---
title: "Final Report - Kallikrein genes"
date: "19.07.2021"
output:
  pdf_document: default
  html_document:
    df_print: paged
header-includes:
   - \usepackage{float}
   - \usepackage{caption}
   - \captionsetup[figure]{font=footnotesize}
---
\thispagestyle{empty}
\hrule
\vspace{0.3cm}
\begin{center}
	\vspace{1cm}
  \large
	\begin{tabular}[c]{l}
	 \\

Anouk Dupe, David Eckey, Dustin Schilling, Maria Yemane \\
\\
	 \\
	Supervisor:
	Dr. Maria Dinkelacker \\
	 \\
	Tutor:
	Nils Mechtel \\
	 \\
	 \\
	Data Analysis for students of Molecular Biotechnology \\
	\\
	Heidelberg University 
	\vspace{0.3cm} \\
	\end{tabular}
	\end{center}

---

\pagenumbering{gobble}
\pagebreak
\tableofcontents
\pagebreak
\pagenumbering{arabic}

```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8)
```

```{r Load Libaries, include=FALSE}
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(hexbin)
library(ggplot2)
library(pheatmap)
library(dendextend) #install.packages("dendextend")
library(factoextra) #install.packages("factoextra")
library(gtools) #install.packages("gtools")
library(plyr)
library(tidyverse) #install.packages("tidyverse")
library(ggfortify)
library(patchwork)
library(RColorBrewer) #install.packages("RColorBrewer")
#library(dendextend)#install.packages("dendextend")
library(float) #install.packages("float")
library(ggdendro) #install.packages("ggdendro")
library(here)
#set the top level folder with here
here::here()
```


```{r Read CEL Files, eval=FALSE, include=FALSE}
#creating a list of the CEL file names, for all three datasets
setwd(here("Rawdata/GSE65216 breast cancer TNBC Her2 LumA LumB"))
cels.breast <- list.files(pattern = 'CEL')

setwd(here("Rawdata/GSE149507 lung cancer"))
cels.lung <- list.files(pattern = 'CEL')
```

```{r Affybatch, eval=FALSE, include=FALSE}
#Breast 
breast <- ReadAffy(filenames = paste(here("Rawdata/GSE65216 breast cancer TNBC Her2 LumA LumB"), cels.breast, sep = "/"), verbose = TRUE)
breast@cdfName <- "HGU133Plus2_Hs_ENST"

orig.breast=colnames(exprs(breast))
new.breast <- substr(orig.breast, 1, nchar(cels.breast)-4)
colnames(exprs(breast)) <- new.breast
rownames(breast@phenoData) <- new.breast
rownames(breast@protocolData) <- new.breast

#Lung
lung <- ReadAffy(filenames = paste(here("Rawdata/GSE149507 lung cancer"), cels.lung, sep = "/"), verbose = TRUE)
lung@cdfName <- "HGU133Plus2_Hs_ENST"

orig.lung=colnames(exprs(lung))
new.lung <- substr(orig.lung, 1, nchar(cels.lung)-4)
colnames(exprs(lung)) <- new.lung
rownames(lung@phenoData) <- new.lung
rownames(lung@protocolData) <- new.lung

```

```{r Load in Data, eval=FALSE, include=FALSE}
breast.vsnrma <- vsnrma(breast)
breastExprs = exprs(breast.vsnrma)

lung.vsnrma <- vsnrma(lung)
lungExprs = exprs(lung.vsnrma)
```


```{r Read ensembl, eval=FALSE, include=FALSE}

setwd(here("Tables"))

ensembl_df = read.csv("ensembl.103.txt",sep="\t")
```

```{r Dataset breast cancer, eval=FALSE, include=FALSE}
breastExprs <- breastExprs[grepl("ENST", rownames(breastExprs)), ]
dim(breastExprs) #  95659    20

# remove .x_at suffix
rownames(breastExprs) <- unlist(lapply(strsplit(rownames(breastExprs), split = "\\."), "[", 1))

setwd(here("Scripts"))

#Setting up annotation
transcriptIDs <- as.character(ensembl_df[, "Transcript.stable.ID"])
geneSymbol <- as.character(ensembl_df[, "HGNC.symbol"])
names(geneSymbol) <- transcriptIDs

# replace ensemble IDs with HGNC symbol
chipIDs.breast <- rownames(breastExprs)
noMatchIDs.breast <- chipIDs.breast[!chipIDs.breast %in% transcriptIDs]
# transcript IDs that are not in ensemble_103.txt
length(noMatchIDs.breast) #112

newChipIDs.breast <- chipIDs.breast[chipIDs.breast %in% transcriptIDs]
breastExprs <- breastExprs[newChipIDs.breast, ]
dim(breastExprs) ## 95547    10

symbol <- geneSymbol[newChipIDs.breast]
symbol.breast <- geneSymbol[newChipIDs.breast]
rownames(breastExprs) <- make.names(as.character(symbol.breast), unique = TRUE)  # Matrix to work with

head(breastExprs)
```


```{r Dataset lung cancer, eval=FALSE, include=FALSE}
lungExprs <- lungExprs[grepl("ENST", rownames(lungExprs)), ]
dim(lungExprs) #  95659    20
 
# remove .x_at suffix
rownames(lungExprs) <- unlist(lapply(strsplit(rownames(lungExprs), split = "\\."), "[", 1))

setwd(here("Scripts"))

# replace ensemble IDs with HGNC symbol
chipIDs.lung <- rownames(lungExprs)
noMatchIDs.lung <- chipIDs.lung[!chipIDs.lung %in% transcriptIDs]
# transcript IDs that are not in ensemble_103.txt
length(noMatchIDs.lung) #112

newChipIDs.lung <- chipIDs.lung[chipIDs.lung %in% transcriptIDs]
lungExprs <- lungExprs[newChipIDs.lung, ]
dim(lungExprs) ## 95547    10


# 
symbol.lung <- geneSymbol[newChipIDs.lung]
rownames(lungExprs) <- make.names(as.character(symbol.lung), unique = TRUE)  # Matrix to work with

head(lungExprs)

```


## 1. Introduction
Tissue-restricted antigens (TRAs) are good drug targets in cancer and cancer immunotherapy. In general, TRAs are genes, which are highly expressed in specific tissue compared to others (Kont et al., 2008). One group of TRAs are Kallikrein genes (KLK).
KLKs are a family of 15 mammalian secreted serine proteases. Analysis has shown that the KLK locus is located on chromosome 19 and forms the largest cluster of contiguous proteases in the entire genome. (Yousef et al., 2000).  
All 15 Kallikrein genes are proteolytic enzymes under steroid hormone regulation and are involved in the regulation of blood pressure, tissue remodeling, skin desquamation, and many other processes. The structure of KLK are similar with two beta-drums, two alpha-helices and a distinct loop involved in the regulation of activity and selectivity. Currently, the specific role of each Kallikrein is unclear. It is known that they are involved in the complex regulatory processes, more specifically in those different signaling cascades.  
Dysregulation of KLKs are frequently associated with cancer. Their expression in different tissues and their involvement in different physiological processes make them potential tumor expression markers (Fischer and Meyer-Hoffert, 2013).   Different expression of Kallikrein genes has been found in many cancer types.


## 2. Quality control

To assure the qualtity of the data the steps  presented in "R Course Micoarray Analysis" by Dr. Maria Dinkelacker (2019) were followed.
The main goal of the quality control is to identify and remove microchips, which show significantly altered gene expression. 
These differences would be difficult to remove via variance stabilizing normalisation (vsn) and could interfere with the rest of the data. The quality control was performed on the breast cancer microarray dataset GSE65216 (Maire et al., 2013) and the small cell lung cancer microarray dataset GSE149507 (Cai et al., 2021).


### 2.1 Quality control - GSE65216 breast cancer

The examination of the individual arrays showed no alteration which indicate physical damage.
The boxplots showed low fluctuation in gene expression for the 20 arrays after normalisation. In addition, none of the chips deviate strongly from each other. In both the density and RNA degradation plot (before and after normalisation). 

### 2.2 Quality control - GSE149507 lung cancer
One of the chips of the small cell lung cancer microarrays displayed non-linear relationships in the scatterplots. 
Since the samples of dataset GSE149507 for normal and carcinoma tissue are linked to one patient each. Therefore,  the two chips GSM4504109_SCLC_05_ca and GSM4504110_SCLC_05_n were replaced. The substitute microarrays were tested again and did not show any discrepancies. 
```{r Broken chip - lung qc, echo=FALSE, fig.align="center", fig.cap="Scatter plot example of broken chip breast cancer GSE65216.", out.width="30%", fig.pos="H", out.extra =''}
knitr::include_graphics("images/broken_chip_lung.jpg")
```

## 3. TRA data
```{r Import TRA data, eval=FALSE, include=FALSE}
setwd(here("TRA Daten"))


# Import of tra.human.5median
tra.human.5median=read.csv(file="Human_protein_atlas_TRA_5median_genes_annotated.tsv", sep=",")

# Import of tra.human
tra.human=read.csv(file="tra.2014.human.5x.table.tsv", sep="\t")

# Import of tra.human.roth
tra.human.roth=read.csv(file="tra.2014.human.roth.5x.table.tsv", sep="\t")

# Import of tra.mouse.4301
tra.mouse.4301=read.csv(file="tra.2014.mouse.4301.5x.table.tsv", sep="\t")

# Import of tra.human.gtex
tra.human.gtex=read.csv(file="tra.2017.human.gtex.5x.table.tsv", sep="\t")

#Import of tra.mouse 
tra.mouse=read.csv(file="tra.2014.mouse.5x.table.tsv",sep="\t")

```

```{r Union of TRA data, eval=FALSE, include=FALSE}
### Filter all Klk from the TRA datasets

# For tra.human  
ind.human=grepl("^KLK",tra.human$gene.symbol)
ind.human =which(ind.human == TRUE)
Klk.TRA1 = as.data.frame(tra.human[ind.human,])

# For tra.human.5median
ind.5median=grepl("^KLK",tra.human.5median$Symbol)
ind.5median =which(ind.5median == TRUE)
Klk.TRA2 = as.data.frame(tra.human.5median[ind.5median,])

# For tra.human.roth
ind.roth=grepl("^KLK",tra.human.roth$gene.symbol)
ind.roth =which(ind.roth == TRUE)
Klk.TRA3 = as.data.frame(tra.human.roth[ind.roth,])

# For tra.mouse
ind.mouse=grepl("^Klk",tra.mouse$gene.symbol)
ind.mouse =which(ind.mouse == TRUE)
Klk.TRA4 = as.data.frame(tra.mouse[ind.mouse,])

# For tra.mouse.4301
ind.4301=grepl("^Klk",tra.mouse.4301$gene.symbol)
ind.4301 =which(ind.4301 == TRUE)
Klk.TRA5 = as.data.frame(tra.mouse.4301[ind.4301,])

# For tra.human.gtex 
ind.gtex=grepl("^KLK",tra.human.gtex$ensembl.symbol)
ind.gtex=which(ind.gtex == TRUE)
Klk.TRA6=as.data.frame(tra.human.gtex[ind.gtex,])



### Creating a unified TRA dataset, that only contains important information for our overview
nrow.x = nrow(Klk.TRA1) + nrow(Klk.TRA2) + nrow(Klk.TRA3) +nrow(Klk.TRA4)+ nrow(Klk.TRA5) + nrow(Klk.TRA6)
Union.TRA = as.data.frame(matrix(ncol = 6, nrow=nrow.x))
union.names <- c("ensembl_transcript","ensembl_genes", "Chromosome", "gene.symbol", "max.tissue", "dataset") ## assigning the unified columns
colnames(Union.TRA) <- union.names 
## now assign the rows, we poorly need to do this manually since there is no strict order of the columns of the 5 TRA datasets
## Watch out!! Chromosomes on the mice are not comparable to humans!!!
Union.TRA[1:nrow(Klk.TRA1),c(1:5)] <- Klk.TRA1[,c(1,2, 7, 3, 11)] # For Klk.TRA1
Union.TRA[nrow(Klk.TRA1)+1:nrow(Klk.TRA2),c(2:5)] <- Klk.TRA2[,c(1, 2, 6, 11)] # For Klk.TRA2
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+1:nrow(Klk.TRA3),c(1:5)] <- Klk.TRA3[,c(1,2, 7, 3, 11)] # For Klk.TRA3
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+1:nrow(Klk.TRA4),c(1:5)] <- Klk.TRA4[,c(1,2, 7, 3, 11)] # For Klk.TRA4
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+nrow(Klk.TRA4)+1:nrow(Klk.TRA5),c(1:5)] <- Klk.TRA5[,c(1,2, 7, 3, 11)] # For Klk.TRA5
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+nrow(Klk.TRA4)+nrow(Klk.TRA5)+1:nrow(Klk.TRA6),c(1:5)] <- Klk.TRA6[,c(1,2,6,3, 10)] # For Klk.TRA6

## now annotate the the dataset
#->> VERY HELPFUL for later plots -> e.g. selecting rows that only apply for the human KLKs: 
#(Union.TRA$dataset == "tra.human")!!!!
Union.TRA[1:nrow(Klk.TRA1), 6] <- "tra.human"
Union.TRA[nrow(Klk.TRA1)+1:nrow(Klk.TRA2),6] <- "tra.human.5median"
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+1:nrow(Klk.TRA3),6] <- "tra.human.roth"
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+1:nrow(Klk.TRA4),6] <- "tra.mouse"
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+nrow(Klk.TRA4)+1:nrow(Klk.TRA5),6] <- "tra.mouse.4301"
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+nrow(Klk.TRA4)+nrow(Klk.TRA5)+1:nrow(Klk.TRA6),6] <- "tra.human.gtex"

# Remove identical entries 
new.union <- rownames(unique(Union.TRA[1:4]))
Union.TRA <- Union.TRA[new.union,] 

### all human KLK genes across all TRA datasets

ind.all.tra.human.KLK <- grep("^tra.human", Union.TRA$dataset)
tra.all.human <- Union.TRA[ind.all.tra.human.KLK,]


# all unique human KLK names that are present as TRAs
all.human.KLK.gene.symbols <- unique(Union.TRA$gene.symbol[grep("tra.human", Union.TRA$dataset)])
all.human.KLK.gene.symbols # These are all the available KLK TRAs we can possibly work with
```


To distinguish between TRA KLK genes and non-TRA KLK genes, a total of 6 TRA datasets were utilized (see appendix). These TRA datasets were than unified, which allowed the extraction of tissue-restricted KLKs according to their transcription number. To get an overview of the distribution of the KLK tissue restriction, a pie chart was conducted. Pie charts allow a quick overview of the proportional distribution. The high prevalence of prostatic kallikrein genes, as well as an occurrence in esophagus, thyroid and salivary gland is notable.
Since six datasets were combined, annotations that differed for the same tissue type were fused.

```{r TRA -piechart, echo=FALSE, fig.align="center", fig.cap="Tissue specificy of KLK genes - KLK genes from six TRA datasets are combined and sorted for tissue specificity", out.width="42%", fig.pos="H", out.extra =''}

knitr::include_graphics("images/piechart_TRA.png")
```

## 4. Expression Analysis
```{r Isolating KLK genes, eval=FALSE, include=FALSE}
# All KLK genes
ind.all.KLK=grep("^KLK",symbol)
Kallikreins.all=symbol[ind.all.KLK]

ind.all.human =which(as.character(symbol) %in% as.character(Kallikreins.all)) 
Kallikreins<- as.character(Kallikreins.all)
Kallikreins

## For breast GSE
all.KLK.breast = breastExprs[ind.all.human,]

## For lung GSE 
all.KLK.lung = lungExprs[ind.all.human,]


ind.KLK.all.human.tra <- which(as.character(names(symbol))  %in% 
as.character(unlist(tra.all.human[1])))

## For breast GSE
TRA.KLK.breast = breastExprs[ind.KLK.all.human.tra,]

## For lung GSE 
TRA.KLK.lung = lungExprs[ind.KLK.all.human.tra,]
```

## 4.1 Breast cancer GSE65216 (Maire et al., 2013)
The breast cancer microarray data GSE65216 (Maire et al., 2013) consists of 20 samples. Respectively, five samples derive from four mutation positive tissue: triple negative breast cancer (TNBC), Her2, Luminal A and Luminal B. Notable, in the microarry data some of the expression values of KLK isoforms were identical. Therefore, the Pearson-correlation was determined between all transcripts. Since isoforms with the correlation of one did not contain additional information, all of the identical isofroms besindes one were removed. In the end, 39 identical isoforms are removed, leaving 73 KLK transcripts for the 15 KLK genes for further analysis. Out of the 73 isoforms, 63 are TRAs, while only 10 are regarded as tissue restricted. Furthermore, the KLKs are sorted after their names in ascending order for the later visualization.
```{r eval=FALSE, include=FALSE}
# transform dataset - genes on columns
df.TRA.KLK.breast <- data.frame(t(TRA.KLK.breast))

# remove identical columns
TRA.KLK.breast.clean <- genes.cleanup(df.TRA.KLK.breast)
dim(TRA.KLK.breast.clean)

# amount of identical columns
cor.TRA.KLK.breast <- cor.genes(df.TRA.KLK.breast)
length(which(cor.TRA.KLK.breast == 1))
```


```{r cleanup identical transcripts - breast cancer, eval=FALSE, include=FALSE}
# Define function for calculating correlation 
cor.genes <- function(df){
  df.cor <- cor(df, method = "pearson")
  diag(df.cor)=NA
  df.cor[upper.tri(df.cor)]=NA
  return(data.frame(df.cor))
}

# cleanup function 
genes.cleanup <- function(df){
  df[!duplicated(unclass(df))]
}

# check clean dataframe for correlations
length(which(cor.genes(TRA.KLK.breast.clean)==1)) #0 correlations

# First we need to clean up the isoforms again
df.all.KLK.breast <- data.frame(t(all.KLK.breast))

# Apply on all KLK dataset
cor.all.KLK.breast <- cor.genes(df.all.KLK.breast)
length(which(cor.all.KLK.breast == 1))
#Identical transcripts can cause issues later on for statistical testing etc. 

# remove identical isoforms
all.KLK.breast.clean <- df.all.KLK.breast[!duplicated(unclass(df.all.KLK.breast))]
dim(all.KLK.breast.clean) #removed 39 identical isoforms

# Use mixedsort function from the gtool package to sort the KLKs correctly
order.breast <- mixedsort(colnames(all.KLK.breast.clean))
order.breast <- order.breast[c(1:3,11:17,4:10,18,19,24:27,20:23,28:73)]
all.KLK.breast.clean <- t(all.KLK.breast.clean[,order.breast]) #sorted all.KLK.breast.clean
all.KLK.breast.clean <- data.frame(all.KLK.breast.clean)
# correct some of the ordering
order.breast1 <- mixedsort(colnames(TRA.KLK.breast.clean))
order.breast1 <- order.breast1[c(1:3,11:17,4:10,18,19,23:26,20:22,27:63)]
TRA.KLK.breast.clean <- t(TRA.KLK.breast.clean[,order.breast1])
TRA.KLK.breast.clean <- data.frame(TRA.KLK.breast.clean)
```

### Histogram
```{r eval=FALSE, include=FALSE}
gene.summary <- function(x){
 round(apply(x, 2, summary), digits = 2)
}
gene.summary(breastExprs)[,1]
```

The histogram represent the frequency of the present gene expression in breast cancer samples. It is conspicuous, that the median gene expression of KLKs is much lower than the overall median gene expression. This means that most of the KLK gene expression is normally down-regulated in relation to the whole genome (Yousef et al., 2004).
```{r Histogram - breast , echo=FALSE, fig.align="center", fig.cap="Histogram of breast cancer gene expression.", out.width="35%", fig.pos="H", out.extra =''}
knitr::include_graphics("images/Histogram_breast.png")
```


### Boxplots
The boxplots confirm the fairly low gene expression of KLKs. There are only two isoforms that exceed the median of the whole genome expression of the breast cancer set, KLK4.4 and KLK8.8. KLK4 gene expression was found by Schmitt et al. to be up-regulated in breast cancer tissue as in comparison to healthy breast tissue. Thereby, KLK4.4 is part of the further analysis. In contrast to that, KLK8 seems to be higher expressed in both normal and cancer tissue (Schmitt et al., 2013).

```{r Boxplot - breast , echo=FALSE, fig.align="center", fig.cap="Boxplot of KLK gene expression in breast cancer.", out.width="50%", fig.pos="H", out.extra =''}
knitr::include_graphics("images/Boxplot_breast.png")
```



### Heatmap

In figure X, KLK4.4 forms its own branch independent of all the others. As already shown in the boxplots, KLK4.4 was distinctly up-regulated. To increase the clarity of the heatmap, KLKs are separated into 3 clusters.
```{r Dendrogram - breast , echo=FALSE, fig.align="center", fig.cap="Dendrogram of KLK genes in breast cancer. Clustering is performed after the complete-linkage method. The genes are separated into 3 clusters.", out.width="50%", fig.pos="H", out.extra =''}
knitr::include_graphics("images/Dendrogram_breast.png")
```
Once again, KLK4.4 clearly stands out (cluster 3) with an overall up-regulated gene expression across all samples. In addition, KLK4.4 belongs to the TRA group. Moreover, gene expression in cluster 1 is higher than within the second cluster. There are only few samples which seem to have up-regulated KLK transcrips for certain mutation types. For instance, the samples Tum01_TNBC and Tum016_TNBC got one of the highest expression values across all the KLKs for KLK10.3, KLK6 and KLK5. As well as the samples Tum71_LumA and Tum76_LumA for the transcripts KLK11 and KLK11.2.
```{r Heatmap - breast , echo=FALSE, fig.align="center", fig.cap="Heatmap of KLK gene expression in breast cancer. The samples are annotated corresponding to their mutation type. Additionally, the KLKs are differentiated by their cluster and potential tissue restriction.", out.width="45%", fig.pos="H", out.extra =''}
knitr::include_graphics("images/Heatmap_breast.png")
```

### Principal component analysis
The principal component analysis (PCA) reduces multidimensional datasets into principle components with proportional variance. In this analysis, PCA was executed over the samples. Scaling was not included, due to the data being vsn-normalized. The cumulative variance of the first two principal components (PCs) yield 72\% of the total variance. 
```{r PCA plot - breast , echo=FALSE, fig.align="center", fig.cap="PC1 is plotted against PC2. The upper part shows the distribution of the breast cancer samples annotated by their mutation type, while the lower part depicts the 12 highest loadings of the KLK genes. Centering was enabled, scaling was not included.", out.width="50%", fig.pos="H", out.extra =''}
knitr::include_graphics("images/PCAplot_breast.png")
```
The loadings consists of the the top twelve most differentiated KLK isoforms. This was conducted by adding absolute values of the rotation matrix for each individual KLK isoform. Some samples are more characterized by the expression of KLK11 and KLK11.2. This is mostly the case for Tum71_LumA and Tum76_LumA samples, just as in the heatmap.  Another finding of the PCA is that TNBC mutations are affected by KLK5 and KLK6 expression.  

### K-means clustering
In order to draw conclusions on characteristics and distribution of different KLKs, k-means was performed. The optimal number of clusters k was determined with the elbow method. For different cluster counts the respective within sum of squares (WSS) was computed, a sudden decrease results in a kink. In this case, the optimal number of clusters is six. In figure X two of the six clusters are clearly separated. Cluster 1 contains KLK4.4 and KLK8.8, while cluster 4 contains KLK5, KLK5.3, KLK6 and KLK10.3. The respective KLKs out of these two clusters will be further analyzed.   

```{r K-means plot - breast , echo=FALSE, fig.align="center", fig.cap="K-means cluster analysis with k = 6 clusters for the breast cancer dataset", out.width="35%", fig.pos="H", out.extra =''}

knitr::include_graphics("images/kmeans_6_breast.png")

```

``` {r k-means for breast cancer dataset, eval=FALSE, include=FALSE}


wssplot.breast <- function(breastExprs, nc=15, seed=1234)
{
  wss <- (nrow(breastExprs)-1)*sum(apply(breastExprs,2,var))
  for(i in 2:nc){
    set.seed(seed)
    wss[i] <- sum(kmeans(breastExprs, centers = i)$withinss)}
  plot(1:nc, wss, type="b", xlab="Number of Clusters", ylab="Within groups sum of squares")
}
  

wssplot.breast(breastExprs)



km.breast <- kmeans(scale(breastExprs),3, nstart=25)
km.breast$cluster

fviz_cluster(km.breast, data=breastExprs, 
             palette =c("#2E9FDF","#00AFBB", "#FC4E07"), 
             geom = "point", 
             ellipse.type = "convex",
             ggtheme = theme_bw()
)


#k-means analysis fpr klks from breastcancer
#(all.KLK.breast.clean.noGSM)

tf.klk.breast.clean = t(all.KLK.breast.clean.noGSM)
setwd(here("Plots/Plots_breast"))
pdf("wssplot_breast.pdf")
wssplot.klk.breast <- function(tf.klk.breast.clean, nc=15, seed=1234)
{
  wss <- (nrow(tf.klk.breast.clean)-1)*sum(apply(tf.klk.breast.clean ,2,var))
  for(i in 2:nc){
    set.seed(seed)
    wss[i] <- sum(kmeans(tf.klk.breast.clean , centers = i)$withinss)}
  plot(1:nc, wss, type="b", xlab="Number of Clusters", ylab="Within groups sum of squares")
}

wssplot.klk.breast(tf.klk.breast.clean)

dev.off()

setwd(here("Plots/Plots_breast"))
pdf("kmeans6_breast.pdf")
km.klk.breast <- kmeans(scale(tf.klk.breast.clean),6, nstart=25)
km.klk.breast$cluster

fviz_cluster(km.klk.breast, data=tf.klk.breast.clean, 
             palette =c("#009900","#00AFBB", "#FC4E07", "#E7B800", "#003333","#FF66CC"), 
             geom = c("point","text"), 
             ellipse.type = "convex",
             ggtheme = theme_bw(),
             labelsize = 5,
             repel=TRUE,
             main="K-Means-Clustering, GSE65216 (Maire et al.2013)"
)
dev.off()

```


### Hypothesis testing
The expression values of the KLKs obtained from Marie et al. were not normally distributed. Therefore, the non- parametric Wilcoxon-Mann-Whitney test was applied. The method merges the values of the two tested samples and ranks the values in an increasing order, before calculating the p-value.
First, KLK4.4 (TRA) and KLK8.8 (non-TRA) form k-means cluster 1 were significantly higher expressed than all other KLKs.
Those results correspond with the observations from the heatmap and the k-means clustering. 
Cluster 4 (KLK5, KLK5.3, KLK6, KLK10.3) was isolated in the k-means clustering. Significant over-expression could not be confirmed, due to the fact that was no differentiation between the tumor types. 

The main characteristic of the dataset from Marie et al. is the subdivision into the samples with different mutations (Her2, LumA, LumB, TNBC). In Figure X, these genes are shown with the subdivision into the different mutation types. A recurring pattern in Figure X is the significant over-expression of TNBC compared with Her2. This observation includes KLK5, KLK5.3, KLK10, KLK10.3. 

```{r, Hypothesis test panel plot with significant bars, echo=FALSE, fig.align="center", fig.cap="Panel plot of the PCA loading genes with significant bars. *: p-value <= 0.5, **: p-value <= 0.01 ", out.width="52%", fig.pos="H", out.extra =''}
knitr::include_graphics("images/breast_panel_loadings_test.png")

```



## 4.2 Lung cancer GSE149507 (Cai et al., 2021)
The lung cancer microarray GSE149507 (Cai et al., 2021) derives from six patients with small cell lung cancer. The dataset consists of a total of twelve samples. Carcinoma tissue and healthy lung tissue, which is adjacent to the carcinoma, make up six samples each.  

### Boxplot
Most of the KLK boxplots are lower than the overall median gene expression and thereby clearly down-regulated (Yousef et al., 2004). KLK4.4 clearly stands out again as the highest expressed KLK gene. The boxplot demonstrates that KLK12 and its isoforms have a high variance and their expression patterns are similar. A possible reason is that the lung cancer dataset consists of both normal and healthy tissue, as in comparison to the breast cancer dataset. In this case, KLK12 and its isofroms a subject for further investigation to determine whether they are differently expressed between normal and carcinoma samples.

```{r Boxplot - lung , echo=FALSE, fig.align="center", fig.cap="Boxplot of KLK gene expression in lung cancer.", out.width="50%", fig.pos="H", out.extra =''}
knitr::include_graphics("images/Boxplot_lung.png")
```
    

### Heatmap

The lung dataset was split into three clusters with the same method used for the breast cancer dataset. In addition, the samples are clustered according to their tissue type being lung carcinoma or healthy tissue. In the dendrogram of the sample type it is striking that the normal samples are clustered into one group with additionally two more cancer samples. Whereas, the other four cancer samples all form their own distinct group. The clustering of the samples clearly reflects itself in the KLK11 and KLK12 gene expression. While KLK4.4 is higher expressed for both normal and carcinoma samples, KLK11 and KLK12 isoforms are mainly higher expressed for the carcinoma sample. The only exception are the already mentioned carcinoma samples SCLC_01 and SCLC_03.

Four out of the six cancer samples have slightly up-regulated KLK11 values. The significance will be tested. The two aforementioned carcinoma samples SCLC_01 and SCLC_03 even got down-regulated KLK11 expression.

```{r Heatmap - lung , echo=FALSE, fig.align="center", fig.cap="Heatmap of KLK gene expression in breast cancer. Carcinoma and normal samples are annotated. Additionally, the KLKs are differentiated by their cluster and potential tissue restriction.", out.width="50%", fig.pos="H", out.extra =''}
knitr::include_graphics("images/Heatmap_lung.png")
```


### PCA
The first two PCs explain 84\% of the total variance. This high cumulative variance indicates that the majority of the variance can be explained by only a few transcripts. As expected, the PCA shows a clear separation between normal and carcinoma samples. Considering the top loadings, four of the cancer samples are characterized by KLK12, while the other two tumor samples SCLC_01_ca and SCLC_03_ca are mainly represented by KLK4.4 and KLK6 expression. 

As in the heatmap, four out of the six cancer samples have up-regulated KLK12 expression values, the two cancer samples SCLC_01_ca and SCLC_03_ca form an exception. They are rather defined by KLK6 and KLK4.4 expression.

```{r PCA plot - lung , echo=FALSE, fig.align="center", fig.cap="PC1 is plotted against PC2. The upper part shows the distribution of the lung cancer samples annotated by their tissue type, while the lower part depicts the top 7 loadings of the KLKs.", out.width="39%", fig.pos="H", out.extra =''}
knitr::include_graphics("images/PCAplot_lung.png")
```


### Clustering - kmeans
The optimal cluster count was determined with the same method as for the breast cancer dataset and equaled five. Cluster 5 only consists of KLK12 and its isoforms. KLK4.4 and KLK8.8, which were conspicuous in the heatmap, are part of cluster 1. The other three clusters containing genes, which were low expressed in the heatmap, are located next to each other. 

```{r plotswss, eval=FALSE, include=FALSE}


wssplot.lung <- function(lungExprs, nc=15, seed=1234)
{
  wss <- (nrow(lungExprs)-1)*sum(apply(lungExprs,2,var))
  for(i in 2:nc){
    set.seed(seed)
    wss[i] <- sum(kmeans(lungExprs, centers = i)$withinss)}
  plot(1:nc, wss, type="b", xlab="Number of Clusters", ylab="Within groups sum of squares")
}
  

wssplot.lung(lungExprs)



## bekomme code nicht zum laufen, Anmerkung für später: Anzahl cluster determinieren!!!
#code für cluster lungexprs
km.lung <- kmeans(scale(lungExprs),4, nstart=25)
km.lung$cluster

fviz_cluster(km.lung, data=lungExprs, 
             palette =c("#2E9FDF","#00AFBB", "#FC4E07", "#E7B800"), 
             geom = "point",
             ellipse.type = "convex",
             ggtheme = theme_bw()
)

#k-means analysis fpr klks from lungcancer
#(all.KLK.lung.clean.noGSM)
setwd(here("Plots/Plots_lung"))
pdf("wssplot_lung.pdf")
wssplot.klk.lung <- function(all.KLK.lung.clean.noGSM, nc=15, seed=1234)
{
  wss <- (nrow(all.KLK.lung.clean.noGSM)-1)*sum(apply(all.KLK.lung.clean.noGSM ,2,var))
  for(i in 2:nc){
    set.seed(seed)
    wss[i] <- sum(kmeans(all.KLK.lung.clean.noGSM , centers = i)$withinss)}
  plot(1:nc, wss, type="b", xlab="Number of Clusters", ylab="Within groups sum of squares")
}

wssplot.klk.lung(all.KLK.lung.clean.noGSM)

dev.off()

#approx 5 clusters
setwd(here("Plots/Plots_lung"))
pdf("kmeans5_lung.pdf")
km.klk.lung.5 <- kmeans(scale(all.KLK.lung.clean.noGSM),5, nstart=25)
km.klk.lung.5$cluster


fviz_cluster(km.klk.lung.5, data=all.KLK.lung.clean.noGSM, 
             palette =c("#003333","#00AFBB", "#FC4E07", "#FF66CC", "#009900"), 
             geom = c("point","text") ,
             ellipse.type = "convex",
             ggtheme = theme_bw(),
             labelsize = 5,
             repel=TRUE,
             main="K-Means-Clustering, GSE149507 (Cai et al. 2021)"
             )


dev.off()

km.klk.lung.5
which(km.klk.lung.5$cluster == 3)

```

### Hypothesis testing
The results of the PCA and the k-means indicate that for some KLKs the expression differs between the cancerous and normal tissue. Since the KLK expression is not normally distributed, the Wilcoxon signed-rank test was used. 
In figure X plot A, KLK4.4 was significantly higher expressed in cancer tissue. Unlike KLK10.3, which was significantly higher expressed in normal tissue. Plot B shows that KLK12 and its isoforms are significantly higher expressed in cancer tissue. Also, the plot visualizes the high similarity within isoforms because only identical transcripts were removed during the clean up. 
KLKs that characterize normal samples in the PCA are shown in plot C. In this respect, KLKB1.1 and KLKB1.3 are significantly down-regulated in the cancer tissue.
The KLKs which characterized the cancer tissue are shown in plot D. Here, KLK1.3, KLK4.4, and KLK12 were significantly higher expressed in cancerous tissue. 
In summary, five out of seven loadings were found to have a significant expression difference between the tissue types. Those results confirm the clear separation of cancer and normal tissue microchips in the PCA based on KLK expression. In conclusion, the hypothesis tests confirm the clear separation of cancer and normal tissue sample in the PCA based on KLK expression.

```{r , echo=FALSE, fig.align="center", fig.cap="A) Genes from  k-means cluster 1, B) Genes from k-means cluster 4, C) PCA loading genes oriented in the direction of normal tissue, D) PCA lading genes oriented in the direction of cancer tissue. * indicate a p-value > 0.05 between the expression in the different tissue types, upper and lower tail Wilcoxon signed-rank test were used.", out.width="50%", fig.pos="H", out.extra =''}
knitr::include_graphics("images/Expr_diff_kmeans_PCA_lung.png")
```



## 5. Logistsic regression
Kallikrein mRNA or protein expression are already established in clinical practice as biomarkers especially in prostate cancer (Diamandis et al., 1998).
Testing whether the identified genes with a significant expression difference were likely to predict tissue type, logistic regression was chosen.
The basic assumptions for logistic regression are:
1. Independence of errors, every observation has to be separate from the others.
2. Linearity of the continuous variables in logit - the relationship between the variable and their logit transformed outcome should be linear. 
3. Absence of multicollinearity or redundancy.
4. No outliners with a strong influence. 
5. For every independent variable there should be at least ten outcomes (Stoltzfus 2011).

These assumptions reveal the shortcomings of the used data and explain the experienced problems with logistic regression. First, the main limitation of the used dataset is the low number of included microchips. The number of microchips has also been further reduced by splitting the data in a training dataset (eight microchips) and testing dataset (four microchips).
Therefore, expected problems of high standard errors and large beta-coefficients for the independent variables were  encountered when including more then one independent variable. This phenomenon is also called overfit-model.
For most individual genes with a significant expression difference the described problems were encountered. The only exceptions were KLK4.4 and KLK12. But in both cases the p-value was not significant. In contrast, the prediction of these univariant models were surprisingly accurate. The model with KLK4.4 could predict the tissue type of three out of four microchips correctly, whereas the model with KLK12 predicted every tissue type right. 
However, a closer look at the probabilities reveals that these models are not reliable. The probabilities for normal to be cancer tissue were mostly over a quarter, indicating a high uncertainty.   



## 6. Discussion

The aim of this project was to analyze whether the expression of Kallikrein genes in given datasets show potential biomarker characteristics.  
In previous studies regarding breast cancer, a higher expression of KLK8 in TNBC and Her2 positive tumors compared to LumA and LumB positive tumors was reported (Michaelidou et al., 2018). However, the conducted analysis could only confirm significant TNBC over-expression for the isoform KLK8.5 compared to LumA and LumB. Nevertheless, the boxplots of KLK8.4 and KLK8.5 showed the trend of Her2 and TNBC over-expression. 
The expression pattern that TNBC is significantly higher than Her2 (KLK5 and KLK10) could not be validated in the literature. This could be subject for further investigations. In summary, the conducted analysis could partially conform the findings from other research groups. The differences can probably be explained by the small amount of samples used in this analysis. In conclusion Kallikrein gene expression can be used for identifying tumor subtypes and even predict the outcome for a patient (Haritos et al., 2018). 
Analysis of the lung cancer microarray GSE149507 (Cai et al., 2021) demonstrated differences in the expression of some KLKs between the cancerous and normal tissues. KLK4.4 was significantly higher expressed in cancer tissue, although it was not obviously shown in the heatmap. In normal tissue KLK10.3 was expressed significantly higher. The finding of decreased KLK11 expression in lung cancer by Sasaki et al. could not be confirmed. Rather, the median of cancer tissue samples was higher. Although over-expression of KLK12 and its isoforms could not be found in the literature, functional studies showed KLK12 to be an pro-angiogenic factor (Kryzer et al., 2013). 
Regression analysis shows the capability of identifying cancerous tissue according to specific KLKs. Since the sample size was not sufficient enough to include more than one independent variable, it should be increased for further studies.  
In conclusion, our results show the potential of the Kallikrein genes to differentiate between cancerous and healthy tissue, as well as between certain mutation types. Furthermore, the over-expression of certain tissue-restricted KLKs allows for their potential use as drug targets.


\newpage
## References

Ardlie, K.G., Deluca, D.S., Segre, A.V., Sullivan, T.J., Young, T.R., Gelfand, E.T., Trowbridge, C.A., Maller, J.B., Tukiainen, T., Lek, M., et al. (2015). The Genotype-Tissue Expression (GTEx) pilot analysis: Multitissue gene regulation in humans. Science 348, 648-660.  
Borgoño, C.A., and Diamandis, E.P. (2004). The emerging roles of human tissue kallikreins in cancer. Nature Reviews Cancer 4, 876-890.  
Cai, L., Liu, H., Huang, F., Fujimoto, J., Girard, L., Chen, J., Li, Y., Zhang, Y.-A., Deb, D., Stastny, V., et al. (2021). Cell-autonomous immune gene expression is repressed in pulmonary neuroendocrine cells and small cell lung cancer. Communications Biology 4.  
Diamandis, E.P. (1998). Prostate-specific Antigen: Its Usefulness in Clinical Medicine. Trends in Endocrinology & Metabolism 9, 310-316.  
Dinkelacker, M. (2007). A database of genes that are expressed in a tissue-restricted manner to analyse promiscous gene expression in medullary thymic epithelial cells. Diplomarbeit (Albert-Ludwigs-Universitaet).  
Dinkelacker, M. (2019). Chromosomal clustering of tissue restricted antigens. Dissertation (University Heidelberg).
Dubey, A.K., Gupta, U., and Jain, S. (2016). Analysis of k-means clustering approach on the breast cancer Wisconsin dataset. International Journal of Computer Assisted Radiology and Surgery 11, 2033-2047.  
Fischer, J., and Meyer-Hoffert, U. (2013). Regulation of kallikrein-related peptidases in the skin – from physiology to diseases to therapeutic options. Thromb Haemost 110, 442-449.  
Haritos, C., Michaelidou, K., Mavridis, K., Missitzis, I., Ardavanis, A., Griniatsos, J., and Scorilas, A. (2018). Kallikrein-related peptidase 6 (KLK6) expression differentiates tumor subtypes and predicts clinical outcome in breast cancer patients. Clinical and Experimental Medicine 18, 203-213.  
Kont, V., Laan, M., Kisand, K., Merits, A., Scott, H.S., and Peterson, P. (2008). Modulation of Aire regulates the expression of tissue-restricted antigens. Molecular Immunology 45, 25-33.  
Lattin JE, S.K., Su AI, Walker JR, Zhang J, Wiltshire T, Saijo K, Glass CK, Hume DA, Kellie S, Sweet MJ (2008). Expression analysis of G Protein-Coupled Receptors in mouse macrophages. Immunome Res. 4:5.  
Lenga Ma Bonda, W., Iochmann, S., Magnen, M., Courty, Y., and Reverdiau, P. (2018). Kallikrein-related peptidases in lung diseases. Biol Chem 399, 959-971.  
Lonsdale, J., Thomas, J., Salvatore, M., Phillips, R., Lo, E., Shad, S., Hasz, R., Walters, G., Garcia, F., Young, N., et al. (2013). The Genotype-Tissue Expression (GTEx) project. Nature Genetics 45, 580-585.  
Maire, V., Némati, F., Richardson, M., Vincent-Salomon, A., Tesson, B., Rigaill, G., Gravier, E., Marty-Prouvost, B., De Koning, L., Lang, G., et al. (2013). Polo-like Kinase 1: A Potential Therapeutic Option in Combination with Conventional Chemotherapy for the Management of Patients with Triple-Negative Breast Cancer. Cancer Research 73, 813-823.  
Michaelidou, K., Ardavanis, A., and Scorilas, A. (2015). Clinical relevance of the deregulated kallikrein-related peptidase 8 mRNA expression in breast cancer: a novel independent indicator of disease-free survival. Breast Cancer Research and Treatment 152, 323-336.  
Roth, R.B., Hevezi, P., Lee, J., Willhite, D., Lechner, S.M., Foster, A.C., and Zlotnik, A. (2006). Gene expression analyses reveal molecular relationships among 20 regions of the human CNS. Neurogenetics 7, 67-80.  
Sano, A., Sangai, T., Maeda, H., Nakamura, M., Hasebe, T., and Ochiai, A. (2007). Kallikrein 11 expressed in human breast cancer cells releases insulin-like growth factor through degradation of IGFBP-3. Int J Oncol 30, 1493-1498.  
Sasaki, H., Kawano, O., Endo, K., Suzuki, E., Haneda, H., Yukiue, H., Kobayashi, Y., Yano, M., and Fujii, Y. (2006). Decreased Kallikrein 11 Messenger RNA Expression in Lung Cancer. Clinical Lung Cancer 8, 45-48.  
Schmitt, M., Magdolen, V., Yang, F., Kiechle, M., Bayani, J., Yousef, G.M., Scorilas, A., Diamandis, E.P., and Dorn, J. (2013). Emerging clinical importance of the cancer biomarkers kallikrein-related peptidases (KLK) in female and male reproductive organ malignancies. Radiology and Oncology 47, 319-329.  
Su, A.I., Cooke, M.P., Ching, K.A., Hakak, Y., Walker, J.R., Wiltshire, T., Orth, A.P., Vega, R.G., Sapinoso, L.M., Moqrich, A., et al. (2002). Large-scale analysis of the human and mouse transcriptomes. Proceedings of the National Academy of Sciences 99, 4465-4470.  
Su, A.I., Wiltshire, T., Batalov, S., Lapp, H., Ching, K.A., Block, D., Zhang, J., Soden, R., Hayakawa, M., Kreiman, G., et al. (2004). A gene atlas of the mouse and human protein-encoding transcriptomes. Proceedings of the National Academy of Sciences 101, 6062-6067.  
Tailor, P.D., Kodeboyina, S.K., Bai, S., Patel, N., Sharma, S., Ratnani, A., Copland, J.A., She, J.-X., and Sharma, A. (2018). Diagnostic and prognostic biomarker potential of kallikrein family genes in different cancer types. Oncotarget 9, 17876-17888. 10.18632/oncotarget.24947.  
Uhlen, M., Fagerberg, L., Hallstrom, B.M., Lindskog, C., Oksvold, P., Mardinoglu, A., Sivertsson, A., Kampf, C., Sjostedt, E., Asplund, A., et al. (2015). Tissue-based map of the human proteome. Science 347, 1260419-1260419.  
Yousef, G.M., Chang, A., Scorilas, A., and Diamandis, E.P. (2000). Genomic Organization of the Human Kallikrein Gene Family on Chromosome 19q13.3–q13.4. Biochemical and Biophysical Research Communications 276, 125-133.  
Yousef, G.M., Magklara, A., and Diamandis, E.P. (2000). KLK12 Is a Novel Serine Protease and a New Member of the Human Kallikrein Gene Family—Differential Expression in Breast Cancer. Genomics 69, 331-341.  
Yousef, G.M., Yacoub, G.M., Polymeris, M.E., Popalis, C., Soosaipillai, A., and Diamandis, E.P. (2004). Kallikrein gene downregulation in breast cancer. British Journal of Cancer 90, 167-172.  
Zhang, Y., Bhat, I., Zeng, M., Jayal, G., Wazer, D.E., Band, H., and Band, V. (2006). Human kallikrein 10, a predictive marker for breast cancer. 387, 715-721.  