---
title: "Klk_Genes_Data_Analysis"
author: "Anouk Dupe, Maria Yemane, Dustin Schilling, David Eckey"
date: "4 5 2021"
output: html_document
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

###load library
```{r Load Libaries, hide=TRUE}
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(hexbin)
library(ggplot2)

library(pheatmap)

###here package 
library(here)
#set the top level folder with here
here::here()
```

```{r Annotation Data Sets, eval=FALSE, include=FALSE}

#Cancer data Sets 
###"rawdata breast GSE27830": Nagel JH, Peeters JK, Smid M, Sieuwerts AM et al. Gene expression profiling assigns CHEK2 1100delC breast cancers to the luminal intrinsic subtypes. Breast Cancer Res Treat 2012 Apr 
#(Nagel et al. 2012)

###"GSE65216 breast cancer": Maire V, Némati F, Richardson M, Vincent-Salomon A et al. Polo-like kinase 1: a potential therapeutic option in combination with conventional chemotherapy for the management of patients with triple-negative breast cancer. Cancer Res 2013 Jan 15 
#(Maire et al. 2013)

###"GSE149507 lung cancer": Cai L, Liu H, Huang F, Fujimoto J et al. Cell-autonomous immune gene expression is repressed in pulmonary neuroendocrine cells and small cell lung cancer. Commun Biol 2021 Mar 9
#(Cai et al. 2021)



#To cite the origin of the TRA data
### (Su et al. 2002, 2004) ->  mouse Novartis data
####Su et al. 2002, Large-scale analysis of the human and mouse transcriptomes. Proc Natl Acad Sci U S A. 2002 Apr 2
#### AND: Su et al. 2004, A gene atlas of the mouse and human protein-encoding transcriptomes. Proc Natl Acad Sci U S A. 2004 Apr 20

###(Roth et al. 2008) -> human Roth data
####Roth et al. 2006, Gene expression analyses reveal molecular relationships among 20 regions of the human CNS. Neurogenetics. 2006 May

### (Lattin et al. 2006) -> mouse Lattin data
#### Lattin et al. 2008, Expression analysis of G Protein-Coupled Receptors in mouse macrophages. Immunome Res. 2008 Apr 29

### (human GTEX data 2015) -> RNAseq data
#### GTEX 2013, The Genotype-Tissue Expression (GTEx) project. GTEx Consortium. Nat Genet. 2013 Jun. 

###
#### GTEX 2015, Human genomics. The Genotype-Tissue Expression (GTEx) pilot analysis: multitissue gene regulation in humans. GTEx Consortium. Science. 2015 May 8

### Uhlén et al. 2015, Proteomics. Tissue-based map of the human proteome. Science. 2015 Jan 23

```



###Read .CEL Files
```{r Read CEL Files, hide=}
#creating a list of the CEL file names, for all thre data sets
setwd(here("Rawdata/rawdata breast GSE27830"))
cels.rawdata <- list.files(pattern = 'CEL')

setwd(here("Rawdata/GSE65216 breast cancer TNBC Her2 LumA LumB"))
cels.breast <- list.files(pattern = 'CEL')

setwd(here("Rawdata/GSE149507 lung cancer"))
cels.lung <- list.files(pattern = 'CEL')


```


```{r Addybatch, hide=}
#read the CEL files into an Affybatch

#Rawdata
rawdata <- ReadAffy(filenames = paste(here("Rawdata/rawdata breast GSE27830"), cels.rawdata, sep = "/"), verbose = TRUE)
rawdata@cdfName <- "HGU133Plus2_Hs_ENST"


orig.rawdata=colnames(exprs(rawdata))
new.rawdata <- substr(orig.rawdata, 1, nchar(cels.rawdata)-4)
colnames(exprs(rawdata)) <- new.rawdata
rownames(rawdata@phenoData) <- new.rawdata
rownames(rawdata@protocolData) <- new.rawdata

save(rawdata, file = paste0(paste(here("Tables")), "/rawdata.rda"))



#Breast 
breast <- ReadAffy(filenames = paste(here("Rawdata/GSE65216 breast cancer TNBC Her2 LumA LumB"), cels.breast, sep = "/"), verbose = TRUE)
breast@cdfName <- "HGU133Plus2_Hs_ENST"

orig.breast=colnames(exprs(breast))
new.breast <- substr(orig.breast, 1, nchar(cels.breast)-4)
colnames(exprs(breast)) <- new.breast
rownames(breast@phenoData) <- new.breast
rownames(breast@protocolData) <- new.breast

save(breast, file = paste0(paste(here("Tables")), "/breast.rda"))


#Lung
lung <- ReadAffy(filenames = paste(here("Rawdata/GSE149507 lung cancer"), cels.lung, sep = "/"), verbose = TRUE)
lung@cdfName <- "HGU133Plus2_Hs_ENST"

orig.lung=colnames(exprs(lung))
new.lung <- substr(orig.lung, 1, nchar(cels.lung)-4)
colnames(exprs(lung)) <- new.lung
rownames(lung@phenoData) <- new.lung
rownames(lung@protocolData) <- new.lung

save(lung, file = paste0(paste(here("Tables")), "/lung.rda"))
```
```{r Rawdata breast}
# save normalized version of rawdata before continuing

rawdata.vsnrma <- vsnrma(rawdata)
setwd(here("Tables"))
save.image(file = "rawdataNorm.rda")

```


##Quality control - GSE65216 breast cancer TNBC Her2 LumA LumB
Single chip
```{r Breast - Single Chip control}
# read images for GSE65216 breast cancer TNBC Her2 LumA LumB
setwd(here("Plots/Plots_breast_qc"))
pdf("breast_chip_control.pdf")
image(breast, col=rainbow(100,start=0,end = 0.75)[100:1])
dev.off()

# the individual chips look fine
```


```{r Breast - Normalization}
## Normalization
breast.vsnrma <- vsnrma(breast)

setwd(here("Tables"))

save.image(file = "breastNorm.rda")

## MeanSdPlot
setwd(here("Plots/Plots_breast_qc"))
msd.breast <- meanSdPlot(breast.vsnrma)
msd.breast.title <- msd.breast$gg + ggtitle("Mean sd GSE65216 (Maire et al. 2013)")

pdf("breast_meansdPlot.pdf")
print(msd.breast.title)
dev.off()
###The line is approximately horizontal, so there shouldn't be a variance-mean dependence 


```

```{r Breast - Boxplots}

##Boxplot before normalization

x11(w=10, h=7)
par(las=2)

mmi=c(1,0.7,1.0477939,0.5366749)
par(mai=mmi)

setwd(here("Plots/Plots_breast_qc"))
pdf("breast_box_bfnorm.pdf")
boxplot(breast, horizontal = TRUE , col=rainbow(150),cex.axis=0.38,main="GSE65216 gene expression before normalization", las = 2)
dev.off()


##Boxplot after normalization
x11(w=10, h=7)
par(las=2)
mmi=c(1,0.7,1.0477939,0.5366749)
par(mai=mmi)
pdf("breast_box_afnorm.pdf")
boxplot(exprs(breast.vsnrma),horizontal = TRUE,col=rainbow(150),cex.axis=0.5,main="GSE65216 gene expression after normalization" , las=2)
dev.off()

```

```{r Breast - Density function}
## Density function before normalization
setwd(here("Plots/Plots_breast_qc"))

x11(w=10, h=7)
pdf("breast_dlogf_bfnorm.pdf")
hist(breast, col=rainbow(150), main = "GSE65216 (Maire et al. 2013) density function before normalization")
dev.off()



## Density function after normalization
##vsnrma
breastExprs = exprs(breast.vsnrma) ####IMPORTANT

pdf("breast_dflog_afnorm.pdf")
plot(density(breastExprs[,1]), type="n", xlab="log Intensity", ylim=c(0,1), main = "GSE65216 (Maire et al. 2013) density function after normalization")
for(i in 1:ncol(breastExprs)){
  lines(density(breastExprs[,i]), col=rainbow(150)[i])
}
dev.off()


```

```{r Breast - RNA deg Plot}

## RNA degeneration plot
setwd(here("Plots/Plots_breast_qc"))
rnadeg.breast = AffyRNAdeg(breast)

pdf("breast_RNAdeg_ss.pdf")
plotAffyRNAdeg(rnadeg.breast, col = rainbow(150))
title(sub="GSE65216 (Maire et al. 2013)")
dev.off()

pdf("breast_RNAdeg_s.pdf")
plotAffyRNAdeg(rnadeg.breast, col = rainbow(150), transform = "shift.only")
title(sub="GSE65216 (Maire et al. 2013)")
dev.off()
```



```{r Breast - Scatter plots}
# Scatter plots
x11(w=9, h=5)
par(mfrow=c(1,2))


setwd(here("Plots/Plots_breast_qc"))

pdf("breast_scatter.pdf")  
for(i in 1:21){
plot(breastExprs[,c(i, i+1)], pch=".")
abline(0,1,col="red")

title(main=paste("Scatterplot of probe", substr(colnames(breast.vsnrma)[i],1,
nchar(colnames(breast.vsnrma)[i])),"and",substr(colnames(breast.vsnrma)[i+1],1,
nchar(colnames(breast.vsnrma)[i+1])), sep = " ", collapse = NULL))

permission = readline(prompt = "type y for next image: ")

if(permission == "y"){
  
}
else{
break}
}
dev.off()

file.name=paste("scatterplot_", as.character(substr(colnames(breast.vsnrma)[i], 1,
nchar(colnames(breast.vsnrma)[i]))),"_",as.character(substr(colnames(breast.vsnrma)[i+1],1,
nchar(colnames(breast.vsnrma)[i+1]))),".eps",sep="")





## None of the scatterplots seem to deviate significantly from the red line

```



# Quality control - GSE149507 lung cancer
```{r Lung - chip control}
# read images for GSE149507 lung cancer
setwd(here("Plots/Plots_lung_qc"))
pdf("lung_chip_control.pdf") 
image(lung, col=rainbow(100,start=0,end = 0.75)[100:1])
dev.off()
# the individual chips look fine
```


```{r Lung - Normalization and meansd Plot}
## Normalization
lung.vsnrma <- vsnrma(lung)

setwd(here("Tables"))

save.image(file = "lungNorm.rda")

## MeanSdPlot
setwd(here("Plots/Plots_lung_qc"))

msd.lung <- meanSdPlot(lung.vsnrma)
msd.lung.title <- msd.lung$gg + ggtitle("Mean sd GSE149507 (Cai et al. 2021)")

pdf("breast_meansdPlot.pdf")
print(msd.lung.title)
dev.off()

###The line is approximately horizontal, so there shouldn't be a variance-mean dependence 
```

```{r Lung - Boxplots}
setwd(here("Plots/Plots_lung_qc"))

##Boxplot before normalization
x11(w=10, h=7)
par(las=2)
mmi=c(1,0.7,1.0477939,0.5366749)
par(mai=mmi)

pdf("lung_box_bfnorm.pdf")
boxplot(lung, horizontal = TRUE ,col=rainbow(150),cex.axis=0.5,main="GSE149507 gene expression before normalization", las = 2)
dev.off()

##Boxplot after normalization
x11(w=10, h=7)
par(las=2)
mmi=c(1,0.7,1.0477939,0.5366749)
par(mai=mmi)

pdf("lung_box_afnorm.pdf")
boxplot(exprs(lung.vsnrma), horizontal = TRUE, col=rainbow(150),cex.axis=0.5,main="GSE149507 gene expression after normalization" , las=2)
dev.off()
```

```{r Lung- Density function}
## Density function before normalization
setwd(here("Plots/Plots_lung_qc"))

x11(w=10, h=7)
pdf("lung_df_bfnorm.pdf")
hist(lung, col=rainbow(150), main = "GSE149507 (Cai et al. 2021) density function before normalization")
dev.off()


## Density function after normalization
##vsnrma
lungExprs = exprs(lung.vsnrma) ####IMPORTANT

pdf("lung_df_afnorm.pdf")
plot(density(lungExprs[,1]), type="n", xlab="log Intensity", ylim=c(0,1), main = "GSE149507 (Cai et al. 2021) density function after normalization")
for(i in 1:ncol(lungExprs)){
  lines(density(lungExprs[,i]), col=rainbow(150)[i])
}
dev.off()
```

```{r Lung -RNA deg}

## RNA degeneration plot
setwd(here("Plots/Plots_lung_qc"))
rnadeg.lung = AffyRNAdeg(lung)

pdf("lung_RNAdeg_ss.pdf")
plotAffyRNAdeg(rnadeg.lung, col = rainbow(150))
title(sub="GSE149507 (Cai et al. 2021) before normalization")
dev.off()

pdf("lung_RNAdeg_s.pdf")
plotAffyRNAdeg(rnadeg.lung, col = rainbow(150), transform = "shift.only")
title(sub="GSE149507 (Cai et al. 2021)")
dev.off()
```



```{r Lung - Scatter plots}
# Scatter plots
setwd(here("Plots/Plots_lung_qc"))
x11(w=9, h=5)
par(mfrow=c(1,2))

pdf("lung_scatter.pdf")
for(i in 1:12){
  
plot(lungExprs[,c(i, i+1)], pch=".")
abline(0,1,col="red")

title(main=paste("Scatterplot of probe", substr(colnames(lung.vsnrma)[i],1,
nchar(colnames(lung.vsnrma)[i])),"and",substr(colnames(lung.vsnrma)[i+1],1,
nchar(colnames(lung.vsnrma)[i+1])), sep = " ", collapse = NULL))

permission = readline(prompt = "type y for next image: ")

if(permission == "y"){
  
}
else{
break}
}
dev.off()

##GSM4504108 or GSM4504109 might be broken
#GSM4504109 or GSM4503110 too
setwd(here("Plots/Plots_lung_qc"))

pdf("lung_scatter_brokenchip1.pdf")
file.name1=paste("scatterplot_", as.character(substr(colnames(lung.vsnrma)[i], 1,
nchar(colnames(lung.vsnrma)[i]))),"_",as.character(substr(colnames(lung.vsnrma)[i+1],1,
nchar(colnames(lung.vsnrma)[i+1]))),".eps",sep="")
dev.off()



## maybe the plot GSM4504109_SCLC_0 might be broken:
setwd(here("Plots/Plots_lung_qc"))
pdf("lung_scatter_brokenchip2.pdf")
for(i in 1:12){
  
plot(lungExprs[,c(9, i)], pch=".")
abline(0,1,col="red")

title(main=paste("Scatterplot of probe", substr(colnames(lung.vsnrma)[9],1,
nchar(colnames(lung.vsnrma)[9])),"and",substr(colnames(lung.vsnrma)[i],1,
nchar(colnames(lung.vsnrma)[i])), sep = " ", collapse = NULL))

permission = readline(prompt = "type y for next image: ")

if(permission == "y"){
  
}
else{
break}
}
dev.off()

###GSM4504109_SCLC_05_ca is broken!!! -> bana-like shaped scatter plots
### Thereby, change the whole patient 05, so chip GSM4504109_SCLC_05_ca and GSM4504110_SCLC_05_n
### Get 2 new chips for patient 5 in Gene Expression omnbius for GSE149507 lung cancer and do quality control over again
```

#Important
```{r}
breastExprs = exprs(breast.vsnrma)
lungExprs = exprs(lung.vsnrma)

rawdata.vsnrma <- vsnrma(rawdata)
rawdataExprs = exprs(rawdata.vsnrma)
```


# Import of TRA data

```{r}
setwd(here("TRA Daten"))


# Import of tra.human.5median
tra.human.5median=read.csv(file="Human_protein_atlas_TRA_5median_genes_annotated.tsv", sep=",")

# Import of tra.human
tra.human=read.csv(file="tra.2014.human.5x.table.tsv", sep="\t")

# Import of tra.human.roth
tra.human.roth=read.csv(file="tra.2014.human.roth.5x.table.tsv", sep="\t")

# Import of tra.mouse.4301
tra.mouse.4301=read.csv(file="tra.2014.mouse.4301.5x.table.tsv", sep="\t")

# Import of tra.human.gtex
tra.mouse.gtex=read.csv(file="tra.2017.human.gtex.5x.table.tsv", sep="\t")

#Import of tra.mouse 
tra.mouse=read.csv(file="tra.2014.mouse.5x.table.tsv",sep="\t")

```


# Filter all Klk from the TRA data sets

```{r}

# For tra.human  
ind.human=grepl("^KLK",tra.human$gene.symbol)
ind.human =which(ind.human == TRUE)
Klk.TRA1 = as.data.frame(tra.human[ind.human,])

# For tra.human.5median ####DOESNT WORK SINCE CONFLICT DURING IMPORT
ind.5median=grepl("^Klk",tra.human.5median$gene.symbol)
ind.5median =which(ind.5median == TRUE)
Klk.TRA2 = as.data.frame(tra.human.5median[ind.5median,])

# For tra.human.roth
ind.roth=grepl("^KLK",tra.human.roth$gene.symbol)
ind.roth =which(ind.roth == TRUE)
Klk.TRA3 = as.data.frame(tra.human.roth[ind.roth,])

# For tra.mouse
ind.mouse=grepl("^Klk",tra.mouse$gene.symbol)
ind.mouse =which(ind.mouse == TRUE)
Klk.TRA4 = as.data.frame(tra.mouse[ind.mouse,])

# For tra.mouse.4301
ind.4301=grepl("^Klk",tra.mouse.4301$gene.symbol)
ind.4301 =which(ind.4301 == TRUE)
Klk.TRA5 = as.data.frame(tra.mouse.4301[ind.4301,])

# For tra.mouse.gtex ###Here the genes are listed in ensembl.symbol 
#-> should rearrange all genes in same columns before putting it all in one data frame 
ind.gtex=grepl("^KLK",tra.mouse.gtex$ensembl.symbol)
ind.gtex=which(ind.gtex == TRUE)
Klk.TRA6=as.data.frame(tra.mouse.gtex[ind.gtex,])

```

#read in ensembl annotation file
```{r}

setwd(here("Tables"))

ensembl_df = read.csv("ensembl.103.txt",sep="\t")
```


# Creating rawdata breast cancer data set to work with
```{r}
rawdataExprs <- rawdataExprs[grepl("ENST", rownames(rawdataExprs)), ]
#dim = 95659    10

# remove .x_at suffix
rownames(rawdataExprs) <- unlist(lapply(strsplit(rownames(rawdataExprs), split = "\\."), "[", 1))

setwd(here("Scripts"))

#Setting up annotation

transcriptIDs <- as.character(ensembl_df[, "Transcript.stable.ID"])
geneSymbol <- as.character(ensembl_df[, "HGNC.symbol"])
names(geneSymbol) <- transcriptIDs

# replace ensemble IDs with HGNC symbol
chipIDs <- rownames(rawdataExprs)
noMatchIDs <- chipIDs[!chipIDs %in% transcriptIDs] # transcript IDs that are not in ensemble_103.txt
length(noMatchIDs) #112

newChipIDs <- chipIDs[chipIDs %in% transcriptIDs]
rawdataExprs <- rawdataExprs[newChipIDs, ]
dim(rawdataExprs) ## 95547    10


# 
symbol <- geneSymbol[newChipIDs]
rownames(rawdataExprs) <- as.character(symbol)  # Matrix to work with

head(rawdataExprs)

```


# Creating GSE65216 breast cancer (TNBC Her2 LumA LumB) data set to work with
```{r}
breastExprs <- breastExprs[grepl("ENST", rownames(breastExprs)), ]
dim(breastExprs) #  95659    20

# remove .x_at suffix
rownames(breastExprs) <- unlist(lapply(strsplit(rownames(breastExprs), split = "\\."), "[", 1))

setwd(here("Scripts"))

# replace ensemble IDs with HGNC symbol
chipIDs.breast <- rownames(breastExprs)
noMatchIDs.breast <- chipIDs.breast[!chipIDs.breast %in% transcriptIDs]
# transcript IDs that are not in ensemble_103.txt
length(noMatchIDs.breast) #112

newChipIDs.breast <- chipIDs.breast[chipIDs.breast %in% transcriptIDs]
breastExprs <- breastExprs[newChipIDs.breast, ]
dim(breastExprs) ## 95547    10


# 
symbol.breast <- geneSymbol[newChipIDs.breast]
rownames(breastExprs) <- as.character(symbol.breast)  # Matrix to work with

head(breastExprs)

```
# Creating lung cancer data set to work with
```{r}
lungExprs <- lungExprs[grepl("ENST", rownames(lungExprs)), ]
dim(lungExprs) #  95659    20

# remove .x_at suffix
rownames(lungExprs) <- unlist(lapply(strsplit(rownames(lungExprs), split = "\\."), "[", 1))

setwd(here("Scripts"))

# replace ensemble IDs with HGNC symbol
chipIDs.lung <- rownames(lungExprs)
noMatchIDs.lung <- chipIDs.lung[!chipIDs.lung %in% transcriptIDs]
# transcript IDs that are not in ensemble_103.txt
length(noMatchIDs.lung) #112

newChipIDs.lung <- chipIDs.lung[chipIDs.lung %in% transcriptIDs]
lungExprs <- lungExprs[newChipIDs.lung, ]
dim(lungExprs) ## 95547    10


# 
symbol.lung <- geneSymbol[newChipIDs.lung]
rownames(lungExprs) <- as.character(symbol.lung)  # Matrix to work with

head(lungExprs)

```



# Prostate tissue 

```{r}
# Klk genes in any cancer -> filter after tissue or genes?



## Selecting from the vector symbol genes of Klk

# Tissue in prostate as an example -> Here do the process with combined TRA data set

#But firstly executing it on whole TRA mouse dataset
tiss=tra.human[,11]

ind.prostate=which(tiss=="Prostate")

TRA.symbol=tra.human[,3]

prostate.TRA1=TRA.symbol[ind.prostate]

row.ind.prostate=which(as.character(symbol) %in% prostate.TRA1)


## For rawdata
rawdataExprs.prostate = rawdataExprs[row.ind.prostate,] #How genes in prostate Tissue of human TRAs? 
dim(rawdataExprs.prostate) #over the 10 patients there are 151 genes for each 
## Prostate matrix -> "KLK", WATCH OUT, do not write "Klk"

## For breast cancer 
breastExprs.prostate = breastExprs[row.ind.prostate,] #How genes in prostate Tissue of human TRAs? 
dim(breastExprs.prostate)#over the 10 patients there are 151 genes for each 

# For lung cancer
lungExprs.prostate = lungExprs[row.ind.prostate,] #How genes in prostate Tissue of human TRAs? 
dim(lungExprs.prostate)#over the 10 patients there are 151 genes for each 


pheatmap(t(rawdataExprs.prostate))
pheatmap(t(breastExprs.prostate))
pheatmap(t(lungExprs.prostate))

```

# Example Analysis
## KLK genes

## Überlegung: Ich glaube wir sollten nicht nur die Klk gene einzeln nehmen, sondern eher im Gesamtbild eines spezifischen Gewebes wie etwa Prostata oder.. schauen, ob die wenig vorhandenen KLK gene stärker exprimiert werden als andere Gene
## Aufklärung: Die 6 tra... Datensätze sind eher für den spezifischen Gewebetyp gedacht, denke mal hier kann man entweder nach dem gewebe wie prostata filtern und darauf eine Analyse machen (auch wichtig) oder nachschauen in welchem Gewebe die einzelnen TRAs vorhanden sind!! Allerdings nicht zum filtern nach KLK gene geeignet, siehe unten, da dadurch nur 7 KLK gene bei raus kommen
```{r}
## Now how many KLK genes are TRAs
# Use Klk.TRA1 for this, filtered for Klk genes -> use ind.human 


TRA.symbol=tra.human[,3] # lists all TRAs names for humans

KLK.genes.TRA1 = TRA.symbol[ind.human] #ind.human (see chunk 21) filters for KLKs in tra.human

row.ind.klk1=which(as.character(symbol) %in% KLK.genes.TRA1) #row.ind.klk1 gives respective column 

# For first breast cancer data set
data.KLK.rawdata = rawdataExprs[row.ind.klk1,]

dim(data.KLK.rawdata) ## 39 KLK genes in breast cancer rawdata 

setwd(here("Plots/Plots_raw_breast"))
pdf("KLK_TRAs_Heatmap_rawdata.pdf")
pheatmap(data.KLK.rawdata, main = "Expression values of KLK TRAs in human breast cancer GSE27830 ")
## (Nagel et al. 2012)
dev.off()


# For GSE65216 breast cancer
data.KLK.breast = breastExprs[row.ind.klk1,]
dim(data.KLK.breast) ## 39 KLK genes in breast cancer rawdata 

setwd(here("Plots/Plots_breast"))
pdf("KLK_TRAs_Heatmap_breast.pdf")
pheatmap(data.KLK.breast, main = "Expression values of KLK TRAs in human breast cancer GSE65216")
## (Maire et al. 2013)
dev.off()


# For GSE149507 lung cancer
data.KLK.lung = lungExprs[row.ind.klk1,]
dim(data.KLK.lung) ## 39 KLK genes in breast cancer rawdata 

setwd(here("Plots/Plots_lung"))
pdf("KLK_TRAs_Heatmap_lung.pdf")
pheatmap(data.KLK.lung, main = "Expression values of KLK TRAs in human lung cancer GSE149507")
## (Cai et al. 2021)
dev.off()



```

Nicht Sinn der Sache alle KLKs zu nehmen, da TRAs entscheidend
```{r}
### FOR ALL KLK genes

ind.all.KLK=grep("^KLK",symbol)
Kallikreins=symbol[ind.all.KLK]

###FOR rawdata
all.KLK.rawdata = rawdataExprs[ind.all.KLK,] 
### MATRIX data.all.KLK -> all KLK genes in breast cancer
all.KLK.rawdata = t(all.KLK.rawdata)
head(all.KLK.rawdata)
dim(all.KLK.rawdata) #10 112
#112 single KLK genes in breast cancer rawdata

#Lets test out a boxplot
par(las=2)
##Experimental
boxplot(all.KLK.rawdata,col=rainbow(length(rownames(data.KLK.TRA1))),main="gene expression of KLK genes in human breast cancer",cex.axis=0.8)
pheatmap(all.KLK.rawdata)

x1 = cor(data.all.KLK)
head(x1)
pheatmap(x1) # corr between patients
x2 = cor(t(data.all.KLK))
head(x2)
pheatmap(x2)

###FOR breast cancer
all.KLK.breast = breastExprs[ind.all.KLK,] 
all.KLK.breast = t(all.KLK.breast)
dim(all.KLK.breast) #20 112
pheatmap(all.KLK.breast)

###FOR lung cancer
all.KLK.lung = lungExprs[ind.all.KLK,] 
all.KLK.lung = t(all.KLK.lung)
dim(all.KLK.lung) # 112  12
pheatmap(all.KLK.lung)
```


```{r}
##test
ind.HER2 = grep(pattern = "Her2", rownames(all.KLK.breast)) ###GREP FOR Her2
ind.TNBC = grep(pattern = "TNBC", rownames(all.KLK.breast)) ###GREP FOR TNBC
ind.LumA = grep(pattern = "LumA", rownames(all.KLK.breast)) ###GREP FOR LumA
ind.LumB = grep(pattern = "LumB", rownames(all.KLK.breast)) ###GREP FOR LumB


data.new <-rbind(all.KLK.breast[ind.HER2,],all.KLK.breast[ind.TNBC,],all.KLK.breast[ind.LumA,],all.KLK.breast[ind.LumB,])

pheatmap(data.new)
boxplot(data.new,col=rainbow(length(rownames(data.new))),main="gene expression of KLK genes in human breast cancer",cex.axis=0.8)
```


```{r}
ind.all.KLK=grep("^KLK",symbol)

all.KLK.lung = lungExprs[ind.all.KLK,] 

dim(all.KLK.lung) 
```

```{r}
### Expression analysis for lungExprs

# get familiar with dataset
head(lungExprs)
colnames(lungExprs)
rownames(lungExprs)
dim(lungExprs)
class(lungExprs)

#filter for klk,try to save klkexpr as data frame

klkexprs_lung = lungExprs[grep('KLK',rownames(lungExprs)),]
class(klkexprs_lung)
dim(klkexprs_lung)



