---
title: "Klk_Genes_Data_Analysis"
author: "Anouk Dupe, Maria Yemane, Dustin Schilling, David Eckey"
date: "4 5 2021"
output: html_document
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

###load library
```{r Load Libaries, hide=TRUE}
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(hexbin)
library(ggplot2)

library(pheatmap)

library(dendextend) #install.package(dendextend)

library(factoextra)

###here package 
library(here)
#set the top level folder with here
here::here()
```
# 0. Introduction
#Relevance of TRAs in cancer therapy: Tumor restricted antigens only occur in cancer tissue, which makes them a great immunological target. Identifying TRAs overexpressed in specific cancer tissue is imporant for the development of immunotherapeutic drugs....
# Biological question considering KLKs:..

```{r Annotation Data Sets, eval=FALSE, include=FALSE}

#Cancer data Sets 
###"rawdata breast GSE27830": Nagel JH, Peeters JK, Smid M, Sieuwerts AM et al. Gene expression profiling assigns CHEK2 1100delC breast cancers to the luminal intrinsic subtypes. Breast Cancer Res Treat 2012 Apr 
#(Nagel et al. 2012)

###"GSE65216 breast cancer": Maire V, Némati F, Richardson M, Vincent-Salomon A et al. Polo-like kinase 1: a potential therapeutic option in combination with conventional chemotherapy for the management of patients with triple-negative breast cancer. Cancer Res 2013 Jan 15 
#(Maire et al. 2013)

###"GSE149507 lung cancer": Cai L, Liu H, Huang F, Fujimoto J et al. Cell-autonomous immune gene expression is repressed in pulmonary neuroendocrine cells and small cell lung cancer. Commun Biol 2021 Mar 9
#(Cai et al. 2021)



#To cite the origin of the TRA data
### (Su et al. 2002, 2004) ->  mouse Novartis data
####Su et al. 2002, Large-scale analysis of the human and mouse transcriptomes. Proc Natl Acad Sci U S A. 2002 Apr 2
#### AND: Su et al. 2004, A gene atlas of the mouse and human protein-encoding transcriptomes. Proc Natl Acad Sci U S A. 2004 Apr 20

###(Roth et al. 2008) -> human Roth data
####Roth et al. 2006, Gene expression analyses reveal molecular relationships among 20 regions of the human CNS. Neurogenetics. 2006 May

### (Lattin et al. 2006) -> mouse Lattin data
#### Lattin et al. 2008, Expression analysis of G Protein-Coupled Receptors in mouse macrophages. Immunome Res. 2008 Apr 29

### (human GTEX data 2015) -> RNAseq data
#### GTEX 2013, The Genotype-Tissue Expression (GTEx) project. GTEx Consortium. Nat Genet. 2013 Jun. 

###
#### GTEX 2015, Human genomics. The Genotype-Tissue Expression (GTEx) pilot analysis: multitissue gene regulation in humans. GTEx Consortium. Science. 2015 May 8

### Uhlén et al. 2015, Proteomics. Tissue-based map of the human proteome. Science. 2015 Jan 23

```



###Read .CEL Files
```{r Read CEL Files, hide = TRUE}
#creating a list of the CEL file names, for all three data sets
setwd(here("Rawdata/rawdata breast GSE27830"))
cels.rawdata <- list.files(pattern = 'CEL')

setwd(here("Rawdata/GSE65216 breast cancer TNBC Her2 LumA LumB"))
cels.breast <- list.files(pattern = 'CEL')

setwd(here("Rawdata/GSE149507 lung cancer"))
cels.lung <- list.files(pattern = 'CEL')


```

```{r eval=FALSE, include=FALSE}
setwd(here("Rawdata/rawdata breast GSE27830"))
cels.rawdata <- c("GSM687017.CEL", "GSM687018.CEL", "GSM687019.CEL", "GSM687020.CEL", "GSM687021.CEL", "GSM687022.CEL", "GSM687023.CEL", "GSM687024.CEL", "GSM687025.CEL", "GSM687026.CEL")

setwd(here("Rawdata/GSE65216 breast cancer TNBC Her2 LumA LumB"))
cels.breast <- c("GSM1588970_Tum01_TNBC.CEL", "GSM1588972_Tum02_Her2.CEL", "GSM1588974_Tum03_TNBC.CEL",  "GSM1588975_Tum04_TNBC.CEL", "GSM1588976_Tum05_TNBC.CEL", "GSM1588977_Tum06_TNBC.CEL", "GSM1588989_Tum020_Her2.CEL", "GSM1588990_Tum015_Her2.CEL", "GSM1588991_Tum016_Her2.CEL", "GSM1588992_Tum016_Her2.CEL", "GSM1589064_Tum071_LumA.CEL", "GSM1589065_Tum073_LumA.CEL", "GSM1589066_Tum074_LumA.CEL", "GSM1589067_Tum075_LumA.CEL", "GSM1589068_Tum076_LumA.CEL", "GSM1589093_Tum101_LumB.CEL", "GSM1589094_Tum102_LumB.CEL", "GSM1589095_Tum103_LumB.CEL", "GSM1589097_Tum104_LumB.CEL", "GSM1589098_Tum105_LumB.CEL")

setwd(here("Rawdata/GSE149507 lung cancer"))
cels.lung <- c("GSM4504101_SCLC_01_ca.CEL", "GSM4504102_SCLC_01_n.CEL", "GSM4504103_SCLC_02_ca.CEL", "GSM4504104_SCLC_02_n.CEL", "GSM4504105_SCLC_03_ca.CEL", "GSM4504106_SCLC_03_n.CEL", "GSM4504107_SCLC_04_ca.CEL", "GSM4504108_SCLC_04_n.CEL", "GSM4504111_SCLC_06_ca.CEL", "GSM4504112_SCLC_06_n.CEL", "GSM4504113_SCLC_07_ca.CEL", "GSM4504114_SCLC_07_n.CEL" )

```



```{r Addybatch, hide= TRUE}
#read the CEL files into an Affybatch

#Rawdata
rawdata <- ReadAffy(filenames = paste(here("Rawdata/rawdata breast GSE27830"), cels.rawdata, sep = "/"), verbose = TRUE)
rawdata@cdfName <- "HGU133Plus2_Hs_ENST"


orig.rawdata=colnames(exprs(rawdata))
new.rawdata <- substr(orig.rawdata, 1, nchar(cels.rawdata)-4)
colnames(exprs(rawdata)) <- new.rawdata
rownames(rawdata@phenoData) <- new.rawdata
rownames(rawdata@protocolData) <- new.rawdata

save(rawdata, file = paste0(paste(here("Tables")), "/rawdata.rda"))



#Breast 
breast <- ReadAffy(filenames = paste(here("Rawdata/GSE65216 breast cancer TNBC Her2 LumA LumB"), cels.breast, sep = "/"), verbose = TRUE)
breast@cdfName <- "HGU133Plus2_Hs_ENST"

orig.breast=colnames(exprs(breast))
new.breast <- substr(orig.breast, 1, nchar(cels.breast)-4)
colnames(exprs(breast)) <- new.breast
rownames(breast@phenoData) <- new.breast
rownames(breast@protocolData) <- new.breast

save(breast, file = paste0(paste(here("Tables")), "/breast.rda"))


#Lung
lung <- ReadAffy(filenames = paste(here("Rawdata/GSE149507 lung cancer"), cels.lung, sep = "/"), verbose = TRUE)
lung@cdfName <- "HGU133Plus2_Hs_ENST"

orig.lung=colnames(exprs(lung))
new.lung <- substr(orig.lung, 1, nchar(cels.lung)-4)
colnames(exprs(lung)) <- new.lung
rownames(lung@phenoData) <- new.lung
rownames(lung@protocolData) <- new.lung

save(lung, file = paste0(paste(here("Tables")), "/lung.rda"))
```
```{r Rawdata breast}
# save normalized version of rawdata before continuing

rawdata.vsnrma <- vsnrma(rawdata)
setwd(here("Tables"))
save.image(file = "rawdataNorm.rda")

```

# 1. Quality Control

##Quality control - GSE65216 breast cancer TNBC Her2 LumA LumB
###Single chip
```{r Breast - Single Chip control}
# read images for GSE65216 breast cancer TNBC Her2 LumA LumB
setwd(here("Plots/Plots_breast_qc"))
pdf("breast_chip_control.pdf")
image(breast, col=rainbow(100,start=0,end = 0.75)[100:1])
dev.off()

# the individual chips look fine
```

### Normalization
```{r Breast - Normalization}
## Normalization
breast.vsnrma <- vsnrma(breast)

setwd(here("Tables"))

save.image(file = "breastNorm.rda")

## MeanSdPlot
setwd(here("Plots/Plots_breast_qc"))
msd.breast <- meanSdPlot(breast.vsnrma)
msd.breast.title <- msd.breast$gg + ggtitle("Mean sd GSE65216 (Maire et al. 2013)")

pdf("breast_meansdPlot.pdf")
print(msd.breast.title)
dev.off()
###The line is approximately horizontal, so there shouldn't be a variance-mean dependence 


```

### Visualization via boxplots
```{r Breast - Boxplots}

##Boxplot before normalization
setwd(here("Plots/Plots_breast_qc"))

x11(w=10, h=7)
pdf("breast_box_bfnorm.pdf")
par(mai=c(0.8,1.5,0.8,0.8))
boxplot(breast, horizontal = TRUE , col=rainbow(150),cex.axis=0.38,main="GSE65216 (Maire et al. 2013) gene expression before normalization", las = 2)
dev.off()


##Boxplot after normalization
x11(w=10, h=7)
pdf("breast_box_afnorm.pdf")
par(mai=c(0.8,1.5,0.8,0.8))
boxplot(exprs(breast.vsnrma),horizontal = TRUE,col=rainbow(150),cex.axis=0.5,main="GSE65216 (Maire et al. 2013) gene expression after normalization" , las=2)
dev.off()

```

### Density function
```{r Breast - Density function}
## Density function before normalization
setwd(here("Plots/Plots_breast_qc"))

x11(w=10, h=7)
pdf("breast_dlogf_bfnorm.pdf")
hist(breast, col=rainbow(150), main = "GSE65216 (Maire et al. 2013) density function before normalization")
dev.off()



## Density function after normalization
##vsnrma
breastExprs = exprs(breast.vsnrma) ####IMPORTANT

pdf("breast_dflog_afnorm.pdf")
plot(density(breastExprs[,1]), type="n", xlab="log Intensity", ylim=c(0,1), main = "GSE65216 (Maire et al. 2013) density function after normalization")
for(i in 1:ncol(breastExprs)){
  lines(density(breastExprs[,i]), col=rainbow(150)[i])
}
dev.off()


```

### RNA degeneration plot
```{r Breast - RNA deg Plot}

## RNA degeneration plot
setwd(here("Plots/Plots_breast_qc"))
rnadeg.breast = AffyRNAdeg(breast)

pdf("breast_RNAdeg_ss.pdf")
plotAffyRNAdeg(rnadeg.breast, col = rainbow(150))
title(sub="GSE65216 (Maire et al. 2013)")
dev.off()

pdf("breast_RNAdeg_s.pdf")
plotAffyRNAdeg(rnadeg.breast, col = rainbow(150), transform = "shift.only")
title(sub="GSE65216 (Maire et al. 2013)")
dev.off()
```


### Scatter plots
#### Examining scatter plots and looking for individual chips with non-linear relationship that indicate  broken chips.
```{r Breast - Scatter plots}
# Scatter plots
x11(w=9, h=5)
par(mfrow=c(1,2))


setwd(here("Plots/Plots_breast_qc"))

pdf("breast_scatter.pdf")  
for(i in 1:21){
plot(breastExprs[,c(i, i+1)], pch=".")
abline(0,1,col="red")

title(main=paste("Scatterplot of probe", substr(colnames(breast.vsnrma)[i],1,
nchar(colnames(breast.vsnrma)[i])),"and",substr(colnames(breast.vsnrma)[i+1],1,
nchar(colnames(breast.vsnrma)[i+1])), sep = " ", collapse = NULL))

permission = readline(prompt = "type y for next image: ")

if(permission == "y"){
  
}
else{
break}
}
dev.off()

file.name=paste("scatterplot_", as.character(substr(colnames(breast.vsnrma)[i], 1,
nchar(colnames(breast.vsnrma)[i]))),"_",as.character(substr(colnames(breast.vsnrma)[i+1],1,
nchar(colnames(breast.vsnrma)[i+1]))),".eps",sep="")





## None of the scatterplots seem to deviate significantly from the red line

```



## Quality control - GSE149507 lung cancer
### Chip control
```{r Lung - chip control}
# read images for GSE149507 lung cancer
setwd(here("Plots/Plots_lung_qc"))
pdf("lung_chip_control.pdf") 
image(lung, col=rainbow(100,start=0,end = 0.75)[100:1])
dev.off()
# the individual chips look fine
```

### Normalization and meansd Plot
```{r Lung - Normalization and meansd Plot}
## Normalization
lung.vsnrma <- vsnrma(lung)

setwd(here("Tables"))

save.image(file = "lungNorm.rda")

## MeanSdPlot
setwd(here("Plots/Plots_lung_qc"))

msd.lung <- meanSdPlot(lung.vsnrma)
msd.lung.title <- msd.lung$gg + ggtitle("Mean sd GSE149507 (Cai et al. 2021)")

pdf("breast_meansdPlot.pdf")
print(msd.lung.title)
dev.off()

###The line is approximately horizontal, so there shouldn't be a variance-mean dependence 
```

### Boxplots
```{r Lung - Boxplots}
setwd(here("Plots/Plots_lung_qc"))

##Boxplot before normalization
x11(w=10, h=7)
pdf("lung_box_bfnorm.pdf")
par(mai=c(0.8,1.5,0.8,0.8))
boxplot(lung, horizontal = TRUE ,col=rainbow(150),cex.axis=0.5,main="GSE149507 (Cai et al. 2021) gene expression before normalization", las = 2)
dev.off()

##Boxplot after normalization
x11(w=10, h=7)
pdf("lung_box_afnorm.pdf")
par(mai=c(0.8,1.5,0.8,0.8))
boxplot(exprs(lung.vsnrma), horizontal = TRUE, col=rainbow(150),cex.axis=0.5,main="GSE149507 (Cai et al. 2021) gene expression after normalization" , las=2)
dev.off()
```

### Density function
```{r Lung- Density function}
## Density function before normalization
setwd(here("Plots/Plots_lung_qc"))

x11(w=10, h=7)
pdf("lung_df_bfnorm.pdf")
hist(lung, col=rainbow(150), main = "GSE149507 (Cai et al. 2021) density function before normalization")
dev.off()


## Density function after normalization
##vsnrma
lungExprs = exprs(lung.vsnrma) ####IMPORTANT

pdf("lung_df_afnorm.pdf")
plot(density(lungExprs[,1]), type="n", xlab="log Intensity", ylim=c(0,1), main = "GSE149507 (Cai et al. 2021) density function after normalization")
for(i in 1:ncol(lungExprs)){
  lines(density(lungExprs[,i]), col=rainbow(150)[i])
}
dev.off()
```

### RNA degenaration plot
```{r Lung -RNA deg}

## RNA degeneration plot
setwd(here("Plots/Plots_lung_qc"))
rnadeg.lung = AffyRNAdeg(lung)

pdf("lung_RNAdeg_ss.pdf")
plotAffyRNAdeg(rnadeg.lung, col = rainbow(150))
title(sub="GSE149507 (Cai et al. 2021) before normalization")
dev.off()

pdf("lung_RNAdeg_s.pdf")
plotAffyRNAdeg(rnadeg.lung, col = rainbow(150), transform = "shift.only")
title(sub="GSE149507 (Cai et al. 2021)")
dev.off()
```


### Scatter plots
#### Examining scatter plots and looking for individual chips with non-linear relationship that indicate  broken chips.
```{r Lung - Scatter plots}
# Scatter plots
setwd(here("Plots/Plots_lung_qc"))
x11(w=9, h=5)
par(mfrow=c(1,2))

pdf("lung_scatter.pdf")
for(i in 1:12){
  
plot(lungExprs[,c(i, i+1)], pch=".")
abline(0,1,col="red")

title(main=paste("Scatterplot of probe", substr(colnames(lung.vsnrma)[i],1,
nchar(colnames(lung.vsnrma)[i])),"and",substr(colnames(lung.vsnrma)[i+1],1,
nchar(colnames(lung.vsnrma)[i+1])), sep = " ", collapse = NULL))

permission = readline(prompt = "type y for next image: ")

if(permission == "y"){
  
}
else{
break}
}
dev.off()

##GSM4504108 or GSM4504109 might be broken
#GSM4504109 or GSM4503110 too
setwd(here("Plots/Plots_lung_qc"))

pdf("lung_scatter_brokenchip1.pdf")
file.name1=paste("scatterplot_", as.character(substr(colnames(lung.vsnrma)[i], 1,
nchar(colnames(lung.vsnrma)[i]))),"_",as.character(substr(colnames(lung.vsnrma)[i+1],1,
nchar(colnames(lung.vsnrma)[i+1]))),".eps",sep="")
dev.off()



## maybe the plot GSM4504109_SCLC_0 might be broken:
setwd(here("Plots/Plots_lung_qc"))
pdf("lung_scatter_brokenchip2.pdf")
for(i in 1:12){
  
plot(lungExprs[,c(9, i)], pch=".")
abline(0,1,col="red")

title(main=paste("Scatterplot of probe", substr(colnames(lung.vsnrma)[9],1,
nchar(colnames(lung.vsnrma)[9])),"and",substr(colnames(lung.vsnrma)[i],1,
nchar(colnames(lung.vsnrma)[i])), sep = " ", collapse = NULL))

permission = readline(prompt = "type y for next image: ")

if(permission == "y"){
  
}
else{
break}
}
dev.off()

###GSM4504109_SCLC_05_ca is broken!!! -> bana-like shaped scatter plots
### Thereby, change the whole patient 05, so chip GSM4504109_SCLC_05_ca and GSM4504110_SCLC_05_n
### Get 2 new chips for patient 5 in Gene Expression omnbius for GSE149507 lung cancer and do quality control over again
```



```{r Load in Data, eval=FALSE, include=FALSE}
breast.vsnrma <- vsnrma(breast)
breastExprs = exprs(breast.vsnrma)

lung.vsnrma <- vsnrma(lung)
lungExprs = exprs(lung.vsnrma)

rawdata.vsnrma <- vsnrma(rawdata)
rawdataExprs = exprs(rawdata.vsnrma)
```


## 2. Further Data acquistion
### Importing TRA data

```{r}
setwd(here("TRA Daten"))


# Import of tra.human.5median
tra.human.5median=read.csv(file="Human_protein_atlas_TRA_5median_genes_annotated.tsv", sep=",")

# Import of tra.human
tra.human=read.csv(file="tra.2014.human.5x.table.tsv", sep="\t")

# Import of tra.human.roth
tra.human.roth=read.csv(file="tra.2014.human.roth.5x.table.tsv", sep="\t")

# Import of tra.mouse.4301
tra.mouse.4301=read.csv(file="tra.2014.mouse.4301.5x.table.tsv", sep="\t")

# Import of tra.human.gtex
tra.human.gtex=read.csv(file="tra.2017.human.gtex.5x.table.tsv", sep="\t")

#Import of tra.mouse 
tra.mouse=read.csv(file="tra.2014.mouse.5x.table.tsv",sep="\t")

```


### Filter all Klk from the TRA data sets
###Union of all six table to visualize 
```{r Filter all Klk from the TRA data sets}

# For tra.human  
ind.human=grepl("^KLK",tra.human$gene.symbol)
ind.human =which(ind.human == TRUE)
Klk.TRA1 = as.data.frame(tra.human[ind.human,])

# For tra.human.5median
ind.5median=grepl("^KLK",tra.human.5median$Symbol)
ind.5median =which(ind.5median == TRUE)
Klk.TRA2 = as.data.frame(tra.human.5median[ind.5median,])

# For tra.human.roth
ind.roth=grepl("^KLK",tra.human.roth$gene.symbol)
ind.roth =which(ind.roth == TRUE)
Klk.TRA3 = as.data.frame(tra.human.roth[ind.roth,])

# For tra.mouse
ind.mouse=grepl("^Klk",tra.mouse$gene.symbol)
ind.mouse =which(ind.mouse == TRUE)
Klk.TRA4 = as.data.frame(tra.mouse[ind.mouse,])

# For tra.mouse.4301
ind.4301=grepl("^Klk",tra.mouse.4301$gene.symbol)
ind.4301 =which(ind.4301 == TRUE)
Klk.TRA5 = as.data.frame(tra.mouse.4301[ind.4301,])

# For tra.human.gtex 
ind.gtex=grepl("^KLK",tra.human.gtex$ensembl.symbol)
ind.gtex=which(ind.gtex == TRUE)
Klk.TRA6=as.data.frame(tra.human.gtex[ind.gtex,])



### Creating a unified TRA data set, that only contains important information for our overview
nrow.x = nrow(Klk.TRA1) + nrow(Klk.TRA2) + nrow(Klk.TRA3) +nrow(Klk.TRA4)+ nrow(Klk.TRA5) + nrow(Klk.TRA6)
Union.TRA = as.data.frame(matrix(ncol = 5, nrow=nrow.x))
union.names <- c("ensembl_genes", "Chromosome", "gene.symbol", "max.tissue", "dataset") ## assigning the unified columns
colnames(Union.TRA) <- union.names 
## now assign the rows, we poorly need to do this manually since there is no strict order of the columns of the 5 TRA data sets
## Watch out!! Chromosomes on the mice are not comparable to humans!!!
Union.TRA[1:nrow(Klk.TRA1),c(1:4)] <- Klk.TRA1[,c(2, 7, 3, 11)] # For Klk.TRA1
Union.TRA[nrow(Klk.TRA1)+1:nrow(Klk.TRA2),c(1:4)] <- Klk.TRA2[,c(1, 2, 6, 11)] # For Klk.TRA2
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+1:nrow(Klk.TRA3),c(1:4)] <- Klk.TRA3[,c(2, 7, 3, 11)] # For Klk.TRA3
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+1:nrow(Klk.TRA4),c(1:4)] <- Klk.TRA4[,c(2, 7, 3, 11)] # For Klk.TRA4
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+nrow(Klk.TRA4)+1:nrow(Klk.TRA5),c(1:4)] <- Klk.TRA5[,c(2, 7, 3, 11)] # For Klk.TRA5
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+nrow(Klk.TRA4)+nrow(Klk.TRA5)+1:nrow(Klk.TRA6),c(1:4)] <- Klk.TRA6[,c(2,6,3, 10)] # For Klk.TRA6

## now annotate the the data set
#->> VERY HELPFUL for later plots -> e.g. selecting rows that only apply for the human KLKs: 
#(Union.TRA$dataset == "tra.human")!!!!
Union.TRA[1:nrow(Klk.TRA1), 5] <- "tra.human"
Union.TRA[nrow(Klk.TRA1)+1:nrow(Klk.TRA2),5] <- "tra.human.5median"
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+1:nrow(Klk.TRA3),5] <- "tra.human.roth"
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+1:nrow(Klk.TRA4),5] <- "tra.mouse"
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+nrow(Klk.TRA4)+1:nrow(Klk.TRA5),5] <- "tra.mouse.4301"
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+nrow(Klk.TRA4)+nrow(Klk.TRA5)+1:nrow(Klk.TRA6),5] <- "tra.human.gtex"


### all human KLK genes across all TRA data sets

ind.all.tra.human.KLK <- grep("^tra.human", Union.TRA$dataset)
tra.all.human <- Union.TRA[ind.all.tra.human.KLK,]


# all unique human KLK names that are present as TRAs
all.human.KLK.gene.symbols <- unique(Union.TRA$gene.symbol[grep("tra.human", Union.TRA$dataset)])
all.human.KLK.gene.symbols # These are all the available KLK TRAs we can possibly work with
```

### ensembl annotation file
```{r Read ensembl}

setwd(here("Tables"))

ensembl_df = read.csv("ensembl.103.txt",sep="\t")
```

#Creating datasets 
Creating rawdata breast cancer data set of expression values
```{r Dataset rawdata breast}
rawdataExprs <- rawdataExprs[grepl("ENST", rownames(rawdataExprs)), ]
#dim = 95659    10

# remove .x_at suffix
rownames(rawdataExprs) <- unlist(lapply(strsplit(rownames(rawdataExprs), split = "\\."), "[", 1))
setwd(here("Scripts"))

#Setting up annotation
transcriptIDs <- as.character(ensembl_df[, "Transcript.stable.ID"])
geneSymbol <- as.character(ensembl_df[, "HGNC.symbol"])
names(geneSymbol) <- transcriptIDs

# replace ensemble IDs with HGNC symbol
chipIDs <- rownames(rawdataExprs)
noMatchIDs <- chipIDs[!chipIDs %in% transcriptIDs] # transcript IDs that are not in ensemble_103.txt
length(noMatchIDs) #112

newChipIDs <- chipIDs[chipIDs %in% transcriptIDs]
rawdataExprs <- rawdataExprs[newChipIDs, ]
dim(rawdataExprs) ## 95547    10


# 
symbol <- geneSymbol[newChipIDs]
rownames(rawdataExprs) <- as.character(symbol)  # Matrix to work with

head(rawdataExprs)

```


Creating GSE65216 breast cancer (TNBC Her2 LumA LumB) data of expression values
```{r Dataset breast cancer}
breastExprs <- breastExprs[grepl("ENST", rownames(breastExprs)), ]
dim(breastExprs) #  95659    20

# remove .x_at suffix
rownames(breastExprs) <- unlist(lapply(strsplit(rownames(breastExprs), split = "\\."), "[", 1))

setwd(here("Scripts"))

# replace ensemble IDs with HGNC symbol
chipIDs.breast <- rownames(breastExprs)
noMatchIDs.breast <- chipIDs.breast[!chipIDs.breast %in% transcriptIDs]
# transcript IDs that are not in ensemble_103.txt
length(noMatchIDs.breast) #112

newChipIDs.breast <- chipIDs.breast[chipIDs.breast %in% transcriptIDs]
breastExprs <- breastExprs[newChipIDs.breast, ]
dim(breastExprs) ## 95547    10


# 
symbol.breast <- geneSymbol[newChipIDs.breast]
rownames(breastExprs) <- as.character(symbol.breast)  # Matrix to work with

head(breastExprs)

```
Creating lung cancer data set to work with
```{r Dataset lung cancer }
lungExprs <- lungExprs[grepl("ENST", rownames(lungExprs)), ]
dim(lungExprs) #  95659    20
 
# remove .x_at suffix
rownames(lungExprs) <- unlist(lapply(strsplit(rownames(lungExprs), split = "\\."), "[", 1))

setwd(here("Scripts"))

# replace ensemble IDs with HGNC symbol
chipIDs.lung <- rownames(lungExprs)
noMatchIDs.lung <- chipIDs.lung[!chipIDs.lung %in% transcriptIDs]
# transcript IDs that are not in ensemble_103.txt
length(noMatchIDs.lung) #112

newChipIDs.lung <- chipIDs.lung[chipIDs.lung %in% transcriptIDs]
lungExprs <- lungExprs[newChipIDs.lung, ]
dim(lungExprs) ## 95547    10


# 
symbol.lung <- geneSymbol[newChipIDs.lung]
rownames(lungExprs) <- as.character(symbol.lung)  # Matrix to work with

head(lungExprs)

```
# 3. Expression Analysis

Overview of gene expression:
The gene expression values are depicted in a logarithmic scale with base 2. This means, that a increase in gene expression by the amount of one unit corresponds with an increase by the factor 2. 
```{r}
gene.summary <- function(x){
 round(apply(x, 2, summary), digits = 2)
}
gene.summary(rawdataExprs)
gene.summary(breastExprs)
gene.summary(lungExprs)
```
We can see here, that the lowest value is about 5 while the highest gene expression value is around 15. Due to the logarithmic scale, an expression value of 15 is 1024 times higher than the value 5. It is also notable, that the median does not fluctuate too much for all samples.This is due to the fact, that we performed vsn normalization.


Ioslating all KLK genes from each dataset 
```{r Isolating KLK genes, results=FALSE}
# All KLK genes
ind.all.KLK=grep("^KLK",symbol)
Kallikreins.all=symbol[ind.all.KLK]

ind.all.human =which(as.character(symbol) %in% as.character(Kallikreins.all)) 
Kallikreins<- as.character(Kallikreins.all)
Kallikreins

## For first breast cancer data set
all.KLK.rawdata = rawdataExprs[ind.all.human,]

## For breast GSE
all.KLK.breast = breastExprs[ind.all.human,]

## For lung GSE 
all.KLK.lung = lungExprs[ind.all.human,]

#TRA: 
## For this we use the human TRA data set (Uhlén et al. 2015) <- right one?
TRA.symbol=tra.human[,3] # lists all TRAs names for humans

KLK.genes.TRA.human = TRA.symbol[ind.human] 

ind.KLK.tra.human =which(as.character(symbol) %in% KLK.genes.TRA.human) #ind.KLK.tra.human gives respective column 

## For first breast cancer data set
TRA.KLK.rawdata = rawdataExprs[ind.KLK.tra.human,]

## For breast GSE
TRA.KLK.breast = breastExprs[ind.KLK.tra.human,]

## For lung GSE 
TRA.KLK.lung = lungExprs[ind.KLK.tra.human,]

```

## 3.1 breast cancer GSE27830 (Nagel et al. 2012)

###Visualization via Heatmaps
```{r Heatmaps}
#transform dataset - genes on columns
df.TRA.KLK.rawdata <- data.frame(t(TRA.KLK.rawdata))

#Define function for calculating correlation 
cor.genes <- function(df){
  df.cor <- cor(df, method = "pearson")
  diag(df.cor)=NA
  df.cor[upper.tri(df.cor)]=NA
  return(data.frame(df.cor))
}

#test function 
cor.TRA.KLK.rawdata <- cor.genes(df.TRA.KLK.rawdata)
length(which(cor.TRA.KLK.rawdata == 1))
#14 isoforms are not unique
#Identical transcripts can cause issues later on for statistical testing ect. 

#remove identical columns
TRA.KLK.rawdata.clean <- df.TRA.KLK.rawdata[!duplicated(unclass(df.TRA.KLK.rawdata))]
dim(TRA.KLK.rawdata)
dim(TRA.KLK.rawdata.clean)
#removed 12 identical isoforms

#check for identicals 
cor.TRA.KLK.rawdata.clean <- cor.genes(TRA.KLK.rawdata.clean)
length(which(cor.TRA.KLK.rawdata.clean == 1))
#no more identical columns!

#Heatmap of KLK TRAs expression withOUT identicals
setwd(here("Plots/Plots_raw_breast"))
pdf("KLK_TRAs_Heatmap_rawdata_noid.pdf")
pheatmap(t(TRA.KLK.rawdata.clean),
         main = "Expression values of KLK TRAs GSE27830 (Nagel et al. 2012)",
         cellwidth = 30,
         cellheight = 10,
         treeheight_col = 20,
         fontsize_row = 8)
dev.off()
```


```{r Heatmaps annotation}
# Creating a heatmap, that displays all KLKs and indicates the ones who are TRAs

# First we need to clean up the isoforms again
df.all.KLK.rawdata <- data.frame(t(all.KLK.rawdata))

#test function 
cor.all.KLK.rawdata <- cor.genes(df.all.KLK.rawdata)
length(which(cor.all.KLK.rawdata == 1))
#40 isoforms are not unique
#Identical transcripts can cause issues later on for statistical testing etc. 

#remove identical isoforms
all.KLK.rawdata.clean <- df.all.KLK.rawdata[!duplicated(unclass(df.all.KLK.rawdata))]
dim(all.KLK.rawdata.clean) #removed 39 identical isoforms


# adding annotations to the KLK genes, for whether they are TRAs or not
# 1. show simple cluster annotated to heatmap

hclust.genes.rawdata <- hclust(dist(t(all.KLK.rawdata.clean)), method = "complete")

as.dendrogram(hclust.genes.rawdata) %>%
  plot(horiz = TRUE)
# Lets try it with 2 clusters in the heatmap, just for visualization purposes
annotation.hmap.rawdata <- cutree(tree = as.dendrogram(hclust.genes.rawdata), k = 2)
annotation.hmap.rawdata  <- data.frame(cluster = ifelse(test = annotation.hmap.rawdata == 1, yes = "cluster 1", no = "cluster 2"))

# 2. differentiate between TRA-/ or Non-TRA-KLKs
TRUEorFALSe <- colnames(all.KLK.rawdata.clean) %in% colnames(TRA.KLK.rawdata.clean)
TRAorNON <- ifelse(TRUEorFALSe == TRUE, "TRA", "Non-TRA")
annotation.hmap.rawdata$TRA_or_NON  <- TRAorNON


# call the annotated Heatmap
setwd(here("Plots/Plots_raw_breast"))
pdf("all_KLK_Heatmap_rawdata.pdf")
pheatmap(t(all.KLK.rawdata.clean),
         main = "Expression values of all KLKs GSE27830 (Nagel et al. 2012)",
         annotation_row = annotation.hmap.rawdata,
         cellwidth = 27,
         cellheight = 5.5,
         treeheight_col = 10,
         fontsize_row = 8,
         cutree_rows = 2
)
dev.off()

```

Further reason for removing the identicals
Now, there are no KLK TRA genes that have identical expression values. This could have caused problems for statistical testing or for performing correlation analysis later on. Before starting off with the analysis, we should get an overview of KLK TRAs for the three data sets via Heatmaps. 

## Visualization of gene expression
Boxplot, Histograms
```{r Boxplot}
#order KLKs alphabetically
order.hand.TRA <- c(1, 14,  2,  5, 7, 8, 9, 10, 15, 16, 17, 18, 19, 21, 22, 23, 25, 3, 4, 6, 11, 12, 13, 20, 24, 26, 27)

TRA.KLK.rawdata.clean = data.frame(TRA.KLK.rawdata.clean)
#data.frame(TRA.KLK.rawdata.clean[,order.hand.TRA])

#Boxplot:
setwd(here("Plots/Plots_raw_breast"))
pdf("KLK_TRAs_boxplot_all_rawdata.pdf")
boxplot(t(TRA.KLK.rawdata.clean),col=rainbow(length(rownames(TRA.KLK.rawdata.clean))),main="Gene expression of KLK TRAs GSE27830 (Nagel et al. 2012)",cex.axis=0.8, sub = "Identical genes removed")
dev.off()
```


##Clustering

##PCA

##Hyposis testing
1. Test whether KLK1 and KLK1.3 have a higher expression than other KLKs
```{r}
#Hypothesis testing with t-tests
#first we need mean values 
TRA.KLK.rawdata.mean = colMeans(TRA.KLK.rawdata.clean)
TRA.KLK.rawdata.mean


#just for fun: check distribution
hist(TRA.KLK.rawdata.mean, breaks = seq(6.1, 7.3, by= 0.05)) 
#do we have a random variable?





```



# 3.2 Breast cancer GSE65216

##Isolating TRAs from the breast cancer dataset
```{r, Isolating KLK TRAs breast + Heatmaps}

#transform dataset - genes on columns
df.TRA.KLK.breast <- data.frame(t(TRA.KLK.breast))

#number of identical columns
cor.TRA.KLK.breast <- cor.genes(df.TRA.KLK.breast)
length(which(cor.TRA.KLK.breast == 1))
# 5 identical isoforms 

#cleanup function 
genes.cleanup <- function(df){
  df[!duplicated(unclass(df))]
}

#remove identical columns
TRA.KLK.breast.clean <- genes.cleanup(df.TRA.KLK.breast)
dim(TRA.KLK.breast.clean)

#check clean dataframe for correlations
length(which(cor.genes(TRA.KLK.breast.clean)==1)) #0 correlations
TRA.KLK.breast.clean

#heatmap withOUT duplicates
setwd(here("Plots/Plots_breast"))
pdf("KLK_TRAs_Heatmap_breast_noid.pdf")
pheatmap(t(TRA.KLK.breast.clean),
         main = "Expression values of KLK TRAs GSE65216 (Maire et al. 2013)",
         cellwidth = 15,
         cellheight = 8,
         treeheight_col = 20,
         fontsize_row = 8,
         fontsize_col = 7)
dev.off()
```

```{r Heatmaps annotation}
# Creating a heatmap, that displays all KLKs and indicates the ones who are TRAs

# First we need to clean up the isoforms again
df.all.KLK.breast <- data.frame(t(all.KLK.breast))

#test function 
cor.all.KLK.breast <- cor.genes(df.all.KLK.breast)
length(which(cor.all.KLK.breast == 1))
#49 isoforms are not unique
#Identical transcripts can cause issues later on for statistical testing etc. 

#remove identical isoforms
all.KLK.breast.clean <- df.all.KLK.breast[!duplicated(unclass(df.all.KLK.breast))]
dim(all.KLK.breast.clean) #removed 39 identical isoforms


# adding annotations to the KLK genes, for whether they are TRAs or not
# 1. show simple cluster annotated to heatmap

hclust.genes.breast <- hclust(dist(t(all.KLK.breast.clean)), method = "complete")

as.dendrogram(hclust.genes.breast) %>%
  plot(horiz = TRUE)
# Lets try it with 2 clusters in the heatmap, just for visualization purposes
annotation.hmap.breast <- cutree(tree = as.dendrogram(hclust.genes.breast), k = 2)
annotation.hmap.breast  <- data.frame(cluster = ifelse(test = annotation.hmap.breast == 1, yes = "cluster 1", no = "cluster 2"))

# 2. differentiate between TRA-/ or Non-TRA-KLKs
annotation.hmap.breast$TRA_or_NON  <- TRAorNON

# 3. To increase visibility, we will cut off the GSM number for the patient names
all.KLK.breast.clean.noGSM <- all.KLK.breast.clean
#rownames(all.KLK.breast.clean.noGSM) <- sub("...........", "", rownames(all.KLK.breast.clean)) 
#There seems to be the problem, that the sample names cant be shortened or else Tum016_Her2 would appear twice, which is not possible
#However these are two different samples, which where taken out of the same tumor, hence the different GSM number
#The different of these samples is their replication protein repA for GSM1588991 and repB for GSM1588992
#Here we will just shorten the name by 2 characters less, so we can differentiate between those two 
rownames(all.KLK.breast.clean.noGSM)[1:8] <- sub("...........", "", rownames(all.KLK.breast.clean)[1:8])
rownames(all.KLK.breast.clean.noGSM)[9:10] <- sub(".........", "", rownames(all.KLK.breast.clean)[9:10])
rownames(all.KLK.breast.clean.noGSM)[11:20] <- sub("...........", "", rownames(all.KLK.breast.clean)[11:20])

 
# call the annotated Heatmap
setwd(here("Plots/Plots_breast"))
pdf("all_KLK_Heatmap_breast.pdf")
pheatmap(t(all.KLK.breast.clean.noGSM),
         main = "Expression values of all KLKs GSE27830 (Nagel et al. 2012)",
         annotation_row = annotation.hmap.breast,
         cellwidth = 14,
         cellheight = 5.3,
         treeheight_col = 10,
         fontsize_row = 8,
         cutree_rows = 2,
         fontsize_row = 2
)
dev.off()

```

## Visualization of gene expression
Boxplots 
```{r}
#boxplots 
setwd(here("Plots/Plots_breast"))
pdf("KLK_TRAs_boxplot_all_breast.pdf")
boxplot(t(TRA.KLK.breast.clean),col=rainbow(length(rownames(TRA.KLK.breast.clean))),main="Expression values of KLK TRAs GSE65216 (Maire et al. 2013)",cex.axis=0.8)
dev.off()

```


## 3.3 Lung cancer GSE149507

###Isolating TRAs from lung cancer dataset
```{r, Isolating KLK TRAs lung + Heatmaps}

#transform dataset - genes on columns
df.TRA.KLK.lung <- data.frame(t(TRA.KLK.lung))

#finding duplicates
cor.TRA.KLK.lung <- cor.genes(df.TRA.KLK.lung)
length(which(cor.TRA.KLK.lung == 1)) #14 identical columns <- why does it always change (5, 14 ,18) <- maybe mistake in ind.human?

#removing identical columns
TRA.KLK.lung <- data.frame(TRA.KLK.lung)
TRA.KLK.lung.clean <- genes.cleanup(df.TRA.KLK.lung)
length(which(cor.genes(TRA.KLK.lung.clean) == 1))

#heatmap with duplicates
setwd(here("Plots/Plots_lung"))
pdf("KLK_TRAs_Heatmap_lung_wid.pdf")
pheatmap(TRA.KLK.lung,
         main = "Expression values of KLK TRAs GSE149507 (Cai et al. 2021)",
         cellwidth = 20,
         cellheight = 8,
         treeheight_col = 20,
         fontsize_row = 7,
         fontsize_col = 7)
dev.off()

#order dataframe rows, columns alphabetically
TRA.KLK.lung.clean = t(TRA.KLK.lung.clean[,order.hand.TRA])

#heatmap withOUT duplicates
setwd(here("Plots/Plots_lung"))
pdf("KLK_TRAs_Heatmap_lung_noid.pdf")
pheatmap(TRA.KLK.lung.clean,
         main = "Expression values of KLK TRAs GSE149507 (Cai et al. 2021)",
         cellwidth = 20,
         cellheight = 8,
         treeheight_col = 20,
         fontsize_row = 8,
         fontsize_col = 7)
dev.off()
```

```{r Heatmaps annotation}
# Creating a heatmap, that displays all KLKs and indicates the ones who are TRAs

# First we need to clean up the isoforms again
df.all.KLK.lung <- data.frame(t(all.KLK.lung))

#test function 
cor.all.KLK.lung <- cor.genes(df.all.KLK.lung)
length(which(cor.all.KLK.lung == 1))
#64 entries
#Identical transcripts can cause issues later on for statistical testing etc. 

#remove identical isoforms
all.KLK.lung.clean <- df.all.KLK.lung[!duplicated(unclass(df.all.KLK.lung))]
dim(all.KLK.lung.clean) #removed 39 identical isoforms


# adding annotations to the KLK genes, for whether they are TRAs or not
# 1. show simple cluster annotated to heatmap

hclust.genes.lung <- hclust(dist(t(all.KLK.lung.clean)), method = "complete")

as.dendrogram(hclust.genes.lung) %>%
  plot(horiz = TRUE)
# Lets try it with 2 clusters in the heatmap, just for visualization purposes
annotation.hmap.lung <- cutree(tree = as.dendrogram(hclust.genes.lung), k = 2)
annotation.hmap.lung  <- data.frame(cluster = ifelse(test = annotation.hmap.lung == 1, yes = "cluster 1", no = "cluster 2"))

# 2. differentiate between TRA-/ or Non-TRA-KLKs
annotation.hmap.lung$TRA_or_NON  <- TRAorNON

# call the annotated Heatmap
setwd(here("Plots/Plots_lung"))
pdf("all_KLK_Heatmap_lung.pdf")
pheatmap(t(all.KLK.lung.clean),
         main = "Expression values of all KLKs GSE27830 (Nagel et al. 2012)",
         annotation_row = annotation.hmap.lung,
         cellwidth = 26,
         cellheight = 4.5,
         treeheight_col = 10,
         fontsize_row = 8,
         cutree_rows = 2,
)
dev.off()

```

## Visualization of gene expression
Boxplots 
```{r}
#order dataframe rows, columns alphabetically
TRA.KLK.lung.clean = t(TRA.KLK.lung.clean[,order.hand.TRA])

#boxplots 
setwd(here("Plots/Plots_lung"))
pdf("KLK_TRAs_boxplot_all_lung.pdf")
boxplot(t(TRA.KLK.lung.clean),col=rainbow(length(rownames(TRA.KLK.lung.clean))),main="Expression values of KLK TRAs GSE149507 (Cai et al. 2021)",cex.axis=0.8)
dev.off()

```

## K means
``` {r}
#determining the optimal number of k clusters, preparing wssplot
wssplot <- function(breastExprs, nc=15, seed=1234)
{
  wss <- (nrow(breastExprs)-1)*sum(apply(breastExprs,2,var))
  for(i in 2:nc){
    set.seed(seed)
    wss[i] <- sum(kmeans(rawdataExprs, centers = i)$withinss)}
  plot(1:nc, wss, type="b", xlab="Number of Clusters", ylab="Within groups sum of squares")
}

#wssplot
wssplot(breastExprs)

#apparently optimal Number of clusters is 2
# ist jetzt nicht squared aber muss ja wenn dann erst bei kmeans plot

#visualization, !!! caution package "factoextra" required
fviz_cluster(results.km, data=rawdataExprs, 
             palette =c("#2E9FDF","#00AFBB"), 
             geom = "point", 
             ellipse.type = "convex",
             ggtheme = theme_bw()
)

