---
title: "Klk_Genes_Data_Analysis"
author: "Anouk Dupe, Maria Yemane, Dustin Schilling, David Eckey"
date: "4 5 2021"
output:
  pdf_document: default
  html_document: default
editor_options:
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

###load library
```{r Load Libaries, hide=TRUE}
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(hexbin)
library(ggplot2)
library(pheatmap)
library(dendextend) #install.packages("dendextend")
library(factoextra) #install.packages("factoextra")
library(gtools) #install.packages("gtools")
library(plyr)
library(tidyverse) #install.packages("tidyverse")

library(RColorBrewer) #install.packages("RColorBrewer")
#library(dendextend) #install.packages("dendextend")
library(ggdendro) #install.packages("ggdendro")

###here package 
library(here)
#set the top level folder with here
here::here()
```
# 1. Introduction
Relevance of TRAs in cancer therapy: Tumor restricted antigens only occur in cancer tissue, which makes them a great immunological target. Identifying TRAs overexpressed in specific cancer tissue is imporant for the development of immunotherapeutic drugs....
Biological question considering KLKs:..

```{r Annotation Data Sets, eval=FALSE, include=FALSE}

#Cancer data Sets 
###"rawdata breast GSE27830": Nagel JH, Peeters JK, Smid M, Sieuwerts AM et al. Gene expression profiling assigns CHEK2 1100delC breast cancers to the luminal intrinsic subtypes. Breast Cancer Res Treat 2012 Apr 
#(Nagel et al. 2012)

###"GSE65216 breast cancer": Maire V, Némati F, Richardson M, Vincent-Salomon A et al. Polo-like kinase 1: a potential therapeutic option in combination with conventional chemotherapy for the management of patients with triple-negative breast cancer. Cancer Res 2013 Jan 15 
#(Maire et al. 2013)

###"GSE149507 lung cancer": Cai L, Liu H, Huang F, Fujimoto J et al. Cell-autonomous immune gene expression is repressed in pulmonary neuroendocrine cells and small cell lung cancer. Commun Biol 2021 Mar 9
#(Cai et al. 2021)



#To cite the origin of the TRA data
### (Su et al. 2002, 2004) ->  mouse Novartis data
####Su et al. 2002, Large-scale analysis of the human and mouse transcriptomes. Proc Natl Acad Sci U S A. 2002 Apr 2
#### AND: Su et al. 2004, A gene atlas of the mouse and human protein-encoding transcriptomes. Proc Natl Acad Sci U S A. 2004 Apr 20

###(Roth et al. 2008) -> human Roth data
####Roth et al. 2006, Gene expression analyses reveal molecular relationships among 20 regions of the human CNS. Neurogenetics. 2006 May

### (Lattin et al. 2006) -> mouse Lattin data
#### Lattin et al. 2008, Expression analysis of G Protein-Coupled Receptors in mouse macrophages. Immunome Res. 2008 Apr 29

### (human GTEX data 2015) -> RNAseq data
#### GTEX 2013, The Genotype-Tissue Expression (GTEx) project. GTEx Consortium. Nat Genet. 2013 Jun. 

###
#### GTEX 2015, Human genomics. The Genotype-Tissue Expression (GTEx) pilot analysis: multitissue gene regulation in humans. GTEx Consortium. Science. 2015 May 8

### Uhlén et al. 2015, Proteomics. Tissue-based map of the human proteome. Science. 2015 Jan 23

```



###Read .CEL Files
```{r Read CEL Files, hide = TRUE}
#creating a list of the CEL file names, for all three data sets
setwd(here("Rawdata/rawdata breast GSE27830"))
cels.rawdata <- list.files(pattern = 'CEL')

setwd(here("Rawdata/GSE65216 breast cancer TNBC Her2 LumA LumB"))
cels.breast <- list.files(pattern = 'CEL')

setwd(here("Rawdata/GSE149507 lung cancer"))
cels.lung <- list.files(pattern = 'CEL')


```

```{r eval=FALSE, include=FALSE}
setwd(here("Rawdata/rawdata breast GSE27830"))
cels.rawdata <- c("GSM687017.CEL", "GSM687018.CEL", "GSM687019.CEL", "GSM687020.CEL", "GSM687021.CEL", "GSM687022.CEL", "GSM687023.CEL", "GSM687024.CEL", "GSM687025.CEL", "GSM687026.CEL")

setwd(here("Rawdata/GSE65216 breast cancer TNBC Her2 LumA LumB"))
cels.breast <- c("GSM1588970_Tum01_TNBC.CEL", "GSM1588972_Tum02_Her2.CEL", "GSM1588974_Tum03_TNBC.CEL",  "GSM1588975_Tum04_TNBC.CEL", "GSM1588976_Tum05_TNBC.CEL", "GSM1588977_Tum06_TNBC.CEL", "GSM1588989_Tum020_Her2.CEL", "GSM1588990_Tum015_Her2.CEL", "GSM1588991_Tum016_Her2.CEL", "GSM1588992_Tum016_Her2.CEL", "GSM1589064_Tum071_LumA.CEL", "GSM1589065_Tum073_LumA.CEL", "GSM1589066_Tum074_LumA.CEL", "GSM1589067_Tum075_LumA.CEL", "GSM1589068_Tum076_LumA.CEL", "GSM1589093_Tum101_LumB.CEL", "GSM1589094_Tum102_LumB.CEL", "GSM1589095_Tum103_LumB.CEL", "GSM1589097_Tum104_LumB.CEL", "GSM1589098_Tum105_LumB.CEL")

setwd(here("Rawdata/GSE149507 lung cancer"))
cels.lung <- c("GSM4504101_SCLC_01_ca.CEL", "GSM4504102_SCLC_01_n.CEL", "GSM4504103_SCLC_02_ca.CEL", "GSM4504104_SCLC_02_n.CEL", "GSM4504105_SCLC_03_ca.CEL", "GSM4504106_SCLC_03_n.CEL", "GSM4504107_SCLC_04_ca.CEL", "GSM4504108_SCLC_04_n.CEL", "GSM4504111_SCLC_06_ca.CEL", "GSM4504112_SCLC_06_n.CEL", "GSM4504113_SCLC_07_ca.CEL", "GSM4504114_SCLC_07_n.CEL" )

```



```{r Affybatch, hide= TRUE}
#read the CEL files into an Affybatch

#Rawdata
rawdata <- ReadAffy(filenames = paste(here("Rawdata/rawdata breast GSE27830"), cels.rawdata, sep = "/"), verbose = TRUE)
rawdata@cdfName <- "HGU133Plus2_Hs_ENST"


orig.rawdata=colnames(exprs(rawdata))
new.rawdata <- substr(orig.rawdata, 1, nchar(cels.rawdata)-4)
colnames(exprs(rawdata)) <- new.rawdata
rownames(rawdata@phenoData) <- new.rawdata
rownames(rawdata@protocolData) <- new.rawdata

save(rawdata, file = paste0(paste(here("Tables")), "/rawdata.rda"))



#Breast 
breast <- ReadAffy(filenames = paste(here("Rawdata/GSE65216 breast cancer TNBC Her2 LumA LumB"), cels.breast, sep = "/"), verbose = TRUE)
breast@cdfName <- "HGU133Plus2_Hs_ENST"

orig.breast=colnames(exprs(breast))
new.breast <- substr(orig.breast, 1, nchar(cels.breast)-4)
colnames(exprs(breast)) <- new.breast
rownames(breast@phenoData) <- new.breast
rownames(breast@protocolData) <- new.breast

save(breast, file = paste0(paste(here("Tables")), "/breast.rda"))


#Lung
lung <- ReadAffy(filenames = paste(here("Rawdata/GSE149507 lung cancer"), cels.lung, sep = "/"), verbose = TRUE)
lung@cdfName <- "HGU133Plus2_Hs_ENST"

orig.lung=colnames(exprs(lung))
new.lung <- substr(orig.lung, 1, nchar(cels.lung)-4)
colnames(exprs(lung)) <- new.lung
rownames(lung@phenoData) <- new.lung
rownames(lung@protocolData) <- new.lung

save(lung, file = paste0(paste(here("Tables")), "/lung.rda"))
```
```{r Rawdata breast}
# save normalized version of rawdata before continuing

rawdata.vsnrma <- vsnrma(rawdata)
setwd(here("Tables"))
save.image(file = "rawdataNorm.rda")

```

# 2. Quality Control

## 2.1 Quality control - GSE65216 breast cancer
### Single chip control
```{r Breast - Single Chip control}
# read images for GSE65216 breast cancer TNBC Her2 LumA LumB
setwd(here("Plots/Plots_breast_qc"))
pdf("breast_chip_control.pdf")
image(breast, col=rainbow(100,start=0,end = 0.75)[100:1])
dev.off()

# the individual chips look fine
```

### Normalization and meanSd plot
```{r Breast - Normalization}
## Normalization
breast.vsnrma <- vsnrma(breast)

setwd(here("Tables"))

save.image(file = "breastNorm.rda")

## MeanSdPlot
setwd(here("Plots/Plots_breast_qc"))
msd.breast <- meanSdPlot(breast.vsnrma)
msd.breast.title <- msd.breast$gg + ggtitle("Mean sd GSE65216 (Maire et al. 2013)")

pdf("breast_meansdPlot.pdf")
print(msd.breast.title)
dev.off()
###The line is approximately horizontal, so there shouldn't be a variance-mean dependence 


```

### Visualization via boxplots
```{r Breast - Boxplots}

##Boxplot before normalization
setwd(here("Plots/Plots_breast_qc"))

pdf("breast_box_bfnorm.pdf")
par(mai=c(0.8,1.5,0.8,0.8))
boxplot(breast, horizontal = TRUE , col=rainbow(150),cex.axis=0.38,main="GSE65216 (Maire et al. 2013) gene expression before normalization", las = 2)
dev.off()


##Boxplot after normalization
pdf("breast_box_afnorm.pdf")
par(mai=c(0.8,1.5,0.8,0.8))
boxplot(exprs(breast.vsnrma),horizontal = TRUE,col=rainbow(150),cex.axis=0.5,main="GSE65216 (Maire et al. 2013) gene expression after normalization" , las=2)
dev.off()

```

### Density function
```{r Breast - Density function}
## Density function before normalization
setwd(here("Plots/Plots_breast_qc"))

pdf("breast_dlogf_bfnorm.pdf")
hist(breast, col=rainbow(150), main = "GSE65216 (Maire et al. 2013) density function before normalization")
dev.off()



## Density function after normalization
##vsnrma
breastExprs = exprs(breast.vsnrma) ####IMPORTANT

pdf("breast_dflog_afnorm.pdf")
plot(density(breastExprs[,1]), type="n", xlab="log Intensity", main = "GSE65216 (Maire et al. 2013) density function after normalization")
for(i in 1:ncol(breastExprs)){
  lines(density(breastExprs[,i]), col=rainbow(150)[i])
}
dev.off()


```

### RNA degeneration plot
```{r Breast - RNA deg Plot}

## RNA degeneration plot
setwd(here("Plots/Plots_breast_qc"))
rnadeg.breast = AffyRNAdeg(breast)

pdf("breast_RNAdeg_ss.pdf")
plotAffyRNAdeg(rnadeg.breast, col = rainbow(150))
title(sub="GSE65216 (Maire et al. 2013)")
dev.off()

pdf("breast_RNAdeg_s.pdf")
plotAffyRNAdeg(rnadeg.breast, col = rainbow(150), transform = "shift.only")
title(sub="GSE65216 (Maire et al. 2013)")
dev.off()
```


### Scatter plots
Examining scatter plots and looking for individual chips with non-linear relationship that indicate  broken chips.
```{r Breast - Scatter plots}
# Scatter plots

setwd(here("Plots/Plots_breast_qc"))

pdf("breast_scatter.pdf")  
for(i in 1:21){
plot(breastExprs[,c(i, i+1)], pch=".")
abline(0,1,col="red")

title(main=paste("Scatterplot of probe", substr(colnames(breast.vsnrma)[i],1,
nchar(colnames(breast.vsnrma)[i])),"and",substr(colnames(breast.vsnrma)[i+1],1,
nchar(colnames(breast.vsnrma)[i+1])), sep = " ", collapse = NULL))

permission = readline(prompt = "type y for next image: ")

if(permission == "y"){
  
}
else{
break}
}
dev.off()

file.name=paste("scatterplot_", as.character(substr(colnames(breast.vsnrma)[i], 1,
nchar(colnames(breast.vsnrma)[i]))),"_",as.character(substr(colnames(breast.vsnrma)[i+1],1,
nchar(colnames(breast.vsnrma)[i+1]))),".eps",sep="")





## None of the scatterplots seem to deviate significantly from the red line

```



## 2.2 Quality control - GSE149507 lung cancer
### Single chip control
```{r Lung - chip control}
# read images for GSE149507 lung cancer
setwd(here("Plots/Plots_lung_qc"))
pdf("lung_chip_control.pdf") 
image(lung, col=rainbow(100,start=0,end = 0.75)[100:1])
dev.off()
# the individual chips look fine
```

### Normalization and meanSd plot
```{r Lung - Normalization and meansd Plot}
## Normalization
lung.vsnrma <- vsnrma(lung)

setwd(here("Tables"))

save.image(file = "lungNorm.rda")

## MeanSdPlot
setwd(here("Plots/Plots_lung_qc"))

msd.lung <- meanSdPlot(lung.vsnrma)
msd.lung.title <- msd.lung$gg + ggtitle("Mean sd GSE149507 (Cai et al. 2021)")

pdf("breast_meansdPlot.pdf")
print(msd.lung.title)
dev.off()

###The line is approximately horizontal, so there shouldn't be a variance-mean dependence 
```

### Boxplots
```{r Lung - Boxplots}
setwd(here("Plots/Plots_lung_qc"))

##Boxplot before normalization
pdf("lung_box_bfnorm.pdf")
par(mai=c(0.8,1.5,0.8,0.8))
boxplot(lung, horizontal = TRUE ,col=rainbow(150),cex.axis=0.5,main="GSE149507 (Cai et al. 2021) gene expression before normalization", las = 2)
dev.off()

##Boxplot after normalization
pdf("lung_box_afnorm.pdf")
par(mai=c(0.8,1.5,0.8,0.8))
boxplot(exprs(lung.vsnrma), horizontal = TRUE, col=rainbow(150),cex.axis=0.5,main="GSE149507 (Cai et al. 2021) gene expression after normalization" , las=2)
dev.off()
```

### Density function
```{r Lung- Density function}
## Density function before normalization
setwd(here("Plots/Plots_lung_qc"))

pdf("lung_df_bfnorm.pdf")
hist(lung, col=rainbow(150), main = "GSE149507 (Cai et al. 2021) density function before normalization")
dev.off()


## Density function after normalization
##vsnrma
lungExprs = exprs(lung.vsnrma) ####IMPORTANT

pdf("lung_df_afnorm.pdf")
plot(density(lungExprs[,1]), type="n", xlab="log Intensity",main = "GSE149507 (Cai et al. 2021) density function after normalization")
for(i in 1:ncol(lungExprs)){
  lines(density(lungExprs[,i]), col=rainbow(150)[i])
}
dev.off()
```

### RNA degenaration plot
```{r Lung -RNA deg}

## RNA degeneration plot
setwd(here("Plots/Plots_lung_qc"))
rnadeg.lung = AffyRNAdeg(lung)

pdf("lung_RNAdeg_ss.pdf")
plotAffyRNAdeg(rnadeg.lung, col = rainbow(150))
title(sub="GSE149507 (Cai et al. 2021) before normalization")
dev.off()

pdf("lung_RNAdeg_s.pdf")
plotAffyRNAdeg(rnadeg.lung, col = rainbow(150), transform = "shift.only")
title(sub="GSE149507 (Cai et al. 2021)")
dev.off()
```


### Scatter plots
Examining scatter plots and looking for individual chips with non-linear relationship that indicate  broken chips.
```{r Lung - Scatter plots}
# Scatter plots
setwd(here("Plots/Plots_lung_qc"))

pdf("lung_scatter.pdf")
for(i in 1:12){
  
plot(lungExprs[,c(i, i+1)], pch=".")
abline(0,1,col="red")

title(main=paste("Scatterplot of probe", substr(colnames(lung.vsnrma)[i],1,
nchar(colnames(lung.vsnrma)[i])),"and",substr(colnames(lung.vsnrma)[i+1],1,
nchar(colnames(lung.vsnrma)[i+1])), sep = " ", collapse = NULL))

permission = readline(prompt = "type y for next image: ")

if(permission == "y"){
  
}
else{
break}
}
dev.off()

##GSM4504108 or GSM4504109 might be broken
#GSM4504109 or GSM4503110 too
setwd(here("Plots/Plots_lung_qc"))

pdf("lung_scatter_brokenchip1.pdf")
file.name1=paste("scatterplot_", as.character(substr(colnames(lung.vsnrma)[i], 1,
nchar(colnames(lung.vsnrma)[i]))),"_",as.character(substr(colnames(lung.vsnrma)[i+1],1,
nchar(colnames(lung.vsnrma)[i+1]))),".eps",sep="")
dev.off()



## maybe the plot GSM4504109_SCLC_0 might be broken:
setwd(here("Plots/Plots_lung_qc"))
pdf("lung_scatter_brokenchip2.pdf")
for(i in 1:12){
  
plot(lungExprs[,c(9, i)], pch=".")
abline(0,1,col="red")

title(main=paste("Scatterplot of probe", substr(colnames(lung.vsnrma)[9],1,
nchar(colnames(lung.vsnrma)[9])),"and",substr(colnames(lung.vsnrma)[i],1,
nchar(colnames(lung.vsnrma)[i])), sep = " ", collapse = NULL))

permission = readline(prompt = "type y for next image: ")

if(permission == "y"){
  
}
else{
break}
}
dev.off()

###GSM4504109_SCLC_05_ca is broken!!! -> banana-like shaped scatter plots
### Thereby, change the whole patient 05, so chip GSM4504109_SCLC_05_ca and GSM4504110_SCLC_05_n
### Get 2 new chips for patient 5 in Gene Expression omnbius for GSE149507 lung cancer and do quality control over again
```



```{r Load in Data, echo=TRUE}
breast.vsnrma <- vsnrma(breast)
breastExprs = exprs(breast.vsnrma)

lung.vsnrma <- vsnrma(lung)
lungExprs = exprs(lung.vsnrma)

rawdata.vsnrma <- vsnrma(rawdata)
rawdataExprs = exprs(rawdata.vsnrma)
```


# 3. Further data acquistion
## Importing TRA data

```{r}
setwd(here("TRA Daten"))


# Import of tra.human.5median
tra.human.5median=read.csv(file="Human_protein_atlas_TRA_5median_genes_annotated.tsv", sep=",")

# Import of tra.human
tra.human=read.csv(file="tra.2014.human.5x.table.tsv", sep="\t")

# Import of tra.human.roth
tra.human.roth=read.csv(file="tra.2014.human.roth.5x.table.tsv", sep="\t")

# Import of tra.mouse.4301
tra.mouse.4301=read.csv(file="tra.2014.mouse.4301.5x.table.tsv", sep="\t")

# Import of tra.human.gtex
tra.human.gtex=read.csv(file="tra.2017.human.gtex.5x.table.tsv", sep="\t")

#Import of tra.mouse 
tra.mouse=read.csv(file="tra.2014.mouse.5x.table.tsv",sep="\t")

```

## Union of TRA data
Union of all six TRA data tables to visualize TRA distribuiton 
```{r Union of TRA data}
### Filter all Klk from the TRA data sets

# For tra.human  
ind.human=grepl("^KLK",tra.human$gene.symbol)
ind.human =which(ind.human == TRUE)
Klk.TRA1 = as.data.frame(tra.human[ind.human,])

# For tra.human.5median
ind.5median=grepl("^KLK",tra.human.5median$Symbol)
ind.5median =which(ind.5median == TRUE)
Klk.TRA2 = as.data.frame(tra.human.5median[ind.5median,])

# For tra.human.roth
ind.roth=grepl("^KLK",tra.human.roth$gene.symbol)
ind.roth =which(ind.roth == TRUE)
Klk.TRA3 = as.data.frame(tra.human.roth[ind.roth,])

# For tra.mouse
ind.mouse=grepl("^Klk",tra.mouse$gene.symbol)
ind.mouse =which(ind.mouse == TRUE)
Klk.TRA4 = as.data.frame(tra.mouse[ind.mouse,])

# For tra.mouse.4301
ind.4301=grepl("^Klk",tra.mouse.4301$gene.symbol)
ind.4301 =which(ind.4301 == TRUE)
Klk.TRA5 = as.data.frame(tra.mouse.4301[ind.4301,])

# For tra.human.gtex 
ind.gtex=grepl("^KLK",tra.human.gtex$ensembl.symbol)
ind.gtex=which(ind.gtex == TRUE)
Klk.TRA6=as.data.frame(tra.human.gtex[ind.gtex,])



### Creating a unified TRA data set, that only contains important information for our overview
nrow.x = nrow(Klk.TRA1) + nrow(Klk.TRA2) + nrow(Klk.TRA3) +nrow(Klk.TRA4)+ nrow(Klk.TRA5) + nrow(Klk.TRA6)
Union.TRA = as.data.frame(matrix(ncol = 6, nrow=nrow.x))
union.names <- c("ensembl_transcript","ensembl_genes", "Chromosome", "gene.symbol", "max.tissue", "dataset") ## assigning the unified columns
colnames(Union.TRA) <- union.names 
## now assign the rows, we poorly need to do this manually since there is no strict order of the columns of the 5 TRA data sets
## Watch out!! Chromosomes on the mice are not comparable to humans!!!
Union.TRA[1:nrow(Klk.TRA1),c(1:5)] <- Klk.TRA1[,c(1,2, 7, 3, 11)] # For Klk.TRA1
Union.TRA[nrow(Klk.TRA1)+1:nrow(Klk.TRA2),c(2:5)] <- Klk.TRA2[,c(1, 2, 6, 11)] # For Klk.TRA2
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+1:nrow(Klk.TRA3),c(1:5)] <- Klk.TRA3[,c(1,2, 7, 3, 11)] # For Klk.TRA3
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+1:nrow(Klk.TRA4),c(1:5)] <- Klk.TRA4[,c(1,2, 7, 3, 11)] # For Klk.TRA4
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+nrow(Klk.TRA4)+1:nrow(Klk.TRA5),c(1:5)] <- Klk.TRA5[,c(1,2, 7, 3, 11)] # For Klk.TRA5
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+nrow(Klk.TRA4)+nrow(Klk.TRA5)+1:nrow(Klk.TRA6),c(1:5)] <- Klk.TRA6[,c(1,2,6,3, 10)] # For Klk.TRA6

## now annotate the the data set
#->> VERY HELPFUL for later plots -> e.g. selecting rows that only apply for the human KLKs: 
#(Union.TRA$dataset == "tra.human")!!!!
Union.TRA[1:nrow(Klk.TRA1), 6] <- "tra.human"
Union.TRA[nrow(Klk.TRA1)+1:nrow(Klk.TRA2),6] <- "tra.human.5median"
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+1:nrow(Klk.TRA3),6] <- "tra.human.roth"
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+1:nrow(Klk.TRA4),6] <- "tra.mouse"
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+nrow(Klk.TRA4)+1:nrow(Klk.TRA5),6] <- "tra.mouse.4301"
Union.TRA[nrow(Klk.TRA1)+nrow(Klk.TRA2)+nrow(Klk.TRA3)+nrow(Klk.TRA4)+nrow(Klk.TRA5)+1:nrow(Klk.TRA6),6] <- "tra.human.gtex"

# Remove identical entries 
new.union <- rownames(unique(Union.TRA[1:4]))
Union.TRA <- Union.TRA[new.union,] 

### all human KLK genes across all TRA data sets

ind.all.tra.human.KLK <- grep("^tra.human", Union.TRA$dataset)
tra.all.human <- Union.TRA[ind.all.tra.human.KLK,]


# all unique human KLK names that are present as TRAs
all.human.KLK.gene.symbols <- unique(Union.TRA$gene.symbol[grep("tra.human", Union.TRA$dataset)])
all.human.KLK.gene.symbols # These are all the available KLK TRAs we can possibly work with
```

## ensembl annotation file
```{r Read ensembl}

setwd(here("Tables"))

ensembl_df = read.csv("ensembl.103.txt",sep="\t")
```

## Setting up rawdata breast cancer data set GSE27830
Creating rawdata breast cancer data set of expression values
```{r Dataset rawdata breast}
rawdataExprs <- rawdataExprs[grepl("ENST", rownames(rawdataExprs)), ]
#dim = 95659    10

# remove .x_at suffix
rownames(rawdataExprs) <- unlist(lapply(strsplit(rownames(rawdataExprs), split = "\\."), "[", 1))
setwd(here("Scripts"))

#Setting up annotation
transcriptIDs <- as.character(ensembl_df[, "Transcript.stable.ID"])
geneSymbol <- as.character(ensembl_df[, "HGNC.symbol"])
names(geneSymbol) <- transcriptIDs

# replace ensemble IDs with HGNC symbol
chipIDs <- rownames(rawdataExprs)
noMatchIDs <- chipIDs[!chipIDs %in% transcriptIDs] # transcript IDs that are not in ensemble_103.txt
length(noMatchIDs) #112

newChipIDs <- chipIDs[chipIDs %in% transcriptIDs]
rawdataExprs <- rawdataExprs[newChipIDs, ]
dim(rawdataExprs) ## 95547    10


# 
symbol <- geneSymbol[newChipIDs]
rownames(rawdataExprs) <- make.names(as.character(symbol), unique = TRUE)  # Matrix to work with

head(rawdataExprs)

```

## Setting up breast cancer data set GSE65216
Creating GSE65216 breast cancer (TNBC Her2 LumA LumB) data of expression values
```{r Dataset breast cancer}
breastExprs <- breastExprs[grepl("ENST", rownames(breastExprs)), ]
dim(breastExprs) #  95659    20

# remove .x_at suffix
rownames(breastExprs) <- unlist(lapply(strsplit(rownames(breastExprs), split = "\\."), "[", 1))

setwd(here("Scripts"))

# replace ensemble IDs with HGNC symbol
chipIDs.breast <- rownames(breastExprs)
noMatchIDs.breast <- chipIDs.breast[!chipIDs.breast %in% transcriptIDs]
# transcript IDs that are not in ensemble_103.txt
length(noMatchIDs.breast) #112

newChipIDs.breast <- chipIDs.breast[chipIDs.breast %in% transcriptIDs]
breastExprs <- breastExprs[newChipIDs.breast, ]
dim(breastExprs) ## 95547    10


# 
symbol.breast <- geneSymbol[newChipIDs.breast]
rownames(breastExprs) <- make.names(as.character(symbol.breast), unique = TRUE)  # Matrix to work with

head(breastExprs)

```
## Setting up lung cancer data set GSE149507 
Creating lung cancer data set to work with
```{r Dataset lung cancer }
lungExprs <- lungExprs[grepl("ENST", rownames(lungExprs)), ]
dim(lungExprs) #  95659    20
 
# remove .x_at suffix
rownames(lungExprs) <- unlist(lapply(strsplit(rownames(lungExprs), split = "\\."), "[", 1))

setwd(here("Scripts"))

# replace ensemble IDs with HGNC symbol
chipIDs.lung <- rownames(lungExprs)
noMatchIDs.lung <- chipIDs.lung[!chipIDs.lung %in% transcriptIDs]
# transcript IDs that are not in ensemble_103.txt
length(noMatchIDs.lung) #112

newChipIDs.lung <- chipIDs.lung[chipIDs.lung %in% transcriptIDs]
lungExprs <- lungExprs[newChipIDs.lung, ]
dim(lungExprs) ## 95547    10


# 
symbol.lung <- geneSymbol[newChipIDs.lung]
rownames(lungExprs) <- make.names(as.character(symbol.lung), unique = TRUE)  # Matrix to work with

head(lungExprs)

```
# 4. Expression Analysis

Overview of gene expression:
The gene expression values are depicted in a logarithmic scale with base 2. This means, that a increase in gene expression by the amount of one unit corresponds with an increase by the factor 2. 

## Extraction of KLK and KLK TRA genes
Ioslating all KLK genes from each dataset 
```{r Isolating KLK genes, results=FALSE}
# All KLK genes
ind.all.KLK=grep("^KLK",symbol)
Kallikreins.all=symbol[ind.all.KLK]

ind.all.human =which(as.character(symbol) %in% as.character(Kallikreins.all)) 
Kallikreins<- as.character(Kallikreins.all)
Kallikreins

## For first breast cancer data set
all.KLK.rawdata = rawdataExprs[ind.all.human,]

## For breast GSE
all.KLK.breast = breastExprs[ind.all.human,]

## For lung GSE 
all.KLK.lung = lungExprs[ind.all.human,]

#TRA: 
## For this we use the human TRA data set (Uhlén et al. 2015) <- right one?
#TRA.symbol=tra.human[,3] # lists all TRAs names for humans
#KLK.genes.TRA.human = TRA.symbol[ind.human] 
#ind.KLK.tra.human =which(as.character(symbol) %in% KLK.genes.TRA.human) 
#ind.KLK.tra.human gives respective column 
## For first breast cancer data set
##TRA.KLK.rawdata = rawdataExprs[ind.KLK.tra.human,]

## For breast GSE
#TRA.KLK.breast = breastExprs[ind.KLK.tra.human,]

## For lung GSE 
#TRA.KLK.lung = lungExprs[ind.KLK.tra.human,]



ind.KLK.all.human.tra <- which(as.character(names(symbol)) %in% 
as.character(unlist(tra.all.human[1])))

## For first breast cancer data set
TRA.KLK.rawdata = rawdataExprs[ind.KLK.all.human.tra,]

## For breast GSE
TRA.KLK.breast = breastExprs[ind.KLK.all.human.tra,]

## For lung GSE 
TRA.KLK.lung = lungExprs[ind.KLK.all.human.tra,]
```

## 4.1 breast cancer GSE27830 (Nagel et al. 2012)

### Clean up identical isoforms
```{r cleanup of identical isoforms}
#transform dataset - genes on columns
df.TRA.KLK.rawdata <- data.frame(t(TRA.KLK.rawdata))

#Define function for calculating correlation 
cor.genes <- function(df){
  df.cor <- cor(df, method = "pearson")
  diag(df.cor)=NA
  df.cor[upper.tri(df.cor)]=NA
  return(data.frame(df.cor))
}

#test function 
cor.TRA.KLK.rawdata <- cor.genes(df.TRA.KLK.rawdata)
length(which(cor.TRA.KLK.rawdata == 1))
#14 isoforms are not unique
#Identical transcripts can cause issues later on for statistical testing ect. 

#remove identical columns
TRA.KLK.rawdata.clean <- df.TRA.KLK.rawdata[!duplicated(unclass(df.TRA.KLK.rawdata))]
dim(TRA.KLK.rawdata)
dim(TRA.KLK.rawdata.clean)
#removed 12 identical isoforms

#check for identicals 
cor.TRA.KLK.rawdata.clean <- cor.genes(TRA.KLK.rawdata.clean)
length(which(cor.TRA.KLK.rawdata.clean == 1))
#no more identical columns!

# First we need to clean up the isoforms again
df.all.KLK.rawdata <- data.frame(t(all.KLK.rawdata))

#test function 
cor.all.KLK.rawdata <- cor.genes(df.all.KLK.rawdata)
length(which(cor.all.KLK.rawdata == 1))
#40 isoforms are not unique
#Identical transcripts can cause issues later on for statistical testing etc. 

#remove identical isoforms
all.KLK.rawdata.clean <- df.all.KLK.rawdata[!duplicated(unclass(df.all.KLK.rawdata))]
dim(all.KLK.rawdata.clean) #removed 39 identical isoforms


# Use mixedsort function from the gtool package to sort the KLKs correctly
order.rawdata <- mixedsort(colnames(all.KLK.rawdata.clean))
order.rawdata <- order.rawdata[c(1:3,11:17,4:10,18,19,24:27,20:23,28:73)]
all.KLK.rawdata.clean <- t(all.KLK.rawdata.clean[,order.rawdata])
#sorted all.KLK.breast.clean
all.KLK.rawdata.clean <- data.frame(all.KLK.rawdata.clean)


order.rawdata1 <- mixedsort(colnames(TRA.KLK.rawdata.clean))
order.rawdata1 <- order.rawdata1[c(1:3,11:17,4:10,18,19,23:26,20:22,27:63)]
TRA.KLK.rawdata.clean <- t(TRA.KLK.rawdata.clean[,order.rawdata1])
TRA.KLK.rawdata.clean <- data.frame(TRA.KLK.rawdata.clean)
```
Further reason for removing the identicals
Now, there are no KLK TRA genes that have identical expression values. This could have caused problems for statistical testing or for performing correlation analysis later on. Before starting off with the analysis, we should get an overview of KLK TRAs for the three data sets via Heatmaps. 

### Visualization of gene expression
#### Overview gene expression
```{r Overview and Histogram rawdata breast}
gene.summary <- function(x){
 round(apply(x, 2, summary), digits = 2)
}
gene.summary(rawdataExprs)

#Histograms of genes expression
#rawdataExprs i snot cleaned of identical values


setwd(here("Plots/Plots_raw_breast"))
pdf("Histogram_gene_expression_lines.pdf")
hist(rawdataExprs, 
     main="Gene expression GSE27830 (Nagel et al. 2021)",
     xlab="Expression value",
     breaks=40
     )
abline(v = median(rawdataExprs), col= "red", lwd = 3)#median all genes
abline(v = median(unlist(all.KLK.rawdata.clean)), col= "blue", lwd = 3) #median all KLKs
legend("topright", c("median all genes","median all KLKs"), col=c("red","blue"), lwd=10)
dev.off()
```
We can see here, that the lowest value is about 5 while the highest gene expression value is around 15. Due to the logarithmic scale, an expression value of 15 is 1024 times higher than the value 5. It is also notable, that the median does not fluctuate too much for all samples.This is due to the fact, that we performed vsn normalization.

#### Boxplots
```{r Boxplots - rawdata breast}
#Boxplot:
setwd(here("Plots/Plots_raw_breast"))
pdf("all_KLK_boxplot_rawdata.pdf")
boxplot(t(all.KLK.rawdata.clean),
        col=rainbow(length(rownames(all.KLK.rawdata.clean))),main="Gene expression of KLKs GSE27830 (Nagel et al. 2012)",
        cex.axis=0.4,
        las = 2,
        ylim = c(4,10.5)
        )
abline(h=median(unlist(all.KLK.rawdata.clean)), col= "red", lwd = 2, lty = 2)
legend("topright", c("median all KLKs"), col=c("red"), lwd=10)
dev.off()

#split the boxplots, because it is hard to read the gene names
#Boxplot1 
setwd(here("Plots/Plots_raw_breast"))
pdf("all_KLK_boxplot_rawdata1.pdf")
boxplot(t(all.KLK.rawdata.clean[1:24,]),
        col=rainbow(length(rownames(all.KLK.rawdata.clean[1:24,]))),main="Gene expression of KLKs GSE27830 (Nagel et al. 2012)",
        cex.axis=0.8,
        las = 2
)
abline(h=median(unlist(all.KLK.rawdata.clean)), col= "red", lwd = 2, lty = 2)
legend("topright", c("median all KLKs"), col=c("red"), lwd=10)
dev.off()

#Boxplot2
setwd(here("Plots/Plots_raw_breast"))
pdf("all_KLK_boxplot_rawdata2.pdf")
boxplot(t(all.KLK.rawdata.clean[25:48,]),
        col=rainbow(length(rownames(all.KLK.rawdata.clean[25:48,]))),main="Gene expression of KLKs GSE27830 (Nagel et al. 2012)",
        cex.axis=0.8,
        las = 2,
        )
abline(h=median(unlist(all.KLK.rawdata.clean)), col= "red", lwd = 2, lty = 2)
legend("topright", c("median all KLKs"), col=c("red"), lwd=10)
dev.off()

#Boxplot3
setwd(here("Plots/Plots_raw_breast"))
pdf("all_KLK_boxplot_rawdata3.pdf")
boxplot(t(all.KLK.rawdata.clean[49:73,]),
        col=rainbow(length(rownames(all.KLK.rawdata.clean[49:73,]))),main="Gene expression of KLKs GSE27830 (Nagel et al. 2012)",
        cex.axis=0.8,
        las = 2
        )
abline(h=median(unlist(all.KLK.rawdata.clean)), col= "red", lwd = 2, lty = 2)
legend("topright", c("median all KLKs"), col=c("red"), lwd=10)
dev.off()
```

#### Heatmap
```{r Heatmaps annotation - rawdata breast}

# We first want to see the dendrogram of the heatmap
# By that we can decide to put certain number of clusters into the heatmap
# This clustering is not meant for analytical purpose -> see k-means later on for this
# This clustering is just intended to increase the visual clarity


# 1. show simple cluster annotated to heatmap
hclust.genes.rawdata <- hclust(dist(all.KLK.rawdata.clean), method = "complete")

dendr.raw <- dendro_data(hclust.genes.rawdata, type = "rectangle")
clust.raw    <- cutree(hclust.genes.rawdata, k=3)
# we tested a few clustering´, however 3 cluster seem to visualize the expression in the more up-regulated genes better
clust.raw.df <- data.frame(label=names(clust.raw), cluster=factor(clust.raw))
dendr.raw[["labels"]] <- merge(dendr.raw[["labels"]],clust.raw.df, by="label")


setwd(here("Plots/Plots_raw_breast"))
pdf("all_KLK_dendrogram_raw.pdf")
ggplot() +
  ggtitle("Dendrogram of all KLKs GSE27830 (Nagel et al. 2012)") +
  ylab("Distance (complete-linkage)") +
  geom_segment(data=segment(dendr.raw), aes(x=x, y=y, xend=xend, yend=yend)) + 
  geom_text(data=label(dendr.raw), aes(x, y, label=label, hjust=0, color=cluster), 
           size=2) +
  coord_flip() + scale_y_reverse(expand=c(0.1, 0)) + 
  theme(axis.line.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.y=element_blank(),
        axis.title.y=element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        legend.position = c(0.05, 0.05),
        legend.justification = c("left", "bottom")
        )
dev.off()



# adding annotations  to the KLK genes, for whether they are TRAs or not

# Lets try it with 3 clusters in the heatmap, just for visualization purposes
annotation.hmap.rawdata <- cutree(tree = as.dendrogram(hclust.genes.rawdata), k = 3)
annotation.hmap.rawdata  <- data.frame(cluster = ifelse(test = annotation.hmap.rawdata == 1, yes = "cluster 1", no = ifelse(test = annotation.hmap.rawdata == 2, yes = "cluster 2", no = "cluster 3" )))

# 2. differentiate between TRA-/ or Non-TRA-KLKs
TRUEorFALSE.raw <- rownames(all.KLK.rawdata.clean) %in% rownames(TRA.KLK.rawdata.clean)
TRAorNON.raw <- ifelse(TRUEorFALSE.raw == TRUE, "TRA", "non-TRA")
annotation.hmap.rawdata$TRA_or_NON  <- TRAorNON.raw
colnames(annotation.hmap.rawdata)[2] <- "TRA or not"

# call the annotated Heatmap
setwd(here("Plots/Plots_raw_breast"))
pdf("all_KLK_Heatmap_rawdata.pdf")
pheatmap(all.KLK.rawdata.clean,
         main = "Expression values of all KLKs GSE27830 (Nagel et al. 2012)",
         annotation_row = annotation.hmap.rawdata,
         cellwidth = 27,
         cellheight = 5.5,
         treeheight_col = 10,
         fontsize_row = 6,
         cutree_rows = 3
)
dev.off()
```



### PCA
```{r PCA - rawdata breast}
# center = TRUE
# scale = FALSE, since we already did this in the vsn normalization
pr.all.KLK.rawdata.clean <- prcomp(all.KLK.rawdata.clean, center = TRUE, scale = FALSE)
#call summary
summary(pr.all.KLK.rawdata.clean)
# normally the first principle component are sufficient

# goal is to get information of roughly 95% of the KLK data -> 95% cumulative variance
# in the summary we can see that this is almost achieved by the 4th pc
# the variance of the pcs rapidly declines, there is only about 5% variance for the 4th pc

# get graphical overview (not really needed though)

screeplot(pr.all.KLK.rawdata.clean, type = "l", npcs = 10, main = "Screeplot of the 10 PCs")
#abline(h = 1, col="red", lty=5)
#legend("topright", legend=c("Eigenvalue = 1"),
##       col=c("red"), lty=5, cex=0.6)

variance.raw <- cumsum(pr.all.KLK.rawdata.clean$sdev^2 / sum(pr.all.KLK.rawdata.clean$sdev^2))
plot(variance.raw[0:10], xlab = "PC #", ylab = "Amount of explained variance", main = "Cumulative variance plot")
abline(v = 3, col="blue", lty=5)
abline(h = 0.9320 , col="blue", lty=5)
legend("topleft", legend=c("Cut-off @ PC3"),
       col=c("blue"), lty=5, cex=0.6)

# we can see in this cumulative variance plot, that we only need about 3 principle components to achieve a a total variance of 93,20 %
# the others do not represent a lot of information

# Plot the PCs after TRAorNOT
color.raw = c(ifelse(annotation.hmap.rawdata[,2] == "TRA", "blue", "red"))

plot(pr.all.KLK.rawdata.clean$x[,1],pr.all.KLK.rawdata.clean$x[,2],
     xlab="PC1 (53,38%)",
     ylab = "PC2 (27,88%)",
     main = "PC1 / PC2 - plot",
     col = color.raw)
legend("topleft", legend=c("TRA", "non-TRA"),
       col=c("blue", "red"), lwd= 10, cex=0.6)

plot(pr.all.KLK.rawdata.clean$x[,1],pr.all.KLK.rawdata.clean$x[,3],
     xlab="PC2 (53,38%)",
     ylab = "PC3 (11,94%)",
     main = "PC2 / PC3 - plot",
    col = color.raw)
legend("topleft", legend=c("TRA", "non-TRA"),
       col=c("blue", "red"), lwd= 10, cex=0.6)
# Since almost all KLKs are TRAs, annotating the color does not really show anything
# Rather annotate the color for the type of cancer
```

### Clustering - kmeans

```{r kmeans - rawdata breast}

```

### Hypothesis testing

```{r Hypothesis testing - rawdata breast}
#Hypothesis testing 
#according to the histogram previously created, KLKs are not normally distributed

#QQ Plot
setwd(here("Plots/Plots_raw_breast"))
pdf("QQPlot_rawdata.pdf")
qqnorm(rawdataExprs, main = "QQ-Plot GSE27830");qqline(rawdataExprs)
dev.off()
#not a normal distribution!!! -> Wilcoxon rank test 

#The defined function below will return a dataframe with all the p-values between the specific gene and all other genes in the dataframe. 

w.test.gene.df.twosided <- function(gene, df){
  apply(as.matrix(df), 1, function(x){
  test <- wilcox.test(as.matrix(gene), x, 
                      alternative = "two.sided", 
                      exact = FALSE, 
                      paired = FALSE,
                      digits.rank = 10)$p.value})
}

#same function but with a different alternative
w.test.gene.df.upper <- function(gene, df){
  apply(as.matrix(df), 1, function(x){
  test <- wilcox.test(as.matrix(gene), x,  
                      alternative = "greater", 
                      exact = FALSE, 
                      paired = FALSE,
                      digits.rank = 10)$p.value})

}


#test use of the function with KLK1
KLK3.vs.all <- w.test.gene.df.upper(all.KLK.rawdata.clean["KLK3", ], all.KLK.rawdata.clean)
KLK3.vs.all
#Some genes have the same p-value. This is due to ties in the rank. The numeric diffrences in those ties should be quite small due to the high digit rank of 10. 

#the function significant.pvalues will return a matrix with two columns (significant, non-significant) - according to the count of entries which are above 0.05 and lower/equal to 0.05. 

significant.pvalues.05 <- function(genevsall) {
  data <- matrix(nrow=1, ncol=2)
  colnames(data) <- c("significant", "non-significant")
  data[ ,1] <- length(which(genevsall <= 0.05))
  data[ ,2] <- length(which(genevsall > 0.05))
  out <- data

  return(out)
  
}

#Same function but with the confidence interval of 0.1 
significant.pvalues.1 <- function(genevsall) {
  data <- matrix(nrow=1, ncol=2)
  colnames(data) <- c("significant", "non-significant")
  data[ ,1] <- length(which(genevsall <= 0.1))
  data[ ,2] <- length(which(genevsall > 0.1))
  out <- data

  return(out)
  
}

#test run 
KLK3.vs.all.sig <- significant.pvalues.05(KLK3.vs.all) 
KLK3.vs.all.sig

#simple barplot 
barplot(KLK3.vs.all.sig)

#the test with KLK3 shows that the gene is significantly different expressed compared to all other genes of the dataset.
```



```{r Test applications}
#Cluster 1 in the heatmap of all KLKs is higher expressed than the other clusters 
#KLK1, KLK1.3, KLK4.4 and KLK8.8 are in this cluster

#Wilcoxon rank sum test on KLK1 - upper sided test 
KLK1.vs.all <- w.test.gene.df.upper(all.KLK.rawdata.clean["KLK1", ], all.KLK.rawdata.clean)
KLK1.vs.all.sig <- significant.pvalues.05(KLK1.vs.all)
KLK1.vs.all.sig
which(KLK1.vs.all > 0.05)
#KLK1 is significantly higher expressed compared to most other genes, except the other genes of cluster 1.

#KLK1.3
KLK1.3.vs.all <- w.test.gene.df.upper(all.KLK.rawdata.clean["KLK1.3", ], all.KLK.rawdata.clean)
KLK1.3.vs.all.sig <- significant.pvalues.05(KLK1.3.vs.all)
KLK1.3.vs.all.sig
which(KLK1.3.vs.all > 0.05)
#KLK1 is significantly higher expressed compared to most other genes, except the other genes of cluster 1. 

#KLK4.4 
KLK4.4.vs.all <- w.test.gene.df.upper(all.KLK.rawdata.clean["KLK4.4", ], all.KLK.rawdata.clean)
KLK4.4.vs.all.sig <- significant.pvalues.05(KLK4.4.vs.all)
KLK4.4.vs.all.sig
which(KLK4.4.vs.all > 0.05)
##KLK1 is significantly higher expressed compared to all other genes.

#KLK8.8 
KLK8.8.vs.all <- w.test.gene.df.upper(all.KLK.rawdata.clean["KLK8.8", ], all.KLK.rawdata.clean)
KLK8.8.vs.all.sig <- significant.pvalues.05(KLK8.8.vs.all)
KLK8.8.vs.all.sig
which(KLK8.8.vs.all > 0.05)
##KLK1 is significantly higher expressed compared to all other genes, except KLK4.4. 

#The tests confirm that cluster 1 is significantly higher expressed than the other genes. 
#Therefore the expression rate of those KLKs could be potential markers for the regression analysis. 
#Unfortunately in this test dataset we do not have data from normal tissue. 
#That is why we will perform the regression analysis on the lung dataset. 

```




## 4.2 Breast cancer GSE65216 (Maire et al. 2013)

### Clean up identical transcripts
```{r cleanup identical transcripts - breast cancer}

#transform dataset - genes on columns
df.TRA.KLK.breast <- data.frame(t(TRA.KLK.breast))

#number of identical columns
cor.TRA.KLK.breast <- cor.genes(df.TRA.KLK.breast)
length(which(cor.TRA.KLK.breast == 1))
# 5 identical isoforms 

#cleanup function 
genes.cleanup <- function(df){
  df[!duplicated(unclass(df))]
}

#remove identical columns
TRA.KLK.breast.clean <- genes.cleanup(df.TRA.KLK.breast)
dim(TRA.KLK.breast.clean)

#check clean dataframe for correlations
length(which(cor.genes(TRA.KLK.breast.clean)==1)) #0 correlations
TRA.KLK.breast.clean

# Creating a heatmap, that displays all KLKs and indicates the ones who are TRAs

# First we need to clean up the isoforms again
df.all.KLK.breast <- data.frame(t(all.KLK.breast))

#test function 
cor.all.KLK.breast <- cor.genes(df.all.KLK.breast)
length(which(cor.all.KLK.breast == 1))
#49 isoforms are not unique
#Identical transcripts can cause issues later on for statistical testing etc. 

#remove identical isoforms
all.KLK.breast.clean <- df.all.KLK.breast[!duplicated(unclass(df.all.KLK.breast))]
dim(all.KLK.breast.clean) #removed 39 identical isoforms


# Use mixedsort function from the gtool package to sort the KLKs correctly
order.breast <- mixedsort(colnames(all.KLK.breast.clean))
order.breast <- order.breast[c(1:3,11:17,4:10,18,19,24:27,20:23,28:73)]
all.KLK.breast.clean <- t(all.KLK.breast.clean[,order.breast]) #sorted all.KLK.breast.clean
all.KLK.breast.clean <- data.frame(all.KLK.breast.clean)


order.breast1 <- mixedsort(colnames(TRA.KLK.breast.clean))
order.breast1 <- order.breast1[c(1:3,11:17,4:10,18,19,23:26,20:22,27:63)]
TRA.KLK.breast.clean <- t(TRA.KLK.breast.clean[,order.breast1])
TRA.KLK.breast.clean <- data.frame(TRA.KLK.breast.clean)


```
### Visualization of gene expression 
#### Overview gene expression
```{r Overview and Histogram - breast cancer}
gene.summary(breastExprs)

setwd(here("Plots/Plots_breast"))
pdf("Histogram_gene_expression_lines.pdf")
hist(breastExprs, 
     main="Gene expression GSE27830 (Nagel et al. 2021)",
     xlab="Expression value",
     breaks=40
     )
abline(v = median(breastExprs), col= "red", lwd = 3)#median all genes
abline(v = median(unlist(all.KLK.breast.clean)), col= "blue", lwd = 3) #median all KLKs
legend("topright", c("median all genes","median all KLKs"), col=c("red","blue"), lwd=10)
dev.off()
```

#### Boxplots 
```{r Boxplots - breast cancer}
#boxplots 
setwd(here("Plots/Plots_breast"))
pdf("all_KLK_boxplot_breast.pdf")
boxplot(t(all.KLK.breast.clean),
        col=rainbow(length(rownames(all.KLK.breast.clean))),
        main="Gene expression of KLKs (Maire et al. 2013)",
        cex.axis=0.4,
        las = 2,
        )
abline(h=median(unlist(all.KLK.breast.clean)), col= "red", lwd = 2, lty = 2)
legend("topright", c("median all KLKs"), col=c("red"), lwd=10)
dev.off()


#split the boxplots, because it is hard to read the gene names
#Boxplot1 
setwd(here("Plots/Plots_breast"))
pdf("all_KLK_boxplot_breast1.pdf")
boxplot(t(all.KLK.breast.clean[1:24,]),
        col=rainbow(length(rownames(all.KLK.breast.clean[1:24,]))),main="Gene expression of KLKs (Maire et al. 2013)",
        cex.axis=0.8,
        las = 2,
        )
abline(h=median(unlist(all.KLK.breast.clean)), col= "red", lwd = 2, lty = 2)
legend("topright", c("median all KLKs"), col=c("red"), lwd=10)
dev.off()

#Boxplot2
setwd(here("Plots/Plots_breast"))
pdf("all_KLK_boxplot_breast2.pdf")
boxplot(t(all.KLK.breast.clean[25:48,]),
        col=rainbow(length(rownames(all.KLK.breast.clean[25:48,]))),main="Gene expression of KLKs (Maire et al. 2013)",
        cex.axis=0.8,
        las = 2,
        )
abline(h=median(unlist(all.KLK.breast.clean)), col= "red", lwd = 2, lty = 2)
legend("topright", c("median all KLKs"), col=c("red"), lwd=10)
dev.off()

#Boxplot3
setwd(here("Plots/Plots_breast"))
pdf("all_KLK_boxplot_breast3.pdf")
boxplot(t(all.KLK.breast.clean[49:73,]),
        col=rainbow(length(rownames(all.KLK.breast.clean[49:73,]))),main="Gene expression of KLKs (Maire et al. 2013)",
        cex.axis=0.8,
        las = 2,
        )
abline(h=median(unlist(all.KLK.breast.clean)), col= "red", lwd = 2, lty = 2)
legend("topright", c("median all KLKs"), col=c("red"), lwd=10)
dev.off()


```


#### Heatmap
```{r Heatmap annotation - breast cancer}
# We first want to see the dendrogram of the heatmap
# By that we can decide to put certain number of clusters into the heatmap
# This clustering is not meant for analytical purpose -> see k-means later on for this
# This clustering is just intended to increase the visual clarity


# 1. show simple cluster annotated to heatmap
hclust.genes.breast <- hclust(dist(all.KLK.breast.clean), method = "complete")

dendr.breast <- dendro_data(hclust.genes.breast, type = "rectangle")
clust.breast    <- cutree(hclust.genes.breast, k=3)
# we tested a few clustering´, however 3 cluster seem to visualize the expression in the more up-regulated genes better
clust.breast.df <- data.frame(label=names(clust.breast), cluster=factor(clust.breast))
dendr.breast[["labels"]] <- merge(dendr.breast[["labels"]],clust.breast.df, by="label")


setwd(here("Plots/Plots_breast"))
pdf("all_KLK_dendrogram_breast.pdf")
ggplot() +
  ggtitle("Dendrogram of all KLKs GSE65216 (Maire et al. 2013)") +
  ylab("Distance (complete-linkage)") +
  geom_segment(data=segment(dendr.breast), aes(x=x, y=y, xend=xend, yend=yend)) + 
  geom_text(data=label(dendr.breast), aes(x, y, label=label, hjust=0, color=cluster), 
           size=2) +
  coord_flip() + scale_y_reverse(expand=c(0.1, 0)) + 
  theme(axis.line.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.y=element_blank(),
        axis.title.y=element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        legend.position = c(0.05, 0.95),
        legend.justification = c("left", "top")
        )
dev.off()


# adding annotations to the KLK genes, for whether they are TRAs or not
# Lets try it with 3 clusters in the heatmap, just for visualization purposes
annotation.hmap.breast <- cutree(tree = as.dendrogram(hclust.genes.breast), k = 3)
annotation.hmap.breast  <- data.frame(cluster = ifelse(test = annotation.hmap.breast == 1, yes = "cluster 1", no = ifelse(test = annotation.hmap.breast == 2, yes = "cluster 2", no = "cluster 3" )))

# 2. differentiate between TRA-/ or Non-TRA-KLKs
# 2. differentiate between TRA-/ or Non-TRA-KLKs
TRUEorFALSE.breast <- rownames(all.KLK.breast.clean) %in% rownames(TRA.KLK.breast.clean)
TRAorNON.breast <- ifelse(TRUEorFALSE.breast == TRUE, "TRA", "non-TRA")
annotation.hmap.breast$TRA_or_NON  <- TRAorNON.breast
colnames(annotation.hmap.breast)[2] <- "TRA or not"

# 3. To increase visibility, we will cut off the GSM number for the patient names
all.KLK.breast.clean.noGSM <- all.KLK.breast.clean
#rownames(all.KLK.breast.clean.noGSM) <- sub("...........", "", rownames(all.KLK.breast.clean)) 
#There seems to be the problem, that the sample names cant be shortened or else Tum016_Her2 would appear twice, which is not possible
#However these are two different samples, which where taken out of the same tumor, hence the different GSM number
#The different of these samples is their replication protein repA for GSM1588991 and repB for GSM1588992
#Here we will just shorten the name by 2 characters less, so we can differentiate between those two 
colnames(all.KLK.breast.clean.noGSM)[1:8] <- sub("...........", "", colnames(all.KLK.breast.clean)[1:8])
colnames(all.KLK.breast.clean.noGSM)[9:10] <- sub(".........", "", colnames(all.KLK.breast.clean)[9:10])
colnames(all.KLK.breast.clean.noGSM)[11:20] <- sub("...........", "", colnames(all.KLK.breast.clean)[11:20])

# annotation for the samples
samples.annotation.breast <- data.frame(matrix(ncol = 1, nrow = ncol(all.KLK.breast.clean.noGSM)))
colnames(samples.annotation.breast) <- "tumor type"                                 
rownames(samples.annotation.breast) <- colnames(all.KLK.breast.clean.noGSM)
ph.ind.Her2 = grep(pattern = "Her2", colnames(all.KLK.breast.clean.noGSM))
ph.ind.LumA = grep(pattern = "LumA", colnames(all.KLK.breast.clean.noGSM))
ph.ind.LumB = grep(pattern = "LumB", colnames(all.KLK.breast.clean.noGSM))
ph.ind.TNBC = grep(pattern = "TNBC", colnames(all.KLK.breast.clean.noGSM))
samples.annotation.breast[ph.ind.Her2,1] <- "Her2"
samples.annotation.breast[ph.ind.LumA,1] <- "LumA"
samples.annotation.breast[ph.ind.LumB,1] <- "LumB"
samples.annotation.breast[ph.ind.TNBC,1] <- "TNBC"

# call the annotated Heatmap
setwd(here("Plots/Plots_breast"))
pdf("all_KLK_Heatmap_breast.pdf")
pheatmap(all.KLK.breast.clean.noGSM,
         main = "Expression values of all KLKs GSE65216 (Maire et al. 2013)",
         annotation_row = annotation.hmap.breast,
         annotation_col = samples.annotation.breast,
         cellwidth = 12.7,
         cellheight = 5,
         treeheight_col = 10,
         fontsize_row = 6,
         cutree_rows = 3
)
dev.off()
```

 ### PCA
```{r PCA - breast cancer}
# run PCA on the KLK data frame
# center and scale = TRUE ? does vsnrma not already do this?
pr.all.KLK.breast.clean <- prcomp(all.KLK.breast.clean, center = TRUE, scale = FALSE)
#call summary
summary(pr.all.KLK.breast.clean)
# goal is to get information of roughly 95% of the KLK data -> 95% cumulative variance
# in the summary we can see that this is almost achieved by the 4th pc
# the variance of the pcs rapidly declines, there is only about 5% variance for the 4th pc

# get graphical overview (not really needed though)

screeplot(pr.all.KLK.breast.clean, type = "l", npcs = 10, main = "Screeplot of the first 10 PCs")
#abline(h = 1, col="red", lty=5)
#legend("topright", legend=c("Eigenvalue = 1"),
#       col=c("red"), lty=5, cex=0.6)

variance.breast <- cumsum(pr.all.KLK.breast.clean$sdev^2 / sum(pr.all.KLK.breast.clean$sdev^2))
plot(variance.breast[0:20], xlab = "PC #", ylab = "Amount of explained variance", main = "Cumulative variance plot")
abline(v = 3, col="blue", lty=5)
abline(h = 0.86583, col="blue", lty=5)
legend("topleft", legend=c("PC3"),
       col=c("blue"), lty=5, cex=0.6)

# we can see in this cumulative variance plot, that we only need about 3 principle components to achieve a a total variance of 86,6 %
# This seems to be a very low number of pcs, the others do not represent a lot of information

# Plot the PCs
# for TRAorNON
color.breast = c(ifelse(annotation.hmap.breast[,2] == "TRA", "blue", "red"))

plot(pr.all.KLK.breast.clean$x[,1],pr.all.KLK.breast.clean$x[,2],
     xlab="PC1 (51,8%)",
     ylab = "PC2 (21,47%)",
     pch = 18,
     main = "PC1 / PC2 - plot",
     col = color.breast)
legend("topleft", legend=c("TRA", "non-TRA"),
       col=c("blue", "red"), lwd= 10, cex=0.6)

plot(pr.all.KLK.breast.clean$x[,1],pr.all.KLK.breast.clean$x[,3],
     xlab="PC1 (51,8%)",
     ylab = "PC3 (13,41%)",
     pch = 18,
     main = "PC1 / PC3 - plot",
     col = color.breast)
legend("topleft", legend=c("TRA", "non-TRA"),
       col=c("blue", "red"), lwd= 10, cex=0.6)
# Since almost all KLKs are TRAs, annotating the color does not really show anything
# Rather annotate the color for the type of cancer

## Group after treatment
color.breast.mutation = c(ifelse(samples.annotation.breast == "TNBC", yes = "blue", no = ifelse(samples.annotation.breast == "Her2", yes = "red", no = ifelse(samples.annotation.breast == "LumA", yes = "green", no = "yellow"))))

plot(pr.all.KLK.breast.clean$x[,1],pr.all.KLK.breast.clean$x[,2],
     xlab="PC1 (51,8%)",
     ylab = "PC2 (21,47%)",
     pch = 18,
     main = "PC1 / PC2 - plot, mutation",
     col = color.breast.mutation)
legend("topleft", legend=c("TNBC", "Her2","LumA", "LumB"),
       col=c("blue", "red", "green", "yellow"), lwd= 7, cex=0.6)

plot(pr.all.KLK.breast.clean$x[,1],pr.all.KLK.breast.clean$x[,3],
     xlab="PC1 (51,8%)",
     ylab = "PC3 (13,41%)",
     main = "PC1 / PC3 - plot",
     pch = 18,
     col = color.breast.mutation)
legend("bottomleft", legend=c("TNBC", "Her2","LumA", "LumB"),
       col=c("blue", "red", "green", "yellow"), lwd= 7, cex=0.6)

### DO IT WITH CLUSTERING FROM k-means
### Mutation and tra-or-not differentation does not represent any information

```

### Clustering - kmeans
```{r k-means - breast cancer}

```

###Hypothesis testing 
```{r Hypothesis testing - breast}
#checking the normaility of the breast cancer dataset 
#QQ Plot
setwd(here("Plots/Plots_breast"))
png("QQPlot_breast.png")
qqnorm(breastExprs, main = "QQ-Plot GSE65216 (Maire et al. 2013)");qqline(breastExprs)
dev.off()
#the data is not normally distributed - therefore we will use the Wilcoxon Rank Sum test/ Mann-Whitney U test

#In the heatmap KLK4.4 in its own cluster 2 seems to be higher expressed than the other genes.
#Test weather KLK4.4 is significantly higher expressed
b.KLK4.4vsall<- w.test.gene.df.upper(all.KLK.breast.clean["KLK4.4", ], all.KLK.breast.clean)
b.KLK4.4.vsall.sig <- significant.pvalues.05(b.KLK4.4vsall)
b.KLK4.4.vsall.sig
which(b.KLK4.4vsall > 0.05)
#KLK4.4 is significantly higher expressed than all other genes, execpt itself.
```


```{r Hypothesis testing - cluster 2}
#Test whether genes in cluster 3 are higher expressed than the other genes. 
#KLK10.3, KLK6, KLK5, KLK5.3 are in cluster 3. 

#KLK10.3
b.KLK10.3vsall <- w.test.gene.df.upper(all.KLK.breast.clean["KLK10.3", ], all.KLK.breast.clean)
b.KLK10.3.vsall.sig <- significant.pvalues.05(b.KLK10.3vsall)
b.KLK10.3.vsall.sig
#KLK10.3 is not significantly higher expressed, compared to the other genes. 

#KLK6
b.KLK6vsall <- w.test.gene.df.upper(all.KLK.breast.clean["KLK6", ], all.KLK.breast.clean)
b.KLK6vsall.sig <- significant.pvalues.05(b.KLK6vsall)
b.KLK6vsall.sig
#KLK6 is only significantly higher expressed than two thirds of the genes. 

#KLK5
b.KLK5vsall <- w.test.gene.df.upper(all.KLK.breast.clean["KLK5", ], all.KLK.breast.clean)
b.KLK5vsall.sig <- significant.pvalues.05(b.KLK5vsall)
b.KLK5vsall.sig
#KLK5 is not significantly higher expressed than most of the other genes

#KLK5.3
b.KLK5.3vsall <- w.test.gene.df.upper(all.KLK.breast.clean["KLK5.3", ], all.KLK.breast.clean)
b.KLK5.3vsall.sig <- significant.pvalues.05(b.KLK5.3vsall)
b.KLK5.3vsall.sig
#KLK5.3 is only significantly higher expressed than half of the genes. 
```
Conclusion: the KLKs in cluster two are not clearly higher expressed than the other genes. They only got in the same cluster together because the genes of cluster two are higher expressed in 5 of the chips (Tum01+0.4+06_TNBC, Tum103_LumB, Tum075_LumA). 

```{r, Hypothesis testing  KLK8.3, 8.8}
#In the heatmap KLK8.3, KLK8.8 seem to be higher expressed than the other genes. 

#Hypothesis test on KLK8.3 (TRA)
b.KLK8.3vsall <- w.test.gene.df.upper(all.KLK.breast.clean["KLK8.3", ], all.KLK.breast.clean)
b.KLK8.3vsall.sig <- significant.pvalues.05(b.KLK8.3vsall)
b.KLK8.3vsall.sig
which(b.KLK8.3vsall > 0.05)
#KLK8.3 is significantly higher expressed than all genes, except KLK4.4 and KLK8.8

#KLK8.8 (NON-TRA)
b.KLK8.8vsall <- w.test.gene.df.upper(all.KLK.breast.clean["KLK8.8", ], all.KLK.breast.clean)
b.KLK8.8vsall.sig <- significant.pvalues.05(b.KLK8.8vsall)
b.KLK8.8vsall.sig
which(b.KLK8.8vsall > 0.05)
#KLK8.8 is significantly higher expressed than all other genes, except KLK4.4
```
Conclusion: KLK8.3 and KLK8.8 are significantly over-expressed, compared to the other genes, except KLK4.4. Especially KLK8.8 is interesting because it is not an TRA. This means there is likely no central tolerance against this KLK, which makes it a good drug target. 

## 4.3 Lung cancer GSE149507 (Cai et al. 2021)

### Clean up identical isoforms
```{r clean up idencticals - lung cancer}

#transform dataset - genes on columns
df.TRA.KLK.lung <- data.frame(t(TRA.KLK.lung))

#finding duplicates
cor.TRA.KLK.lung <- cor.genes(df.TRA.KLK.lung)
length(which(cor.TRA.KLK.lung == 1)) #14 identical columns <- why does it always change (5, 14 ,18) <- maybe mistake in ind.human?

#removing identical columns
TRA.KLK.lung <- data.frame(TRA.KLK.lung)
TRA.KLK.lung.clean <- genes.cleanup(df.TRA.KLK.lung)
length(which(cor.genes(TRA.KLK.lung.clean) == 1))

# First we need to clean up the isoforms again
df.all.KLK.lung <- data.frame(t(all.KLK.lung))

#test function 
cor.all.KLK.lung <- cor.genes(df.all.KLK.lung)
length(which(cor.all.KLK.lung == 1))
#64 entries
#Identical transcripts can cause issues later on for statistical testing etc. 

#remove identical isoforms
all.KLK.lung.clean <- df.all.KLK.lung[!duplicated(unclass(df.all.KLK.lung))]
dim(all.KLK.lung.clean) #removed 39 identical isoforms

# Use mixedsort function from the gtool package to sort the KLKs correctly
order.lung <- mixedsort(colnames(all.KLK.lung.clean))
order.lung <- order.lung[c(1:3,11:17,4:10,18,19,24:27,20:23,28:73)]
all.KLK.lung.clean <- t(all.KLK.lung.clean[,order.lung]) #sorted all.KLK.lung.clean
all.KLK.lung.clean <- data.frame(all.KLK.lung.clean)


order.lung1 <- mixedsort(colnames(TRA.KLK.lung.clean))
order.lung1 <- order.lung1[c(1:3,11:17,4:10,18,19,23:26,20:22,27:63)]
TRA.KLK.lung.clean <- t(TRA.KLK.lung.clean[,order.lung1])
TRA.KLK.lung.clean <- data.frame(TRA.KLK.lung.clean)


```

### Visualization of gene expression 
#### Overview gene expression
```{r Histogram - lung cancer}
gene.summary(lungExprs)

#Histograms of genes expression
setwd(here("Plots/Plots_lung"))
png("Histogram_gene_expression_lines.png")
hist(lungExprs, 
     main="Gene expression GSE149507 (Cai et al. 2021)",
     xlab="Expression value",
     xlim = c(4,15),
     breaks=40
     )
abline(v = median(lungExprs), col= "red", lwd = 3)#median all genes
abline(v = median(unlist(all.KLK.lung.clean)), col= "blue", lwd = 3) #median all KLKs
legend("topright", c("median all genes","median all KLKs"), col=c("red","blue"), lwd=10)
dev.off()
```

#### Boxplots 
```{r Boxplots - lung cancer}
#boxplots 
setwd(here("Plots/Plots_lung"))
png("all_KLK_boxplot_lung.png")
boxplot(t(all.KLK.lung.clean),
        col=rainbow(length(rownames(all.KLK.lung.clean))),
        main="Gene expression of KLKs GSE149507 (Cai et al. 2021)",
        cex.axis=0.4,
        las = 2,
        )
abline(h=median(unlist(all.KLK.lung.clean)), col= "red", lwd = 2, lty = 2)
legend("topright", c("median all KLKs"), col=c("red"), lwd=10)
dev.off()


#split the boxplots, because it is hard to read the gene names
#Boxplot1 
setwd(here("Plots/Plots_lung"))
png("all_KLK_boxplot_lung1.png")
boxplot(t(all.KLK.lung.clean[1:24,]),
        col=rainbow(length(rownames(all.KLK.lung.clean[1:24,]))),main="Gene expression of KLKs GSE149507 (Cai et al. 2021)",
        cex.axis=0.8,
        las = 2,
        )
abline(h=median(unlist(all.KLK.lung.clean)), col= "red", lwd = 2, lty = 2)
legend("topright", c("median all KLKs"), col=c("red"), lwd=10)
dev.off()

#Boxplot2
setwd(here("Plots/Plots_lung"))
png("all_KLK_boxplot_lung2.png")
boxplot(t(all.KLK.lung.clean[25:48,]),
        col=rainbow(length(rownames(all.KLK.lung.clean[25:48,]))),main="Gene expression of KLKs GSE149507 (Cai et al. 2021)",
        cex.axis=0.8,
        las = 2,
        )
abline(h=median(unlist(all.KLK.lung.clean)), col= "red", lwd = 2, lty = 2)
legend("topright", c("median all KLKs"), col=c("red"), lwd=10)
dev.off()

#Boxplot3
setwd(here("Plots/Plots_lung"))
png("all_KLK_boxplot_lung3.png")
boxplot(t(all.KLK.lung.clean[49:73,]),
        col=rainbow(length(rownames(all.KLK.lung.clean[49:73,]))),main="Gene expression of KLKs GSE149507 (Cai et al. 2021)",
        cex.axis=0.8,
        las = 2,
        )
abline(h=median(unlist(all.KLK.lung.clean)), col= "red", lwd = 2, lty = 2)
legend("topright", c("median all KLKs"), col=c("red"), lwd=10)
dev.off()
```

#### Heatmap
```{r Heatmaps annotation - lung cancer}
# We first want to see the dendrogram of the heatmap
# By that we can decide to put certain number of clusters into the heatmap
# This clustering is not meant for analytical purpose -> see k-means later on for this
# This clustering is just intended to increase the visual clarity


# 1. show simple cluster annotated to heatmap
hclust.genes.lung <- hclust(dist(all.KLK.lung.clean), method = "complete")

dendr.lung <- dendro_data(hclust.genes.lung, type = "rectangle")
clust.lung    <- cutree(hclust.genes.lung, k=3)
# we tested a few clustering´, however 3 cluster seem to visualize the expression in the more up-regulated genes better
clust.lung.df <- data.frame(label=names(clust.lung), cluster=factor(clust.lung))
dendr.lung[["labels"]] <- merge(dendr.lung[["labels"]],clust.lung.df, by="label")


setwd(here("Plots/Plots_lung"))
png("all_KLK_dendrogram_lung.png")
ggplot() +
  ggtitle("Dendrogram of all KLKs GSE149507 (Cai et al. 2021)") +
  ylab("Distance (complete-linkage)") +
  geom_segment(data=segment(dendr.lung), aes(x=x, y=y, xend=xend, yend=yend)) + 
  geom_text(data=label(dendr.lung), aes(x, y, label=label, hjust=0, color=cluster), 
           size=2) +
  coord_flip() + scale_y_reverse(expand=c(0.1, 0)) + 
  theme(axis.line.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.y=element_blank(),
        axis.title.y=element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        legend.position = c(0.05, 0.95),
        legend.justification = c("left", "top")
        )
dev.off()

setwd(here("Plots/Plots_lung"))
pdf("all_KLK_dendrogram_lung.pdf")
ggplot() +
  ggtitle("Dendrogram of all KLKs GSE149507 (Cai et al. 2021)") +
  ylab("Distance (complete-linkage)") +
  geom_segment(data=segment(dendr.lung), aes(x=x, y=y, xend=xend, yend=yend)) + 
  geom_text(data=label(dendr.lung), aes(x, y, label=label, hjust=0, color=cluster), 
           size=2) +
  coord_flip() + scale_y_reverse(expand=c(0.1, 0)) + 
  theme(axis.line.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.y=element_blank(),
        axis.title.y=element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        legend.position = c(0.05, 0.95),
        legend.justification = c("left", "top")
        )
dev.off()


# 3 clusters in the heatmap, just for visualization purposes
annotation.hmap.lung <- cutree(tree = as.dendrogram(hclust.genes.lung), k = 3)
annotation.hmap.lung  <- data.frame(cluster = ifelse(test = annotation.hmap.lung == 1, yes = "cluster 1", no = ifelse(test = annotation.hmap.lung == 2, yes = "cluster 2", no = "cluster 3" )))

# 2. differentiate between TRA-/ or Non-TRA-KLKs
TRUEorFALSE.lung <- rownames(all.KLK.lung.clean) %in% rownames(TRA.KLK.lung.clean)
TRAorNON.lung <- ifelse(TRUEorFALSE.lung == TRUE, "TRA", "non-TRA")
annotation.hmap.lung$TRA_or_NON  <- TRAorNON.lung
colnames(annotation.hmap.lung)[2] <- "TRA or not"

# 3. To increase visibility, we will cut off the GSM number for the patient names
all.KLK.lung.clean.noGSM <- all.KLK.lung.clean

colnames(all.KLK.lung.clean.noGSM)[1:12] <- sub("...........", "", colnames(all.KLK.lung.clean)[1:12])

# annotation for the samples
samples.annotation.lung <- data.frame(matrix(ncol = 1, nrow = ncol(all.KLK.lung.clean.noGSM)))
colnames(samples.annotation.lung) <- "sample"                                 
rownames(samples.annotation.lung) <- colnames(all.KLK.lung.clean.noGSM)
ph.ind.normal = grep(pattern = "n", colnames(all.KLK.lung.clean.noGSM))
ph.ind.cancer = grep(pattern = "ca", colnames(all.KLK.lung.clean.noGSM))
samples.annotation.lung[ph.ind.normal,1] <- "normal"
samples.annotation.lung[ph.ind.cancer,1] <- "cancer"

# call the annotated Heatmap
setwd(here("Plots/Plots_lung"))
png("all_KLK_Heatmap_lung.png")
pheatmap(all.KLK.lung.clean.noGSM,
         main = "Expression values of all KLKs GSE149507 (Cai et al. 2021)",
         annotation_row = annotation.hmap.lung,
         annotation_col = samples.annotation.lung,
         cellwidth = 20.6,
         cellheight = 5.3,
         treeheight_col = 10,
         fontsize_row = 6,
         cutree_rows = 3,
)
dev.off()

setwd(here("Plots/Plots_lung"))
pdf("all_KLK_Heatmap_lung.pdf")
pheatmap(all.KLK.lung.clean.noGSM,
         main = "Expression values of all KLKs GSE149507 (Cai et al. 2021)",
         annotation_row = annotation.hmap.lung,
         annotation_col = samples.annotation.lung,
         cellwidth = 20.6,
         cellheight = 5.3,
         treeheight_col = 10,
         fontsize_row = 6,
         cutree_rows = 3,
)
dev.off()
```



### PCA
```{r PCA - lung cancer}
# run PCA on the KLK data frame
# center and scale = TRUE ? does vsnrma not already do this?
pr.all.KLK.lung.clean <- prcomp(all.KLK.lung.clean, center = TRUE, scale = FALSE)
#call summary
summary(pr.all.KLK.lung.clean)
# goal is to get information of roughly 95% of the KLK data -> 95% cumulative variance
# in the summary we can see that this is almost achieved by the 4th pc
# the variance of the pcs rapidly declines, there is only about 5% variance for the 4th pc

# get graphical overview (not really needed though)

screeplot(pr.all.KLK.lung.clean, type = "l", npcs = 10, main = "Screeplot of the first 10 PCs")
abline(h = 1, col="red", lty=5)
legend("topright", legend=c("Eigenvalue = 1"),
       col=c("red"), lty=5, cex=0.6)

variance.lung <- cumsum(pr.all.KLK.lung.clean$sdev^2 / sum(pr.all.KLK.lung.clean$sdev^2))
plot(variance.lung[0:12], xlab = "PC #", ylab = "Amount of explained variance", main = "Cumulative variance plot")
abline(v = 3, col="blue", lty=5)
abline(h = 0.9525, col="blue", lty=5)
legend("topleft", legend=c("PC3"),
       col=c("blue"), lty=5, cex=0.6)

# we can see in this cumulative variance plot, that we only need about 3 principle components to achieve a a total variance of 95%
# This seems to be a very low number of pcs, the others do not represent a lot of information

# Plot the PCs

### GROUP AFTER MUTATION

color.lung = c(ifelse(annotation.hmap.lung[,2] == "TRA", "blue", "red"))

plot(pr.all.KLK.lung.clean$x[,1],pr.all.KLK.lung.clean$x[,2],
     xlab="PC1 (80,23%)",
     ylab = "PC2 (12,13%)",
     main = "PC1 / PC2 - plot",
     col = color.lung)
legend("topleft", legend=c("TRA", "non-TRA"),
       col=c("blue", "red"), lwd= 10, cex=0.6)

plot(pr.all.KLK.lung.clean$x[,1],pr.all.KLK.lung.clean$x[,3],
     xlab="PC1 (80,23%)",
     ylab = "PC3 (2,89%)",
     main = "PC1 / PC3 - plot",
     col = color.lung)
legend("topleft", legend=c("TRA", "non-TRA"),
       col=c("blue", "red"), lwd= 10, cex=0.6)


# Since almost all KLKs are TRAs, annotating the color does not really show anything

color.lung.cancer = c(ifelse(samples.annotation.lung == "cancer", yes = "blue", no = "red"))

plot(pr.all.KLK.lung.clean$x[,1],pr.all.KLK.lung.clean$x[,2],
     xlab="PC1 (80,23%)",
     ylab = "PC2 (12,13%)",
     pch = 18,
     main = "PC1 / PC2 - plot",
     col = color.lung.cancer)
legend("left", legend=c("TRA", "non-TRA"),
       col=c("blue", "red"), lwd= 10, cex=0.6)

plot(pr.all.KLK.lung.clean$x[,1],pr.all.KLK.lung.clean$x[,3],
     xlab="PC1 (80,23%)",
     ylab = "PC3 (2,89%)",
     pch = 18,
     main = "PC1 / PC3 - plot",
     col = color.lung.cancer)
legend("bottomleft", legend=c("cancer", "normal"),
       col=c("blue", "red"), lwd= 10, cex=0.6)


# NOW annotate the clustering after k-means 

```


### K-means
``` {r K-means - rawdata breast}

rawdataExprs.nodupl = unique(rawdataExprs)


#determining the optimal number of k clusters, preparing wssplot
wssplot.rawdata <- function(rawdataExprs.nodupl, nc=15, seed=1234)
{
  wss <- (nrow(rawdataExprs.nodupl)-1)*sum(apply(rawdataExprs.nodupl,2,var))
  for(i in 2:nc){
    set.seed(seed)
    wss[i] <- sum(kmeans(rawdataExprs.nodupl, centers = i)$withinss)}
  plot(1:nc, wss, type="b", xlab="Number of Clusters", ylab="Within groups sum of squares")
}

#wssplot
wssplot.rawdata(rawdataExprs.nodupl)

#apparently optimal Number of clusters is 2
# ist jetzt nicht squared aber muss ja wenn dann erst bei kmeans plot

#visualization, !!! caution package "factoextra" required
results.km <- kmeans(scale(rawdataExprs.nodupl),4, nstart=25)
results.km$cluster

fviz_cluster(results.km, data=rawdataExprs.nodupl, 
             palette =c("#2E9FDF","#00AFBB","#FC4E07", "#E7B800"), 
             geom = "point", 
             ellipse.type = "convex",
             ggtheme = theme_bw(),
)

#für klk aus breastrawdata 
#all.KLK:rawdata.clean, zuerst transformieren

tf.KLK.rawdata.clean = data.frame(t(all.KLK.rawdata.clean))



wssplot <- function(tf.KLK.rawdata.clean, nc=15, seed=1234)
{
  wss <- (nrow(tf.KLK.rawdata.clean)-1)*sum(apply(tf.KLK.rawdata.clean ,2,var))
  for(i in 2:nc){
    set.seed(seed)
    wss[i] <- sum(kmeans(tf.KLK.rawdata.clean , centers = i)$withinss)}
  plot(1:nc, wss, type="b", xlab="Number of Clusters", ylab="Within groups sum of squares")
}
wssplot(tf.KLK.rawdata.clean)


#approx 4 clusters

results.km <- kmeans(scale(tf.KLK.rawdata.clean),4, nstart=25)
results.km$cluster

fviz_cluster(results.km, data=tf.KLK.rawdata.clean, 
             palette =c("#2E9FDF","#00AFBB", "#FC4E07", "#E7B800"), 
             geom = "point", 
             ellipse.type = "convex",
             ggtheme = theme_bw()
)

#Cluster4: KLK8.8 , KLK1.3 , KLK1 , KLK4.4 

#elbow not clearly visible, trying 3 & 5 centers
#3 

results.km <- kmeans(scale(tf.KLK.rawdata.clean),3, nstart=25)
results.km$cluster

fviz_cluster(results.km, data=tf.KLK.rawdata.clean, 
             palette =c("#2E9FDF", "#FC4E07", "FF3399"), 
             geom = "point", 
             ellipse.type = "convex",
             ggtheme = theme_bw()
)

#here cluster 2 shows overlap with cluster 3, and also combines points that are really far away(in dim2)
#now 5:

results.km <- kmeans(scale(tf.KLK.rawdata.clean),5, nstart=25)
results.km$cluster

fviz_cluster(results.km, data=tf.KLK.rawdata.clean, 
             palette =c("#2E9FDF", "#FC4E07", "#FF66FF", "#0099FF", "#99CCFF"), 
             geom = "point", 
             ellipse.type = "convex",
             ggtheme = theme_bw()
)
#additional little cluster in direction of of dim2
#4 centers seems good
```

``` {r k-means for lung cancer dataset}
#determining the optimal number of k clusters, preparing wssplot

wssplot <- function(lungExprs, nc=15, seed=1234)
{
  wss <- (nrow(lungExprs)-1)*sum(apply(lungExprs,2,var))
  for(i in 2:nc){
    set.seed(seed)
    wss[i] <- sum(kmeans(lungExprs, centers = i)$withinss)}
  plot(1:nc, wss, type="b", xlab="Number of Clusters", ylab="Within groups sum of squares")
}
  
#wssplot
wssplot(lungExprs)

## bekomme code nicht zum laufen, Anmerkung für später: Anzahl cluster determinieren!!!
#code für cluster lungexprs
results.km <- kmeans(scale(lungExprs),4, nstart=25)
results.km$cluster

fviz_cluster(results.km, data=lungExprs, 
             palette =c("#2E9FDF","#00AFBB", "#FC4E07", "#E7B800"), 
             geom = "point", 
             ellipse.type = "convex",
             ggtheme = theme_bw()
)

#k-means analysis fpr klks from lungcancer
#transformieren
tf.all.KLK.lung.clean = data.frame(t(all.KLK.lung.clean.noGSM))

wssplot <- function(tf.all.KLK.lung.clean, nc=15, seed=1234)
{
  wss <- (nrow(tf.all.KLK.lung.clean)-1)*sum(apply(tf.all.KLK.lung.clean ,2,var))
  for(i in 2:nc){
    set.seed(seed)
    wss[i] <- sum(kmeans(tf.all.KLK.lung.clean , centers = i)$withinss)}
  plot(1:nc, wss, type="b", xlab="Number of Clusters", ylab="Within groups sum of squares")
}
wssplot(tf.all.KLK.lung.clean)

#approx ?? clusters

results.km <- kmeans(scale(tf.all.KLK.lung.clean),4, nstart=25)
results.km$cluster

fviz_cluster(results.km, data=tf.all.KLK.lung.clean, 
             palette =c("#2E9FDF","#00AFBB", "#FC4E07", "#E7B800"), 
             geom = "point", 
             ellipse.type = "convex",
             ggtheme = theme_bw()
)

#text
#text
```

``` {r k-means for breast cancer dataset}

#determining the optimal number of k clusters, preparing wssplot

wssplot <- function(breastExprs, nc=15, seed=1234)
{
  wss <- (nrow(breastExprs)-1)*sum(apply(breastExprs,2,var))
  for(i in 2:nc){
    set.seed(seed)
    wss[i] <- sum(kmeans(breastExprs, centers = i)$withinss)}
  plot(1:nc, wss, type="b", xlab="Number of Clusters", ylab="Within groups sum of squares")
}
  
#wssplot
wssplot(breastExprs)

## bekomme code nicht zum laufen, Anmerkung für später: Anzahl cluster determinieren!!!
#code für cluster breastExprs

results.km <- kmeans(scale(breastExprs),4, nstart=25)
results.km$cluster

fviz_cluster(results.km, data=breastExprs, 
             palette =c("#2E9FDF","#00AFBB", "#FC4E07", "#E7B800"), 
             geom = "point", 
             ellipse.type = "convex",
             ggtheme = theme_bw()
)


#k-means analysis fpr klks from breastcancer
#transformieren

tf.all.KLK.breast.clean = data.frame(t(all.KLK.breast.clean.noGSM))

wssplot <- function(tf.all.KLK.breast.clean, nc=15, seed=1234)
{
  wss <- (nrow(tf.all.KLK.breast.clean)-1)*sum(apply(tf.all.KLK.breast.clean ,2,var))
  for(i in 2:nc){
    set.seed(seed)
    wss[i] <- sum(kmeans(tf.all.KLK.breast.clean , centers = i)$withinss)}
  plot(1:nc, wss, type="b", xlab="Number of Clusters", ylab="Within groups sum of squares")
}
wssplot(tf.all.KLK.breast.clean)

#approx ?? clusters

results.km <- kmeans(scale(tf.all.KLK.breast.clean),4, nstart=25)
results.km$cluster

fviz_cluster(results.km, data=tf.all.KLK.breast.clean, 
             palette =c("#2E9FDF","#00AFBB", "#FC4E07", "#E7B800"), 
             geom = "point", 
             ellipse.type = "convex",
             ggtheme = theme_bw()
)

#später auskommentieren

```
###Hypothesis testing lung 
```{r Hypothesis testing}
#check the normality of the dataset with a QQ-plot
setwd(here("Plots/Plots_lung"))
png("QQPlot_lung.png")
qqnorm(breastExprs, main = "QQ-Plot GSE149507 (Cai et al. 2021)");qqline(breastExprs)
dev.off()
#The data is not normally distributed - we must use a non-parametric test!
#But in this dataset we have the the gene expression values of the genes in non-cancer and cancer tissues.
#Therefore this time we do not use the Mann-Whitney-U test (unpaired Wilcoxon rank sum test), but the Wilcoxon signed rank test, with 2 paired probes. 

#In the heatmap the genes of cluster 3, except KLK4.4) seem to be higher expressed in the cancer sample than in the healthy sample. 
#Cluster 3: KLK11, KLK11.2, KLK12, KLK12.3, KLK12.4, KLK12.5, KLK12.6

#new function to check for each gene weather their are higher expressed in the cancer probes for multiple KLKs

#Sub dataset with all cancer mikrochips
all.KLK.lung.clean.ca <- all.KLK.lung.clean[, grepl("ca", colnames(all.KLK.lung.clean))]
all.KLK.lung.clean.ca

#Sub dataset with all normal tissue mikrochips
all.KLK.lung.clean.n <- all.KLK.lung.clean[, grepl("n", colnames(all.KLK.lung.clean))]
all.KLK.lung.clean.n

cluster3.lung.names <- c("KLK11", "KLK11.2", "KLK12", "KLK12.3", "KLK12.4", "KLK12.5", "KLK12.6")

results <- matrix(data=NA, nrow=7, ncol=3)
results[ ,1:2] <- cluster3.lung.names
colnames(results) <- c("ca", "n", "p.value")
results

for( ca in all.KLK.lung.clean.ca[cluster3.lung.names
                                , ]){
  for( n in all.KLK.lung.clean.n[cluster3.lung.names
                                 , ]){
    if (identical(rownames(ca),rownames(n))){
      print(wilcox.test(ca, n, 
               alternative = "greater",
               paired = TRUE,
               exact = NULL)$p.value)
    }
    else{break} 
  }
}

wilcox.test(as.matrix(all.KLK.lung.clean.ca["KLK11", ]) ,as.matrix(all.KLK.lung.clean.n["KLK11", ]),
            paired= TRUE,                                         alternative = "greater")

median(as.matrix(all.KLK.lung.clean.ca["KLK11", ]))
median(as.matrix(all.KLK.lung.clean.n["KLK11", ]))




dim(all.KLK.lung.clean.ca)




```
```{r, Backup code testing }

# <- lungExprs[grepl("ENST", colnames(lungExprs)), ]

w.test.gene.df.upper <- function(gene, df){
  apply(as.matrix(df), 1, function(x){
  test <- wilcox.test(as.matrix(gene), x,  
                      alternative = "greater", 
                      exact = FALSE, 
                      paired = FALSE,
                      digits.rank = 10)$p.value})

}




t.data <- adply(combinations, 2, function(x) {
  test <- t.test(all.KLK.rawdata.clean[x[1], ],
                 all.KLK.rawdata.clean[x[2], ],
                 alternative = "two.sided") 

  



  out <- rbind.fill(data.frame(
    "Var1" = colnames(all.KLK.rawdata.clean[ ,x[1]]),
    "Var2" = colnames(all.KLK.rawdata.clean[ ,x[2]]),
                    "W.value"= test$statistic,
                    "df"= test$parameter,
                    "p.value" = test$p.value
                    ))
   
  return(out)
  })
```



#Piechart: Distribution KLK TRA
```{r Piechart TRA}

#convert dataframe into vector first
#we will work with Union.TRA dataframe, since KLK type and tissue type is already given

KLK.TRA.tissue = Union.TRA[,5]       

KLK.TRA.table = table(KLK.TRA.tissue)      

KLK.TRA.frame = as.data.frame(KLK.TRA.table)  

#in this frame we still include the mouse TRA data - shall we only look at human?
#skin adjust
KLK.TRA.frame$Freq[14] = 6
KLK.TRA.frame <- KLK.TRA.frame %>% slice(-c(15,16))

#prostate adjust
KLK.TRA.frame$Freq[12] = 46
KLK.TRA.frame <- KLK.TRA.frame %>% slice(-c(13))

#liver adjust
KLK.TRA.frame$Freq[6] = 7
KLK.TRA.frame <- KLK.TRA.frame %>% slice(-c(5))

#esophagus adjust
KLK.TRA.frame$Freq[3] = 16
KLK.TRA.frame <- KLK.TRA.frame %>% slice(-c(4))

#oral mucosa adjust
KLK.TRA.frame$Freq[7] = 15
KLK.TRA.frame <- KLK.TRA.frame %>% slice(-c(9))

#salivary gland adjust(oral mucosa is salivary gland)
KLK.TRA.frame$Freq[5] = 16
KLK.TRA.frame <- KLK.TRA.frame %>% slice(-c(7))

#dataframe needs to be table so pie can work, converting:
piechart.vec = KLK.TRA.frame[,2]
names.piechart = c("Cervix","Colon","Esophagus","Liver","Minor Salivary Gland", "nodose_nucleus","Pancreas",
                   "Prostate", "Skin", "Snoutepidermis", "Spinal Cord", "Testis", "Thyroid", "Tongueepidermis","Vulva")


#expanding color palette
coul <- brewer.pal(10, "Paired")
coul <- colorRampPalette(coul)(25)

#pieeee
pie(piechart.vec, labels = names.piechart,
    edges = 200, radius = 0.9, col=coul ,
    main = "Distribution of KLK-TRA genes over tissue",
    cex =0.55)


```










